{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CARD algorithm\n"
     ]
    }
   ],
   "source": [
    "XH = [1, 3]\n",
    "ZL = 10\n",
    "from SY import *\n",
    "HX.LB(XH, ZL)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Todolist\n",
    "\n",
    "\n",
    "\n",
    "mIE4 VZV fails to downregulate MHC-I on the cell surface\n",
    "\n",
    "CRISPR\n",
    "\n",
    "\n",
    "Upenn递交\n",
    "哥大递交\n",
    "Cornell\n",
    "\n",
    "# CRISPR\n",
    "\n",
    "Expression,purification and characterization of HFBI protein\n",
    "(1)Large-scale Expressionand Purification ofHFBI Protein[17\n",
    "Add20μL of HFBI glycerol stockto 5mL of YPD medium in a small tube and incubate at 30℃,280 rpm in a shaker for approximately  12 hours.When the OD value reaches around 2-6, transfer the entire culture from the small tube to a large bottle containing 500 mL of BMG under sterile  conditions.Continue  shaking  incubation   at  30℃,280  rpm,and  monitor  the   OD  value. When the OD value reaches around 2-6,centrifuge at 4000 rpm and 4℃ to collect the cell pellet, and transfer an appropriate amount of the cell pellet to a 500 mL BMM bottle for the induction of hydrophobic protein expression(the OD value  should be  1  at  this point).After transferring the culture to the BMM bottle,add  1.5mL  of anhydrous methanol to the bottle  every  18  hours  to induce  protein  expression,continuing  for  110  hours.Then,centrifuge  the  culture  supernatant  at\n",
    "\n",
    "4000 rpm and 4℃ to collect it,and perform ultrafiltration and concentration using a large-volume ultrafiltration  column.Next,lyophilize  the  concentrated  supernatant  using  a  freeze  dryer  in preparation   for   purification   by   reverse-phase   high-performance   liquid   chromatography (RP-HPLC).\n",
    "Purify the hydrophobic protein HFBI using a C4 hydrophobic reverse-phase chromatography column.Dissolve  the  lyophilized  protein  powder  in  40%acetonitrile(0.1%trifluoroacetic  acid), centrifuge at 16000 g and 4℃ for 30 minutes,and filter the supernatant.Load the resulting protein solution   onto    the   AKTA    system   and    perform   gradient    elution   with    acetonitrile(0.1% trifluoroacetic acid)at a flow rate of 1 mL/min.Monitor the elution peaks at UV 215 nm,collect the eluted protein fractions,and verify the size and purity of the collected protein samples using Tricine-SDS-PAGE.Place  the  protein   solution  under   a  vacuum  pump  to   evaporate  residual acetonitrile,then lyophilize the remaining solution using a freeze dryer to obtain protein powder, which is stored at room temperature under vacuum.\n",
    "\n",
    "# 1 sc\n",
    "\n",
    "  - Covariates in sample processing\n",
    "\n",
    "    - Sample preparation\n",
    "      - first\n",
    "      - least standardized\n",
    "\n",
    "    - Tissue composition change\n",
    "      - time-dependent\n",
    "        - min to h\n",
    "      - immune cell extravasation\n",
    "      - vulnerable cells apoptosis\n",
    "\n",
    "      - RNA synthesis shut down\n",
    "        - reducing detectable transcripts\n",
    "        - nonrandom transcriptional alterations occur\n",
    "        - tissue-specific variation in mitochondrial RNA content\n",
    "        - splicing dysregulation\n",
    "        - increased alternative isoforms\n",
    "        - gene inactivation\n",
    "          - carbohydrate metabolism\n",
    "          - immune responses\n",
    "          - cell cycling\n",
    "          - stress responses\n",
    "          - cell necrosis\n",
    "\n",
    "\n",
    "    - intrinsic features\n",
    "      - Chemicals in tissues\n",
    "        - lyse cells\n",
    "        - degrade RNA\n",
    "        - e.g.\n",
    "          - stomach acid\n",
    "          - bile acid\n",
    "          - intestinal enzyme\n",
    "          - pancreatic enzyme\n",
    "      - Adherent cells\n",
    "        - fibroblasts\n",
    "        - difficult to detach\n",
    "      - large filopodia cells\n",
    "        - macrophages\n",
    "        - adipocytes\n",
    "      - short-lived cells\n",
    "        - neutrophils\n",
    "        - missed\n",
    "        - filtered-out\n",
    "\n",
    "      - Enzymatic digestion\n",
    "        - degrades surface antigens\n",
    "          - reducing antibody-mediated detectability\n",
    "            - flow cytometry\n",
    "            - CITE-seq\n",
    "            - REAP-seq\n",
    "\n",
    "\n",
    "    - Experimental strategies\n",
    "      - mitigate these dissociation biases\n",
    "      - introduced new confounders in exchange\n",
    "\n",
    "      - Single-nucleus RNA-seq (snRNA-seq) \n",
    "        - isolates nuclear RNA\n",
    "          - frozen\n",
    "          - difficult-to-dissociate tissues\n",
    "            - neurons\n",
    "            - glial cells\n",
    "          - require customized protocols\n",
    "            - to minimize dissociation biases\n",
    "\n",
    "\n",
    "        - susceptible to contamination\n",
    "        - excludes key cytoplasmic transcripts\n",
    "          - involved in microglial activation\n",
    "          - regulatory RNA stored in processing bodies\n",
    "          - nuclear RNA during mitosis\n",
    "            - lack nuclear membrane\n",
    "\n",
    "      - Chemical fixation\n",
    "        - preserve transcriptional contents\n",
    "        - corrupts cells ultrastructure\n",
    "          - challenging to whole-transcriptome seq\n",
    "\n",
    "        - 10x Genomics Flex Kit\n",
    "          - from fixed tissues\n",
    "          - predetermined probe\n",
    "            - mouse\n",
    "            - human\n",
    "          - minimizes effect\n",
    "            - sample processing\n",
    "            - technical batch\n",
    "          - RNA quantification\n",
    "            - limited to curated genomic regions\n",
    "            - nuclei remains susceptible\n",
    "\n",
    "        - ST\n",
    "          - preserve tissue context\n",
    "          - without dissociation\n",
    "\n",
    "    - minimizing experimental variability\n",
    "      - Reducing sample preparation time\n",
    "      - optimizing digestion buffers\n",
    "      - developing community standards\n",
    "    \n",
    "  - Sample integration & batch correction\n",
    "    \n",
    "    - Factor\n",
    "      - sequencing-related factors\n",
    "        - library size\n",
    "        - PCR cycles\n",
    "        - sequencing instrument\n",
    "        - resulting in sc clustering\n",
    "        - technical batch ＞ biological similarity\n",
    "      \n",
    "      - Other factors\n",
    "        - sampled individuals\n",
    "          - age\n",
    "          - sex\n",
    "          - ancestry\n",
    "        \n",
    "        - heterogeneity in features\n",
    "          - cell cycle stage\n",
    "          - mitochondrial RNA content\n",
    "\n",
    "    - Linear adjustment models\n",
    "\n",
    "      - microarray\n",
    "      - bulk RNA-seq data\n",
    "      - e.g.\n",
    "        - ComBat\n",
    "        - limma\n",
    "        \n",
    "      - challenge\n",
    "        - sparsity稀疏性\n",
    "        - variability\n",
    "\n",
    "    - MNN\n",
    "      - Batch effect\n",
    "      - Mutual nearest neighbours\n",
    "      - Identify similar neighbours\n",
    "      - Integrate batches\n",
    "      - Correct gene expression\n",
    "\n",
    "    - Reduce dimension\n",
    "      - gene expression profiles\n",
    "      - expedite search for shared elements\n",
    "      - improve robustness to noise\n",
    "\n",
    "      - e.g.\n",
    "        - principal components analysis\n",
    "          - fastMNN\n",
    "          - BBKNN\n",
    "          - Harmony\n",
    "\n",
    "        - singular value decomposition\n",
    "          - Scanorama\n",
    "\n",
    "        - integrative non-negative matrix factorization\n",
    "          - LIGER\n",
    "\n",
    "        - canonical correlation analysis, CCA\n",
    "          - Seurat\n",
    "\n",
    "      - require high input\n",
    "      - parameter tuning\n",
    "\n",
    "      - challenge\n",
    "        - accurate correct high-dimensional data\n",
    "        - batch-corrected gene expression analysis\n",
    "          - not desirable\n",
    "          - type II error\n",
    "            - eliminating true biological differences\n",
    "\n",
    "          - type I error\n",
    "            - introducing spurious differentially expressed genes\n",
    "\n",
    "    - Differential gene-expression analysis\n",
    "\n",
    "      - Pseudobulk analysis\n",
    "        - aggregates gene expression\n",
    "          - individual → group-level\n",
    "\n",
    "        - minimizes\n",
    "          - sparsity\n",
    "          - noise\n",
    "          - collinearity among\n",
    "        - ignores heterogeneity\n",
    "          - cell clusters\n",
    "          - cell abundance\n",
    "          - inter-batch variability\n",
    "        - e.g.\n",
    "          - DESeq2\n",
    "          - edgeR\n",
    "      \n",
    "      - Mixed-effects modelling\n",
    "        - estimate different factors\n",
    "          - sc\n",
    "          - batch covariates\n",
    "\n",
    "        - Nonparametric methods\n",
    "          - Wilcoxon rank-sum test\n",
    "            - × assume underlying distribution\n",
    "            - limitation\n",
    "              - simple experimental design\n",
    "              - two-group comparisons\n",
    "\n",
    "        - gene-level meta-analysis statistics\n",
    "          - across groups\n",
    "          - scenarios\n",
    "            - batches contain\n",
    "              - case samples\n",
    "              - control samples\n",
    "            - differential gene expression analysis\n",
    "              - within samples\n",
    "              - rather than across samples\n",
    "          - e.g.\n",
    "            - MAST\n",
    "            - muscat\n",
    "\n",
    "        - Clustering-free strategies\n",
    "          - emerged identify differentially abundant cell states\n",
    "          - across batches\n",
    "          - × need for predefining cell subpopulations\n",
    "          - nebula\n",
    "            - fast negative binomial mixed model\n",
    "            - differential or co-expression analysis\n",
    "            - large-scale multi-subject sc\n",
    "\n",
    "    - Neural network-based\n",
    "      - improve scalability\n",
    "      - aligning large batches speed\n",
    "      - low-dimensional embedding space\n",
    "      - original gene expression space\n",
    "\n",
    "      - multi-tasking techniques\n",
    "        - data normalization\n",
    "        - clustering\n",
    "        - dimensional reduction\n",
    "        - differential gene expression analysis\n",
    "          - integrated space\n",
    "          - corrected space\n",
    "\n",
    "      - risk overcorrecting\n",
    "      - eliminating desired signal\n",
    "        - strategy to preserve signal\n",
    "          - semi-supervised methods\n",
    "            - informed by cell annotations\n",
    "            - e.g.\n",
    "              - scANVI\n",
    "              - scGEN\n",
    "            - outperform unsupervised methods\n",
    "              - batch correction\n",
    "              - bio variation preservation\n",
    "      - e.g.\n",
    "        - scVI\n",
    "        - CarDEC\n",
    "        - Pareto MTL\n",
    "        - scGen\n",
    "          - predict sc perturbation response\n",
    "        - scGAN\n",
    "          - generative adversarial network\n",
    "        - scANVI\n",
    "\n",
    "    - Deficiency\n",
    "      - distinguish biological signals\n",
    "      - from confounding variables\n",
    "      - proactive inclusion of technical control groups\n",
    "        - pool-of-controls\n",
    "          - CellANOVA\n",
    "            - Signal recovery in batch correction\n",
    "\n",
    "  - Large sc atlases\n",
    "\n",
    "    - large cell atlas\n",
    "      - 190 tissues\n",
    "      - 76 species\n",
    "\n",
    "    - goal\n",
    "      - annotated\n",
    "      - publicly accessible dictionary\n",
    "        - cell types\n",
    "        - states\n",
    "      - reference\n",
    "        - cell labelling\n",
    "        - transcriptional signatures\n",
    "        - disease association\n",
    "        - hypothesis generation\n",
    "\n",
    "    - challenge\n",
    "      - quality control\n",
    "      - data integration\n",
    "        - across variable sample processing\n",
    "      - sequencing modalities persist\n",
    "\n",
    "    - address\n",
    "      - Human Tumor Atlas Network\n",
    "\n",
    "      - annotating sc data use ref\n",
    "      - marker gene detection\n",
    "        - CellAssign\n",
    "        - Garnett\n",
    "        - ScType\n",
    "        - scMAP\n",
    "        - scGSEA\n",
    "        - SciBet\n",
    "      - gene enrichment analysis\n",
    "      - correlation-based techniques\n",
    "        - SingleR\n",
    "        - scmap\n",
    "        - CHETAH\n",
    "        - CellAnn\n",
    "      - general-purpose classifiers\n",
    "        - scID\n",
    "        - scClassify\n",
    "        - scGraph\n",
    "        - scPred\n",
    "      - maximum-likelihood estimators\n",
    "\n",
    "    - Large language models\n",
    "      - GPT4\n",
    "      - annotate based on\n",
    "        - prompts chain\n",
    "        - user-provided marker genes\n",
    "      - GPTCelltype\n",
    "\n",
    "    - Transfer learning strategies\n",
    "      - map annotations \n",
    "        - from single-cell references\n",
    "        - to small-scale datasets\n",
    "      - discover disease-specific cell states\n",
    "        - individuals with COVID-19\n",
    "        - idiopathic pulmonary fibrosis\n",
    "      - e.g.\n",
    "        - treeArches\n",
    "        - scArches\n",
    "\n",
    "    - Challenges in curation\n",
    "      - sc reference need\n",
    "      - more detailed cell-level\n",
    "      - sample-level annotations\n",
    "      - unified cell state ontology establishment\n",
    "        - various tissues\n",
    "        - organs\n",
    "        - disease conditions\n",
    "      - precise alignment\n",
    "        - query datasets\n",
    "        - to condition-matched reference atlases\n",
    "\n",
    "    - potential option\n",
    "      - assign cell to ‘novel’ class\n",
    "\n",
    "  - Identifying rare, subtle event\n",
    "\n",
    "    - discovery\n",
    "      - tissue-resident stem cells\n",
    "      - progenitor cells\n",
    "      - immune disease drivers\n",
    "      - tumour cell lineage states\n",
    "  \n",
    "    - challenges\n",
    "\n",
    "      - harsh dissociation\n",
    "        - eliminate vulnerable cells\n",
    "        - induce stress-related changes\n",
    "          - activate quiescent muscle stem cells\n",
    "        - loss cell populations\n",
    "          - poor characterization \n",
    "            - quiescent stem cell population\n",
    "          - misclassification\n",
    "            - dissociation artifact \n",
    "            - as new cell state\n",
    "\n",
    "      - frequency within population\n",
    "\n",
    "        - Tissue-resident stem cells\n",
    "          - exceptionally rare\n",
    "          - miss\n",
    "            - large-scale scRNA-seq\n",
    "            - without targeted enrichment strategy\n",
    "\n",
    "        - Hoxb5+\n",
    "\n",
    "          - frequency\n",
    "            - 0.001%\n",
    "            - in bone marrow\n",
    "\n",
    "          - enrichment\n",
    "            - specific cell-surface markers\n",
    "              - KIT\n",
    "              - SCA-1\n",
    "            - depleting mature markers cells\n",
    "            - increased 1,000-fold\n",
    "\n",
    "          - Other distribution features\n",
    "            - surface markers\n",
    "            - anatomic location\n",
    "            - embryonic stage\n",
    "            - human neural and skeletal stem cells\n",
    "\n",
    "    - modest but reproducible difference\n",
    "      - cell cycle\n",
    "      - differentiation trajectory\n",
    "      - metabolic state\n",
    "\n",
    "    - plate-based full-length sequencing\n",
    "      - vs droplet-based short-read sequencing\n",
    "        - costlier\n",
    "        - time-consuming\n",
    "\n",
    "      - RNA capture\n",
    "        - deeper\n",
    "        - broader\n",
    "        - from individual cells\n",
    "\n",
    "      - e.g.\n",
    "        - MAS-seq\n",
    "        - VASA-seq\n",
    "      - expanded transcriptome coverage\n",
    "        - non-coding\n",
    "        - alternatively spliced transcript isoforms\n",
    "      - revealed uncommon transcripts\n",
    "        - tumour-infiltrating T cell differentiation\n",
    "        - mouse development\n",
    "\n",
    "\n",
    "\n",
    "    - analysis pipeline\n",
    "      \n",
    "      - challenge\n",
    "        - skewed by irrelevant covariates\n",
    "        - insensitive to biological signals\n",
    "        - complexity and diversity of datasets\n",
    "          - size\n",
    "          - batches\n",
    "      \n",
    "      - Cells, rare, subtle\n",
    "        - clustering-based \n",
    "          - hierarchical clustering\n",
    "            - RaceID393\n",
    "          - inter-cluster distance\n",
    "            - GapClust95\n",
    "        - density-based\n",
    "          - FiRE94\n",
    "\n",
    "      - Marker genes, rare, subtle\n",
    "        - combined with clustering strategies ??\n",
    "        - GiniClust3\n",
    "          - Gini coefficient\n",
    "            - measures gene expression inequality\n",
    "\n",
    "        - CellSIUS\n",
    "          - generating highly correlated\n",
    "            - bimodal distribution\n",
    "            - differential expression\n",
    "            - in pre-defined cluster\n",
    "\n",
    "        - scPNMF\n",
    "          - reveal gene expression patterns\n",
    "            - projective non-negative matrix factorization (NMF)\n",
    "\n",
    "          - select biologically relevant gene\n",
    "            - correlation screening\n",
    "            - multimodal distributions\n",
    "\n",
    "\n",
    "      - false discovery risk\n",
    "        - same data used\n",
    "          - clustering \n",
    "          - before differential expression analysis\n",
    "        - statistical + aforementioned tools\n",
    "\n",
    "      - Deep generative models\n",
    "        - scLDS2\n",
    "        - learn sc distribution\n",
    "        - utilize interpretable classifier\n",
    "          - infer rare cells\n",
    "        \n",
    "      - SEAcells101\n",
    "        - adaptive Gaussian kernel\n",
    "          - capture major variation sources\n",
    "          - applies archetypal analysis \n",
    "            - over dimensionally reduced graph\n",
    "        - identify\n",
    "          - biologically coherent cell groups\n",
    "          - meta-cells\n",
    "    \n",
    "  - sc Trace\n",
    "\n",
    "    - dynamic bio processes\n",
    "      - development\n",
    "      - differentiation\n",
    "      - cell cycle\n",
    "      - responses to genomic perturbations\n",
    "      - responses to metabolic perturbations\n",
    "\n",
    "    - Trajectory inference\n",
    "\n",
    "      - step\n",
    "        - generate low-dimensional embedding\n",
    "        - nodes identified\n",
    "          - graph-based\n",
    "          - clustering-based\n",
    "        - path delineation\n",
    "\n",
    "\n",
    "      - challenge\n",
    "        - susceptible to unwanted covariates\n",
    "          - dimensional reduction\n",
    "          - clustering\n",
    "        \n",
    "        - rely on accuracy\n",
    "          - data integration\n",
    "          - batch correction\n",
    "\n",
    "        - more dynamic processes together\n",
    "          - e.g.\n",
    "            - cell cycle\n",
    "            - differentiation\n",
    "\n",
    "          - reduce into 1 part\n",
    "            - draw trajectories without connections\n",
    "              - unrelated cells\n",
    "              - static cells\n",
    "\n",
    "          - require \n",
    "            - significant user engagement\n",
    "            - parameter tuning\n",
    "\n",
    "      - transcriptional evolution direction\n",
    "\n",
    "        - RNA velocity analyse\n",
    "          - rate\n",
    "          - direction\n",
    "        - by comparing\n",
    "          - newly transcribed pre-mRNA\n",
    "          - to mature mRNA\n",
    "        - incorporates protein velocity\n",
    "          - give richer insight\n",
    "\n",
    "        - molecular kinetics\n",
    "          - metabolite labelling\n",
    "            - nascent RNA\n",
    "              - NASC-seq\n",
    "              - scSlam-seq\n",
    "\n",
    "        - capture cell transitions\n",
    "          - short time scale\n",
    "            - min to h\n",
    "          - longer time scale\n",
    "            - d to m\n",
    "            - less reliable\n",
    "\n",
    "        - Integration\n",
    "          - epigenomic\n",
    "          - proteomic data\n",
    "          - guide transitions direction\n",
    "            - chromatin accessibility\n",
    "            - protein expression\n",
    "\n",
    "      - lineage tracing\n",
    "        - sc resolution barcoding\n",
    "          - offer ground-truth information\n",
    "\n",
    "          - engineered genomic labels\n",
    "            - recombination events\n",
    "              - PolyLox\n",
    "            - viral integration\n",
    "              - TraceSeq\n",
    "            - CRISPR-based barcoding\n",
    "            - natural markers\n",
    "              - somatic mutations\n",
    "              - mitochondrial DNA variants\n",
    "              - DNA methylation patterns\n",
    "\n",
    "    - Measuring cellular plasticity\n",
    "\n",
    "      - Cellular plasticity \n",
    "        - potential fates diversity\n",
    "        - genomic alterations independently\n",
    "        - normal cellular responses to disease \n",
    "          - diverse T cell states in infection\n",
    "\n",
    "        - cancer cells adaptation\n",
    "          - during tumorigenesis\n",
    "          - therapy evasion\n",
    "\n",
    "      - probabilistic process\n",
    "        - cell reach terminal points\n",
    "        - in low-dimensional emb\n",
    "\n",
    "      - infer complex topographies\n",
    "        - Markov chain model\n",
    "          - Palantir\n",
    "          - VIA\n",
    "          - MARGARET\n",
    "\n",
    "      - FateID\n",
    "        - random forest classifier\n",
    "        - trained on pre-defined terminal states\n",
    "\n",
    "      - GpFates\n",
    "        - overlapping Gaussian processes mixture\n",
    "\n",
    "      - STEMNET\n",
    "        - generalized linear regression model\n",
    "          - coefficient weights\n",
    "            - contributions to pre-defined terminal states\n",
    "\n",
    "      - Pitfall\n",
    "        - reliance on prior knowledge\n",
    "          - start point\n",
    "          - end point\n",
    "\n",
    "        - CellRank\n",
    "          - ensemble unbiased tools\n",
    "            - RNA velocity\n",
    "            - CytoTRACE\n",
    "          - infer \n",
    "            - origins\n",
    "            - destinations\n",
    "          - Markov chain model\n",
    "            - compute fate probabilities\n",
    "\n",
    "      - time series data\n",
    "        - Waddington-OT\n",
    "          - optimal transport models\n",
    "        - rajectoryNet\n",
    "          - optimal transport approach\n",
    "          - continuous trajectories\n",
    "          - nonlinear trajectories\n",
    "\n",
    "      - master regulatory genes\n",
    "        - CellOracle\n",
    "          - leverages gene regulatory networks\n",
    "          - derived from sc multi-omics analyses\n",
    "          - in silico transcription-factor perturbations\n",
    "\n",
    "  - Developmental potential\n",
    "\n",
    "    - plasticity category\n",
    "      - rise to more differentiated cells\n",
    "\n",
    "      - zygote\n",
    "        - rise to entire organism\n",
    "        - highest developmental potential\n",
    "\n",
    "      - neutrophils\n",
    "        - terminally differentiated cells\n",
    "        - lowest developmental potential\n",
    "\n",
    "      - hallmark\n",
    "        - stem cells\n",
    "        - progenitor cells \n",
    "        - during\n",
    "          - tissue growth\n",
    "          - homoeostasis\n",
    "          - regeneration\n",
    "\n",
    "\n",
    "\n",
    "    - predicting strategy\n",
    "      - measuring active transcriptional programmes\n",
    "      - during early embryogenesis\n",
    "\n",
    "      - ORIGINS\n",
    "        - protein–protein network\n",
    "          - derived from Gene Ontology\n",
    "            - Cell Differentiation Biological Process\n",
    "\n",
    "      - mRNAsi\n",
    "        - logistic regression model\n",
    "        - trained on gene expression data\n",
    "        - from pluripotent stem cells\n",
    "\n",
    "      - PluriNet\n",
    "        - gene set \n",
    "          - derived from protein–protein network\n",
    "          - shared by pluripotent stem cells\n",
    "\n",
    "\n",
    "\n",
    "    - post-embryogenic developmental processes\n",
    "      - assumes\n",
    "        - same programmes\n",
    "          - maintain pluripotency \n",
    "            - embryonic stem cells\n",
    "          - also govern plasticity\n",
    "            - adult normal cells\n",
    "            - tumour cells\n",
    "\n",
    "      - FitDevo \n",
    "        - expand scope\n",
    "        - training generalized linear model\n",
    "          - fetal cells\n",
    "          - adult cells \n",
    "          - during differentiation\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "    - transcriptional diversity\n",
    "\n",
    "      - high chromatin accessibility\n",
    "        - high transcriptional diversity\n",
    "        - high developmental potential\n",
    "\n",
    "      - CytoTRACE\n",
    "        - total genes expressed\n",
    "        - cellular ontogeny\n",
    "          - from fertilized zygote \n",
    "          - to terminally differentiated cells\n",
    "\n",
    "        - human breast cancer data\n",
    "          - identified population\n",
    "            - tumorigenic luminal progenitor cells \n",
    "            - characterized by GULP1 expression\n",
    "\n",
    "      - ATAC-seq\n",
    "        - assay for transposase-accessible chromatin with sequencing \n",
    "        - finding less-differentiated cells \n",
    "          - in normal tissues\n",
    "\n",
    "      - computing Shannon entropy\n",
    "        - popular uncertainty or randomness in data\n",
    "\n",
    "        - varied in data representation\n",
    "          - entropy across entire transcriptome\n",
    "          - the most highly expressed genes\n",
    "          - curated gene–gene network\n",
    "          - gene set–gene set network\n",
    "          - protein–protein network\n",
    "\n",
    "\n",
    "      - flexible machine learning model\n",
    "        - publicly available dataset\n",
    "          - time point\n",
    "          - ontogenetic annotations\n",
    "        - learns latent features in scRNA-seq data\n",
    "        - improves predicting developmental potential\n",
    "\n",
    "# 2 ST\n",
    "\n",
    "# 4 VZV\n",
    "\n",
    "  - Ectopically IE4 + HLA-C\n",
    "\n",
    "    - yeast two-hybrid assay\n",
    "      - candidate target protein\n",
    "      - HLA-C\n",
    "    \n",
    "    - immunofluorescence assays\n",
    "      - HeLa + IE4-YFP\n",
    "        - cytoplasmic punctate localization\n",
    "      - HeLa + HLA-C-Red\n",
    "        - located in cytoplasm\n",
    "      - HeLa cotransfected\n",
    "        - IE4\n",
    "        - HLA-C\n",
    "        - in cytoplasm\n",
    "        - punctate staining patterns\n",
    "      - 24h after transfection\n",
    "      - 40× oil objective\n",
    "\n",
    "    - Co-IP\n",
    "      - HEK293T cotransfected\n",
    "        - IE4-flag\n",
    "        - HLA-C-HA\n",
    "      - 36h after transfection\n",
    "      - anti-flag mAb (IP: flag)\n",
    "      - nonspecific mAb (IgG)\n",
    "      - IE4 interact HLA-C\n",
    "\n",
    "  - HLA-C + IE4 + Golgi \n",
    "    \n",
    "    - previous study\n",
    "      - VZV\n",
    "        - retains MHC-I in Golgi\n",
    "        - downregulate cellular surface MHC-I\n",
    "    \n",
    "    - IE4\n",
    "      - punctate distribution\n",
    "      - hypothesize\n",
    "        - IE4 target MHC-I to Golgi\n",
    "    \n",
    "    - immunofluorescence assays\n",
    "    \n",
    "      - HeLa, IE4-Red, Golgi-YFP\n",
    "        - IE4, Golgi co-localized\n",
    "      - HeLa, HLA-C-Red, Golgi-YFP\n",
    "        - HLA, Golgi × co-localized\n",
    "      - HeLa, IE4-HA, HLA-C-Red, Golgi-YFP\n",
    "        - HLA sequestered in Golgi\n",
    "\n",
    "    - flow cytometry\n",
    "    \n",
    "      - MeWo + IE4-flag\n",
    "      \n",
    "        - stained with APC-HLA\n",
    "          - APC MFI\n",
    "            - downregulated MHC surface distribution\n",
    "            - n>3\n",
    "            - P<0.05\n",
    "    \n",
    "    - WB\n",
    "      - HEK293T + IE4-HA\n",
    "        - not affect total HLA-C\n",
    "\n",
    "  - mIE4 × ↓ MHC-I\n",
    "\n",
    "    - Program UniProt\n",
    "      - potential Golgi localization signal\n",
    "      - 105-DVE-107 to 105-AVA-107\n",
    "      - abrogated distribution\n",
    "\n",
    "    - immunofluorescence assays\n",
    "      \n",
    "      - HeLa, IE4-Red, Golgi-YFP\n",
    "        - IE4, Golgi co-localized\n",
    "      \n",
    "      - HeLa, mIE4-Red, Golgi-YFP\n",
    "        - mIE4, Golgi × co-localized\n",
    "      \n",
    "      - HeLa, mIE4-YFP, HLA-C-Red\n",
    "        - mIE4, HLA co-localized\n",
    "    \n",
    "    - HEK293T + mIE4-HA\n",
    "      - 36 hours after transfection\n",
    "      - IP\n",
    "        - anti-HA\n",
    "          - pull mIE4-HA\n",
    "        - anti-HLA\n",
    "          - pull HLA\n",
    "      - IB\n",
    "        - HA\n",
    "    \n",
    "    - flow cytometry\n",
    "      - MeWo + mIE4-HA\n",
    "        - stained with APC-HLA\n",
    "\n",
    "  - mIE4 VZV × ↓ MHC-I\n",
    "\n",
    "# 3 HCC & CRLM\n",
    "\n",
    "  - sc\n",
    "    \n",
    "    - Quality control criteria\n",
    "      - mitochondrial content\n",
    "        - ＜ 20%\n",
    "      - UMI counts\n",
    "        - 200-20,000\n",
    "      - gene counts\n",
    "        - 200-5,000\n",
    "    \n",
    "    - Seurat functions\n",
    "      - NormalizeData\n",
    "        - Data normalization\n",
    "      - FindVariableFeatures\n",
    "        - selection of highly variable genes\n",
    "        - 2000 genes\n",
    "      - ScaleData\n",
    "        - data transformation\n",
    "        - eliminating cell cycle effects with the parameters\n",
    "        - vars.to.regress = c(\"S.Score\", \"G2M.Score\")\n",
    "    \n",
    "    - Batch effect correction\n",
    "      - Harmony\n",
    "\n",
    "    - 32 clusters\n",
    "      - Markers\n",
    "\n",
    "        - epithelial cell\n",
    "          - EPCAM\n",
    "          - KRT18\n",
    "          - KRT19\n",
    "          - CDH1\n",
    "\n",
    "        - fibroblast\n",
    "          - DCN\n",
    "          - THY1\n",
    "          - COL1A1\n",
    "          - COL1A2\n",
    "\n",
    "        - endothelial cell\n",
    "          - PECAM1\n",
    "          - CLDN5\n",
    "          - FLT1\n",
    "          - RAMP2\n",
    "\n",
    "        - T cell \n",
    "          - CD3D\n",
    "          - CD3E\n",
    "          - CD3G\n",
    "          - TRAC\n",
    "\n",
    "        - NK cell \n",
    "          - NKG7\n",
    "          - GNLY\n",
    "          - NCAM1\n",
    "          - KLRD1\n",
    "\n",
    "        - B cell\n",
    "          - CD79A\n",
    "          - IGHM\n",
    "          - IGHG3\n",
    "          - IGHA2\n",
    "\n",
    "        - mast cell \n",
    "          - KIT\n",
    "          - MS4A2\n",
    "          - GATA2\n",
    "\n",
    "    - Sample origins tSNE\n",
    "\n",
    "    - Cluster annotation\n",
    "      \n",
    "    - Pearson correlation matrix\n",
    "      - correlation between\n",
    "        - clustering results\n",
    "        - cell type classification\n",
    "    \n",
    "    - Confirm cell-type identification\n",
    "      - marker gene expression heatmap\n",
    "      - marker gene expression tSNE\n",
    "\n",
    "    - Comparison\n",
    "      - cell type composition\n",
    "      - patient origin\n",
    "      - cell count\n",
    "      - transcript detection\n",
    "      - CNV status\n",
    "\n",
    "  - ST\n",
    "\n",
    "    - Post-quality control\n",
    "      - SpaceRanger software\n",
    "      - average #spots\n",
    "        - 4054\n",
    "      - average UMI\n",
    "        - 15317.4\n",
    "      - gene counts\n",
    "        - 3890.1\n",
    "      - mitochondrial content\n",
    "        - 4.5%\n",
    "    \n",
    "    - SCTtransform algorithm\n",
    "      - data normalization\n",
    "      - selection of highly variable genes\n",
    "      - data transformation\n",
    "    \n",
    "    - CNV detection\n",
    "      - InferCNV software\n",
    "    \n",
    "    - deconvolution analysis\n",
    "      - CARD algorithm\n",
    "    \n",
    "    - H&E staining\n",
    "      - CNV score distribution\n",
    "      - cell type composition\n",
    "\n",
    "    - tumor vs. non-tumor\n",
    "\n",
    "      - dimensionality reduction\n",
    "      - clustering analysis\n",
    "\n",
    "      - Tumor\n",
    "        - tSNE plot\n",
    "          - Tumor 8 subgroup\n",
    "          - Tumor sample origin\n",
    "          - Tumor CNV status\n",
    "          - Tumor patient composition\n",
    "        - Tumor marker genes heatmap\n",
    "\n",
    "\n",
    "      - Non-tumor\n",
    "        - tSNE plot\n",
    "          - Non-tumor 5 subgroup\n",
    "          - Non-tumor sample origin\n",
    "          - Non-tumor CNV status\n",
    "          - Non-tumor patient composition\n",
    "        - Non-tumor marker gene heatmap\n",
    "\n",
    "    - cell types distribution\n",
    "      - LIHC2 patients\n",
    "      - enrichment\n",
    "        - endothelial cells\n",
    "        - tumor cells\n",
    "        - myeloid cells\n",
    "\n",
    "    - cell composition heatmap\n",
    "      - HCC\n",
    "        - LIHCT1-4\n",
    "        - tumor cells\n",
    "      - CRLM\n",
    "        - LIHC1-4\n",
    "        - fibroblasts\n",
    "        - endothelial cells\n",
    "        - myeloid cells\n",
    "\n",
    "  - Tumor Cells Subclassification\n",
    "  \n",
    "    - dimensionality reduction\n",
    "    - clustering analysis\n",
    "    - 16 clusters\n",
    "      - merged\n",
    "      - based on marker gene functions\n",
    "      - 7 tumor cell subgroups\n",
    "\n",
    "        - cell composition\n",
    "          - HCC\n",
    "            - APOA2+NEO cells\n",
    "          - CRLM\n",
    "            - TFF3+NEO cells\n",
    "  \n",
    "    - Differential gene expression heatmap\n",
    "      - FindAllMarkers function\n",
    "        - p-value < 0.05\n",
    "        - log2FC > 0.25\n",
    "        - expression proportion > 0.1\n",
    "    \n",
    "    - GOBP\n",
    "      - GO Biological Process enrichment analysis\n",
    "      - revealing corresponding functions\n",
    "\n",
    "    - CNV differences\n",
    "    \n",
    "    - Transcription factor analysis heatmap\n",
    "\n",
    "    - Subsequent cell trajectory analysis\n",
    "      - multiple paths\n",
    "        - leading to different endpoints\n",
    "      - potential existence \n",
    "        - different subtypes\n",
    "      - starting branches\n",
    "        - from HCC\n",
    "        - APOA2+NEO cells\n",
    "        - tumor cells\n",
    "\n",
    "    - Pseudotime analysis\n",
    "      - Monocle2 software\n",
    "        - DDRTree\n",
    "          - dimensionality reduction algorithm\n",
    "          - default parameters\n",
    "          - cell differentiation\n",
    "      - starting points\n",
    "        - APOA2+NEO cells\n",
    "      - endpoints\n",
    "        - other cells\n",
    "\n",
    "      - pseudotime gene heatmap\n",
    "\n",
    "      - pseudotime scatter plots\n",
    "        - tumor marker genes\n",
    "          - APOA2\n",
    "          - HIST1H2AG\n",
    "          - HMGB2\n",
    "          - IGKV1-5\n",
    "          - LCN2\n",
    "          - TFF3\n",
    "\n",
    "  - APOA2+NEO Cells Functional Analysis\n",
    "\n",
    "    - APOA2+NEO cells difference\n",
    "      - HCC\n",
    "        - large number\n",
    "      - CRLM\n",
    "        - fewer\n",
    "\n",
    "    - Differential APOA2+NEO heatmap\n",
    "      - APOA2+NEO cells\n",
    "      - non-APOA2+NEO cells\n",
    "\n",
    "    - Lipid metabolism-related pathway\n",
    "      - selected from MSigDB\n",
    "      - signature scores heatmap\n",
    "        - Seurat\n",
    "          - GSVA package\n",
    "          - AddModelScore\n",
    "\n",
    "    - FATTY_ACID_BETA_OXIDATION\n",
    "      - SCENIC software\n",
    "        - across cell subgroups\n",
    "        - default parameters\n",
    "          - RcisTarget\n",
    "          - GRNBoost motif databases\n",
    "\n",
    "      - RcisTarget package\n",
    "        - overexpressed TF binding motifs\n",
    "\n",
    "      - AUCell package\n",
    "        - activity score\n",
    "          - each regulatory module for \n",
    "          - each cell type\n",
    "      - transcription factor IKZF1\n",
    "      - enriched in APOA2+NEO cells\n",
    "      - visualized in ST\n",
    "        - FATTY_ACID_BETA_OXIDATION scores distribution\n",
    "        - IKZF1 expression levels distribution\n",
    "\n",
    "    - violin plot\n",
    "      - FATTY_ACID_BETA_OXIDATION scores differences\n",
    "      - between two cell groups\n",
    "\n",
    "  - T, NK, and B Cells Subclassification\n",
    "\n",
    "    - dimensionality reduction\n",
    "    - clustering analysis\n",
    "    - 20 cell subgroups\n",
    "\n",
    "    - cell origins bar plot\n",
    "\n",
    "    - Differential gene volcano plot\n",
    "\n",
    "    - T, NK, and B Cells Marker gene tSNE\n",
    "\n",
    "    - Hallmark gene heatmap\n",
    "\n",
    "    - immune gene heatmap\n",
    "\n",
    "    - cytotoxicity scores\n",
    "      - CD8+ cells\n",
    "\n",
    "    - exhaustion scores\n",
    "      - CD8+ cells\n",
    "\n",
    "    - immune gene distribution\n",
    "      - GZMB\n",
    "      - GZMH\n",
    "      - PRF1\n",
    "      - GNLY\n",
    "      - CD8+ cell\n",
    "\n",
    "    - C12 and C14 subgroups\n",
    "      - CRLM\n",
    "        - C12\n",
    "          - TIMP1+CD8 cells\n",
    "            - poor survival\n",
    "      - HCC\n",
    "        - C14\n",
    "          - APOA2+CD8 cells\n",
    "            - better survival\n",
    "\n",
    "  - Myeloid Cells Subgroup Analysis\n",
    "  \n",
    "    - dimensionality reduction\n",
    "    - clustering analysis\n",
    "    - 17 cell subgroups\n",
    "    - Myeloid cell origin\n",
    "    - Myeloid cell annotation\n",
    "      - Sc-type software\n",
    "    \n",
    "    - Myeloid cell marker gene tSNE\n",
    "    \n",
    "    - Myeloid cell differential gene volcano plot\n",
    "    \n",
    "    - cell composition pie chart\n",
    "    \n",
    "    - Myeloid cell hallmark gene heatmap\n",
    "    \n",
    "    - macrophage-related trajectory analysis\n",
    "      - multiple differentiation paths\n",
    "      \n",
    "    - macrophage-related gene heatmap\n",
    "    \n",
    "    - M1, M2 macrophage scores tSNE\n",
    "    \n",
    "    - M1, M2 macrophage scores violin plot\n",
    "\n",
    "  - Stromal Cells Subgroup Analysis\n",
    "\n",
    "    - dimensionality reduction\n",
    "    - clustering analysis\n",
    "    - 12 cell subgroups\n",
    "    - Stromal cell origins\n",
    "\n",
    "    - Stromal cell Marker genes tSNE\n",
    "\n",
    "    - Stromal cell differential gene heatmap\n",
    "      - CAF\n",
    "      - endothelial cells\n",
    "\n",
    "    - Stromal Hallmark gene heatmap\n",
    "\n",
    "    - Stromal cell ST\n",
    "      - fibroblasts\n",
    "      - endothelial cells\n",
    "\n",
    "    - correlation scatter plot\n",
    "      - + correlation\n",
    "        - fibroblast\n",
    "        - endothelial cell\n",
    "\n",
    "  - Functional and Trajectory Analysis\n",
    "\n",
    "    - Stromal, CAFs Marker genes volcano\n",
    "\n",
    "    - CAF, myoCAF difference\n",
    "\n",
    "    - Stromal, CAFs Enrichment analysis\n",
    "      - GO database\n",
    "      - Type\n",
    "        - CAF1\n",
    "        - CAF2\n",
    "        - Endo\n",
    "        - myoCAF1\n",
    "\n",
    "    - CAFs Trajectory analysis\n",
    "      - two distinct subtypes\n",
    "\n",
    "  - Cell Communication Analysis\n",
    "\n",
    "    - separately for HCC, CRLM\n",
    "      - cell-cell communication\n",
    "        - CellChat package\n",
    "        - normalized gene expression matrix imported\n",
    "\n",
    "        - Data preprocessing\n",
    "          - identifyOverExpressedGenes\n",
    "          - identifyOverExpressedInteraction\n",
    "          - ProjectData\n",
    "          - default parameters\n",
    "    \n",
    "    - communication network diagrams\n",
    "      - aggregateNet function\n",
    "    \n",
    "    - communication scatter plots\n",
    "\n",
    "    - MIF signaling pathways\n",
    "      - differential chord diagram comparisons\n",
    "      - ligand-receptor interaction bar plots\n",
    "        - computeCommunProb\n",
    "        - filterCommunication\n",
    "        - computeCommunProbPathway\n",
    "\n",
    "    - GALECTIN signaling pathways\n",
    "      - differential chord diagram comparisons\n",
    "      - ligand-receptor interaction bar plots\n",
    "        - computeCommunProb\n",
    "        - filterCommunication\n",
    "        - computeCommunProbPathway\n",
    "\n",
    "    - Differential ligand-receptor interaction bubble plots\n",
    "      - CAF to other\n",
    "      - other to CAF\n",
    "      - higher communication capabilities in HCC\n",
    "        - LGALS9_CD44\n",
    "        - LGALS9_CD45\n",
    "        - LGALS9 in ST\n",
    "          - LGALS9_CD44 ＞ LGALS9_CD45\n",
    "\n",
    "  - Statistical Analysis\n",
    "\n",
    "    - R 4.1.3\n",
    "      - data processing\n",
    "      - statistical analysis\n",
    "      - plotting\n",
    "\n",
    "    - Processing\n",
    "      - TPM format\n",
    "      - log2-transformed\n",
    "\n",
    "    - chi-square test\n",
    "      - categorical variables\n",
    "\n",
    "    - continuous variables\n",
    "      - Wilcoxon rank-sum test\n",
    "      - T-test\n",
    "\n",
    "    - survminer package\n",
    "      - determine optimal cutoff values\n",
    "\n",
    "    - survival package\n",
    "      - Cox regression\n",
    "      - Kaplan-Meier analyses\n",
    "\n",
    "# 10 Hallmarks of Cancer\n",
    "\n",
    "  - Self-Sufficiency in Growth Signals\n",
    "\n",
    "    - mitogenic growth signals (GS)\n",
    "      - quiescent state → active proliferative state\n",
    "      - ensure a proper behavior of the various cell types within a tissue.\n",
    "      - transmitted\n",
    "        - transmembrane receptors\n",
    "      - signaling molecules\n",
    "        - diffusible growth factors\n",
    "        - extracellular matrix components\n",
    "        - cell-to-cell adhesion/interaction molecules\n",
    "    \n",
    "      - oncogenes\n",
    "        - mimicking GS\n",
    "\n",
    "# Interview\n",
    "\n",
    "# Forth\n",
    "\n",
    "## PChem\n",
    "\n",
    "  - Rate Processes & Statistical Mechanics\n",
    "    \n",
    "    - Arrhenius\n",
    "      - k = A * e^(-Ea / (R * T))\n",
    "      - A\n",
    "        - pre-exponential factor\n",
    "        - collision rate\n",
    "        - empirical\n",
    "      - Ea\n",
    "        - elementary reaction\n",
    "          - energy required to break or reorganise existing bonds in the reactants to create products\n",
    "        - complex reaction\n",
    "          - parameter describes T dependence of k\n",
    "      - Limitations\n",
    "        - entirely empirical\n",
    "        - only elementary reaction\n",
    "          - Ea, A are T independent\n",
    "      - × pressure effects\n",
    "      - × extreme temperatures\n",
    "        - alternative mechanisms\n",
    "          - quantum mechanical tunnelling \n",
    "          - higher-order saddle points\n",
    "\n",
    "    - Complex reactions\n",
    "      - complex rate laws \n",
    "        - MUST have complex mechanisms\n",
    "      - complex mechanisms \n",
    "        - × NECESSARILY complex rate laws\n",
    "      - SSA\n",
    "        - Steady-state approximation\n",
    "        - conc. reaction intermediates\n",
    "          - low \n",
    "          - × change\n",
    "        - e.g.\n",
    "          - N₂O₅\n",
    "\n",
    "    - SCT\n",
    "      \n",
    "      - Simple collision theory\n",
    "\n",
    "\n",
    "## Inorg\n",
    "\n",
    "## Org\n",
    "\n",
    "# 64\n",
    "\n",
    "  - 天\n",
    "    - 天\n",
    "      - 乾\n",
    "    - 泽\n",
    "      - 屯\n",
    "    - 火\n",
    "      - 同人\n",
    "    - 雷\n",
    "      - 无妄\n",
    "    - 风\n",
    "      - 讼\n",
    "    - 水\n",
    "      - 需\n",
    "    - 山\n",
    "      - 大畜\n",
    "    - 地\n",
    "      - 泰\n",
    "\n",
    "  - 泽\n",
    "    - 天\n",
    "      - 夬\n",
    "    - 泽\n",
    "      - 兑\n",
    "    - 火\n",
    "      - 革\n",
    "    - 雷\n",
    "      - 随\n",
    "    - 风\n",
    "      - 益\n",
    "    - 水\n",
    "      - 困\n",
    "    - 山\n",
    "      - 咸\n",
    "    - 地\n",
    "      - 萃\n",
    "\n",
    "  - 火\n",
    "    - 天\n",
    "      - 大有\n",
    "    - 泽\n",
    "      - 睽\n",
    "    - 火\n",
    "      - 离\n",
    "    - 雷\n",
    "      - 豫\n",
    "    - 风\n",
    "      - 家人\n",
    "    - 水\n",
    "      - 既济\n",
    "    - 山\n",
    "      - 贲\n",
    "    - 地\n",
    "      - 明夷\n",
    "\n",
    "  - 雷\n",
    "    - 天\n",
    "      - 无妄\n",
    "    - 泽\n",
    "      - 随\n",
    "    - 火\n",
    "      - 豫\n",
    "    - 雷\n",
    "      - 震\n",
    "    - 风\n",
    "      - 益\n",
    "    - 水\n",
    "      - 解\n",
    "    - 山\n",
    "      - 小过\n",
    "    - 地\n",
    "      - 复\n",
    "\n",
    "  - 风\n",
    "    - 天\n",
    "      - 讼\n",
    "    - 泽\n",
    "      - 益\n",
    "    - 火\n",
    "      - 家人\n",
    "    - 雷\n",
    "      - 益\n",
    "    - 风\n",
    "      - 巽\n",
    "    - 水\n",
    "      - 涣\n",
    "    - 山\n",
    "      - 渐\n",
    "    - 地\n",
    "      - 观\n",
    "\n",
    "  - 水\n",
    "    - 天\n",
    "      - 需\n",
    "    - 泽\n",
    "      - 困\n",
    "    - 火\n",
    "      - 既济\n",
    "    - 雷\n",
    "      - 解\n",
    "    - 风\n",
    "      - 涣\n",
    "    - 水\n",
    "      - 坎\n",
    "    - 山\n",
    "      - 蹇\n",
    "    - 地\n",
    "      - 比\n",
    "\n",
    "  - 山\n",
    "    - 天\n",
    "      - 大畜\n",
    "    - 泽\n",
    "      - 咸\n",
    "    - 火\n",
    "      - 贲\n",
    "    - 雷\n",
    "      - 小过\n",
    "    - 风\n",
    "      - 渐\n",
    "    - 水\n",
    "      - 蹇\n",
    "    - 山\n",
    "      - 艮\n",
    "    - 地\n",
    "      - 谦\n",
    "\n",
    "  - 地\n",
    "    - 天\n",
    "      - 泰\n",
    "    - 泽\n",
    "      - 萃\n",
    "    - 火\n",
    "      - 明夷\n",
    "    - 雷\n",
    "      - 复\n",
    "    - 风\n",
    "      - 观\n",
    "    - 水\n",
    "      - 比\n",
    "    - 山\n",
    "      - 谦\n",
    "    - 地\n",
    "      - 坤\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
