{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "XH = [64]\n",
    "ZL = 40\n",
    "from SY import *\n",
    "HX.LB(XH, ZL)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# P\n",
    "\n",
    "  - Free Electrons\n",
    "\n",
    "# Hallmarks of Cancer\n",
    "\n",
    "  - Self-Sufficiency in Growth Signals\n",
    "\n",
    "    - mitogenic growth signals (GS)\n",
    "      - quiescent state → active proliferative state\n",
    "      - ensure a proper behavior of the various cell types within a tissue.\n",
    "      - transmitted\n",
    "        - transmembrane receptors\n",
    "      - signaling molecules\n",
    "        - diffusible growth factors\n",
    "        - extracellular matrix components\n",
    "        - cell-to-cell adhesion/interaction molecules\n",
    "    \n",
    "      - oncogenes\n",
    "        - mimicking GS\n",
    "\n",
    "  normal cells in culture, proliferate\n",
    "  appropriate diffusible mitogenic factors\n",
    "  roper substratum\n",
    "  for their integrins\n",
    "\n",
    "  tumor cells\n",
    "  reduced dependence\n",
    "  exogenous growth stimulation\n",
    "  generate their own growth signals\n",
    "\n",
    "  - Deregulating Cellular Energetics and Avoiding Immune Destruction\n",
    "  \n",
    "\n",
    "# 64\n",
    "\n",
    "  - 天\n",
    "    - 天\n",
    "      - 乾\n",
    "    - 泽\n",
    "      - 屯\n",
    "    - 火\n",
    "      - 同人\n",
    "    - 雷\n",
    "      - 无妄\n",
    "    - 风\n",
    "      - 讼\n",
    "    - 水\n",
    "      - 需\n",
    "    - 山\n",
    "      - 大畜\n",
    "    - 地\n",
    "      - 泰\n",
    "\n",
    "  - 泽\n",
    "    - 天\n",
    "      - 夬\n",
    "    - 泽\n",
    "      - 兑\n",
    "    - 火\n",
    "      - 革\n",
    "    - 雷\n",
    "      - 随\n",
    "    - 风\n",
    "      - 益\n",
    "    - 水\n",
    "      - 困\n",
    "    - 山\n",
    "      - 咸\n",
    "    - 地\n",
    "      - 萃\n",
    "\n",
    "  - 火\n",
    "    - 天\n",
    "      - 大有\n",
    "    - 泽\n",
    "      - 睽\n",
    "    - 火\n",
    "      - 离\n",
    "    - 雷\n",
    "      - 豫\n",
    "    - 风\n",
    "      - 家人\n",
    "    - 水\n",
    "      - 既济\n",
    "    - 山\n",
    "      - 贲\n",
    "    - 地\n",
    "      - 明夷\n",
    "\n",
    "  - 雷\n",
    "    - 天\n",
    "      - 无妄\n",
    "    - 泽\n",
    "      - 随\n",
    "    - 火\n",
    "      - 豫\n",
    "    - 雷\n",
    "      - 震\n",
    "    - 风\n",
    "      - 益\n",
    "    - 水\n",
    "      - 解\n",
    "    - 山\n",
    "      - 小过\n",
    "    - 地\n",
    "      - 复\n",
    "\n",
    "  - 风\n",
    "    - 天\n",
    "      - 讼\n",
    "    - 泽\n",
    "      - 益\n",
    "    - 火\n",
    "      - 家人\n",
    "    - 雷\n",
    "      - 益\n",
    "    - 风\n",
    "      - 巽\n",
    "    - 水\n",
    "      - 涣\n",
    "    - 山\n",
    "      - 渐\n",
    "    - 地\n",
    "      - 观\n",
    "\n",
    "  - 水\n",
    "    - 天\n",
    "      - 需\n",
    "    - 泽\n",
    "      - 困\n",
    "    - 火\n",
    "      - 既济\n",
    "    - 雷\n",
    "      - 解\n",
    "    - 风\n",
    "      - 涣\n",
    "    - 水\n",
    "      - 坎\n",
    "    - 山\n",
    "      - 蹇\n",
    "    - 地\n",
    "      - 比\n",
    "\n",
    "  - 山\n",
    "    - 天\n",
    "      - 大畜\n",
    "    - 泽\n",
    "      - 咸\n",
    "    - 火\n",
    "      - 贲\n",
    "    - 雷\n",
    "      - 小过\n",
    "    - 风\n",
    "      - 渐\n",
    "    - 水\n",
    "      - 蹇\n",
    "    - 山\n",
    "      - 艮\n",
    "    - 地\n",
    "      - 谦\n",
    "\n",
    "  - 地\n",
    "    - 天\n",
    "      - 泰\n",
    "    - 泽\n",
    "      - 萃\n",
    "    - 火\n",
    "      - 明夷\n",
    "    - 雷\n",
    "      - 复\n",
    "    - 风\n",
    "      - 观\n",
    "    - 水\n",
    "      - 比\n",
    "    - 山\n",
    "      - 谦\n",
    "    - 地\n",
    "      - 坤\n",
    "\n",
    "\n",
    "# 0 interview\n",
    "\n",
    "  - VZV\n",
    "\n",
    "## 50 HCC & CRLM\n",
    "\n",
    "## 50.0 Acquisition / Processing\n",
    "\n",
    "  - TCGA database \n",
    "    - RNA expression profiles\n",
    "    - clinical data\n",
    "    - via UCSC Xena platform\n",
    "\n",
    "  - survival analysis\n",
    "    - 368 samples\n",
    "  \n",
    "  - Processing\n",
    "    - TPM format\n",
    "    - log2-transformed\n",
    "\n",
    "  - sc\n",
    "\n",
    "    - sc dataset\n",
    "      - GEO\n",
    "      - 10 HCC\n",
    "      - 9 CRLM\n",
    "\n",
    "    - Quality control criteria\n",
    "      - mitochondrial content\n",
    "        - ＜ 20%\n",
    "      - UMI counts\n",
    "        - 200-20,000\n",
    "      - gene counts\n",
    "        - 200-5,000\n",
    "\n",
    "    - Seurat functions \n",
    "      - NormalizeData\n",
    "        - Data normalization\n",
    "      - FindVariableFeatures\n",
    "        - selection of highly variable genes\n",
    "        - 2000 genes\n",
    "      - ScaleData\n",
    "        - data transformation\n",
    "        - eliminating cell cycle effects with the parameters \n",
    "        - vars.to.regress = c(\"S.Score\", \"G2M.Score\")\n",
    "\n",
    "      - Dimensionality reduction methods \n",
    "        - UMAP\n",
    "        - TSNE\n",
    "        - Louvain\n",
    "\n",
    "    - Batch effect correction\n",
    "      - Harmony\n",
    "\n",
    "    - Differential gene expression\n",
    "      - Between\n",
    "        - clusters\n",
    "        - cell types\n",
    "      - FindAllMarkers function\n",
    "        - p-value < 0.05, log2FC > 0.25\n",
    "        - expression proportion > 0.1\n",
    "\n",
    "  - ST\n",
    "\n",
    "    - HCC ST\n",
    "      - 4 HCC samples\n",
    "    - CRLM ST\n",
    "      - 4 CRLM samples\n",
    "\n",
    "    - SpaceRanger software\n",
    "      - Post-quality control\n",
    "\n",
    "    - SCTtransform algorithm\n",
    "      - data normalization\n",
    "      - selection of highly variable genes\n",
    "      - data transformation\n",
    "\n",
    "    - average number of spots\n",
    "      - 4054\n",
    "    - average UMI\n",
    "      - 15317.4\n",
    "    - gene counts\n",
    "      - 3890.1\n",
    "    - mitochondrial content\n",
    "      - 4.5%\n",
    "\n",
    "    - Seurat\n",
    "      - Analysis\n",
    "      - visualization\n",
    "\n",
    "    - CARD algorithm\n",
    "      - conditional autoregression based deconvolution\n",
    "      - Deconvolution analysis\n",
    "\n",
    "    - single-cell annotations\n",
    "      - cell type predictions for each spot\n",
    "\n",
    "    - CARD software\n",
    "      - Visualization of cell types in ST\n",
    "\n",
    "    - Signature scores\n",
    "      - Seurat\n",
    "        - GSVA package\n",
    "        - AddModelScore\n",
    "\n",
    "## 50.1 sc Expression Profile\n",
    "\n",
    "  - Markers\n",
    "\n",
    "    - epithelial cell\n",
    "      - EPCAM\n",
    "      - KRT18\n",
    "      - KRT19\n",
    "      - CDH1\n",
    "\n",
    "    - fibroblast\n",
    "      - DCN\n",
    "      - THY1\n",
    "      - COL1A1\n",
    "      - COL1A2\n",
    "\n",
    "    - endothelial cell\n",
    "      - PECAM1\n",
    "      - CLDN5\n",
    "      - FLT1\n",
    "      - RAMP2\n",
    "\n",
    "    - T cell \n",
    "      - CD3D\n",
    "      - CD3E\n",
    "      - CD3G\n",
    "      - TRAC\n",
    "\n",
    "    - NK cell \n",
    "      - NKG7\n",
    "      - GNLY\n",
    "      - NCAM1\n",
    "      - KLRD1\n",
    "\n",
    "    - B cell\n",
    "      - CD79A\n",
    "      - IGHM\n",
    "      - IGHG3\n",
    "      - IGHA2\n",
    "\n",
    "    - mast cell \n",
    "      - KIT\n",
    "      - MS4A2\n",
    "      - GATA2\n",
    "\n",
    "  - Pearson correlation matrix\n",
    "    - correlation between two continuous variables\n",
    "      - clustering results\n",
    "      - cell type classification\n",
    "    - marker gene expression\n",
    "    - heatmap\n",
    "\n",
    "  - Marker gene expression\n",
    "    - heatmaps\n",
    "    - tSNE plots\n",
    "    - confirming cell-type identification\n",
    "\n",
    "  - Comparisons\n",
    "    - cell type composition\n",
    "    - patient origin\n",
    "    - cell count\n",
    "    - transcript detection\n",
    "    - CNV status\n",
    "\n",
    "## 50.2 ST\n",
    "\n",
    "  - HCC enriched\n",
    "    - LIHCT\n",
    "    - tumor cells\n",
    "  - CRLM enriched\n",
    "    - LIHC\n",
    "    - fibroblasts\n",
    "    - endothelial cells\n",
    "    - myeloid cells\n",
    "\n",
    "  - CNV detection\n",
    "    - InferCNV software\n",
    "    - CNV score distribution\n",
    "\n",
    "  - deconvolution analysis\n",
    "    - CARD algorithm\n",
    "    - based on single-cell data\n",
    "    - cell type distribution\n",
    "\n",
    "    \n",
    "  - H&E staining images\n",
    "\n",
    "  - tumor vs. non-tumor classification\n",
    "\n",
    "    - dimensionality reduction\n",
    "    - clustering analysis\n",
    "\n",
    "\n",
    "\n",
    "    - tumor cells\n",
    "    8 subgroups\n",
    "    LIHCT1\n",
    "    LIHCT3\n",
    "    LIHCT4\n",
    "\n",
    "    tSNE plot\n",
    "    subgroup composition\n",
    "    sample origin\n",
    "    CNV status\n",
    "    patient composition\n",
    "\n",
    "    Marker genes for each subgroup were calculated\n",
    "    expression heatmaps were created\n",
    "\n",
    "\n",
    "\n",
    "    - non-tumor cells\n",
    "    5 subgroups\n",
    "    endothelial cells\n",
    "    tumor cells\n",
    "    myeloid cells\n",
    "\n",
    "    LIHCT2 enrichment\n",
    "\n",
    "    tSNE plot\n",
    "    marker gene expression heatmaps\n",
    "\n",
    "## 50.3 Subclassification Analysis of Tumor Cells\n",
    "\n",
    "Tumor cells were isolated and subjected to dimensionality reduction and clustering analysis, resulting in 16 clusters. These clusters were then merged based on their marker gene functions, yielding 7 tumor cell subgroups (Figure 3A). A bar plot displays the cell composition of each sample, showing that HCC samples have a higher content of APOA2+NEO cells, while CRLM samples have a higher content of TFF3+NEO cells. Differential gene expression analysis was performed for each subgroup, and a heatmap of differentially expressed genes was generated (Figure 3B). GO Biological Process (GOBP) enrichment analysis was conducted for the marker genes of each subgroup, revealing their corresponding functions. We compared the CNV differences among different cell types from different sample origins, finding significant differences in CNV across most cell types (Figure 3C). Transcription factor analysis was also performed for each cell type, with the results visualized in a heatmap showing different transcription factors enriched in various cell types (Figure 3D).\n",
    "\n",
    "Subsequent cell trajectory analysis revealed multiple paths leading to different endpoints, indicating the potential existence of different subtypes. The starting branches contained a large number of APOA2+NEO cells and tumor cells from HCC (Figure 3E). From the pseudotime analysis, APOA2+NEO cells were primarily at the starting points, while most other cells were at the endpoints. Genes related to pseudotime were identified, and their expression heatmap was generated (Figure 3F). Scatter plots of pseudotime for six selected tumor marker genes (APOA2, HIST1H2AG, HMGB2, IGKV1-5, LCN2, TFF3) were created (Figure 3G).\n",
    "\n",
    "## 50.3 Cell Annotation Analysis\n",
    "\n",
    "  - tumor heterogeneity clustering\n",
    "    - epithelial cells\n",
    "    - immune cells\n",
    "    - fibroblasts\n",
    "\n",
    "\n",
    "## 50.4 Subgroup Analysis of Cell Populations\n",
    "\n",
    "  - further distinction for subgroups\n",
    "    - Immune cells\n",
    "    - epithelial cells\n",
    "    - fibroblasts\n",
    "\n",
    "  - standard Seurat workflow\n",
    "  - specific markers for each subgroup\n",
    "  - UMAP plots\n",
    "\n",
    "## 50.5 CNV Analysis of Single Cells\n",
    "\n",
    "  - CNVs in tumor cell subgroups\n",
    "    - InferCNV\n",
    "    - references\n",
    "      - fibroblasts\n",
    "      - endothelial cells\n",
    "    - identify malignant cells within subgroups\n",
    "\n",
    "## 50.6 Pseudotime Analysis of Single Cells\n",
    "\n",
    "  - Monocle2 software\n",
    "    - pseudotime analysis\n",
    "      - epithelial\n",
    "      - macrophage\n",
    "      - CAF cell\n",
    "\n",
    "  - DDRTree\n",
    "    - dimensionality reduction algorithm\n",
    "    - parameters set to default\n",
    "    - cell differentiation\n",
    "\n",
    "## 50.7 Transcription Factor Analysis\n",
    "\n",
    "  - SCENIC software\n",
    "    - transcription factor analysis across cell subgroups\n",
    "    - default parameters\n",
    "      - RcisTarget\n",
    "      - GRNBoost motif databases\n",
    "\n",
    "  - RcisTarget package\n",
    "    - overexpressed transcription factor binding motifs\n",
    "\n",
    "  - AUCell package\n",
    "    - scored the activity of each regulatory module for each cell type\n",
    "\n",
    "## 50.8 Cell-Cell Communication Analysis\n",
    "\n",
    "  - cell-cell communication\n",
    "    - CellChat package\n",
    "    - normalized gene expression matrix was imported\n",
    "\n",
    "  - Data preprocessing\n",
    "    - identifyOverExpressedGenes\n",
    "    - identifyOverExpressedInteraction\n",
    "    - ProjectData\n",
    "    - default parameters\n",
    "\n",
    "  - ligand-receptor interactions\n",
    "    - computeCommunProb\n",
    "    - filterCommunication\n",
    "    - computeCommunProbPathway\n",
    "\n",
    "  - cell communication network\n",
    "    - aggregateNet function\n",
    "\n",
    "## 50.9 Statistical Analysis\n",
    "\n",
    "  - R 4.1.3\n",
    "    - data processing\n",
    "    - statistical analysis\n",
    "    - plotting\n",
    "\n",
    "  - Pearson correlation coefficient\n",
    "    - correlation between two continuous variables\n",
    "\n",
    "  - chi-square test\n",
    "    - categorical variables\n",
    "\n",
    "  - continuous variables\n",
    "    - Wilcoxon rank-sum test\n",
    "    - T-test\n",
    "\n",
    "  - survminer package\n",
    "    - determine optimal cutoff values\n",
    "\n",
    "  - survival package\n",
    "    - Cox regression\n",
    "    - Kaplan-Meier analyses"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
