{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1 Protein folding\n",
    "  \n",
    "  - Why\n",
    "    - Diseases\n",
    "      - Replacement, Glu β6 → Val\n",
    "        - Mutated haemoglobin\n",
    "        - aggregate into filaments in deoxygenated form\n",
    "        - causes sickle-cell anaemia\n",
    "      - Partially unstructured proteins\n",
    "        - aggregation \n",
    "      - Neurodegenerative disease\n",
    "    - Function\n",
    "      - give indication about function\n",
    "    - Design\n",
    "      - Understanding mechanism\n",
    "      - Design secondary structure\n",
    "        - e.g. helical bundles\n",
    "    - Common misfolding\n",
    "      - α-helix to β-sheet\n",
    "      - Exposes hydrophobic aa\n",
    "        - leads to aggregation\n",
    "\n",
    "  - Anfinsen Experiment\n",
    "\n",
    "    - Ribonuclease A Denaturation\n",
    "      - 4 disulphide bonds\n",
    "      - 8 M urea + β-mercaptoethanol\n",
    "      - leads to random coil\n",
    "        - no activity\n",
    "\n",
    "    - refolded protein\n",
    "      - √ native activity\n",
    "      - information necessary for folding\n",
    "        - aa sequence\n",
    "\n",
    "    - no activity\n",
    "      - Remove β-mercaptoethanol only\n",
    "      - oxidation Cys\n",
    "      - remove urea\n",
    "      \n",
    "    - reforms native form\n",
    "      - trace amounts of β-mercaptoethanol\n",
    "\n",
    "    - native form\n",
    "      - thermodynamically most stable\n",
    "\n",
    "\n",
    "  - Golf course landscape\n",
    "    - Single low energy conformation\n",
    "    - Random search\n",
    "    - Very slow folding\n",
    "    - Levinthal Paradox\n",
    "      - M^N\n",
    "        - M\n",
    "          - #possible conformations\n",
    "        - N\n",
    "          - #aa\n",
    "\n",
    "  - Rugged funnel landscape\n",
    "    - Reduces conformational search\n",
    "    - Hierarchical folding\n",
    "      - structure complexity increases gradually\n",
    "    - Allows fast folding\n",
    "      - ms\n",
    "      - s\n",
    "    - Small changes in conformation\n",
    "      - lead to formation\n",
    "        - few H-bonds\n",
    "        - hydrophobic contacts\n",
    "      - decrease in energy\n",
    "    - Different unfolded molecules\n",
    "      - via different routes Funnel landscape\n",
    "    - trapped misfolded states/Intermediates\n",
    "      - productive\n",
    "        - unfold to refold correctly\n",
    "      - unproductive\n",
    "        - Irreversible aggregation\n",
    "        - through intermolecular interactions\n",
    "\n",
    "  - Folding equilibrium\n",
    "    - U\n",
    "      - Solvation\n",
    "        - exposed amide\n",
    "      - Entropy\n",
    "        - flexible chain\n",
    "\n",
    "    - N\n",
    "      - H-bonded secondary structure\n",
    "        - compensate for solvation\n",
    "      - non-polar residues\n",
    "        - Hydrophobic effect\n",
    "      - side chains\n",
    "        - Van der Waals packing\n",
    "\n",
    "  \n",
    "  - Proteins stabilized by\n",
    "    - non-bonding interactions\n",
    "    - e.g.\n",
    "      - H-bond\n",
    "      - Hydrophobic\n",
    "      - Electrostatic\n",
    "      - Van der Walls\n",
    "\n",
    "  - Denaturation\n",
    "\n",
    "    - cooperative process\n",
    "      - partial unfolding destabilises rest of structure\n",
    "\n",
    "    - Factors\n",
    "      - Temperature\n",
    "      - pH\n",
    "      - Detergents\n",
    "      - High conc. of soluble compound\n",
    "        - alcohol\n",
    "      - Salts\n",
    "\n",
    "    - effect\n",
    "      - stability\n",
    "      - energy barriers in two state system\n",
    "\n",
    "    - measurment\n",
    "      - Thermofluor assay\n",
    "        - Protein Stability\n",
    "          - Tm\n",
    "            - melting temperatur\n",
    "            - midpoint in unfolding process\n",
    "          - Cofactors, ligands can stabilise\n",
    "            - Cu on stability of LPMOs\n",
    "        - Thermofluor analysis\n",
    "          - differential scanning fluorimetry\n",
    "\n",
    "        - Protein + fluorescent dye + cuvette\n",
    "          - fluorescence quenched by water\n",
    "          - T ↑\n",
    "          - protein unfold\n",
    "          - dye bind hydrophobic residues\n",
    "          - expelling water\n",
    "          - fluorescence ↑\n",
    "\n",
    "# 2 Protein composition\n",
    "  \n",
    "  - 50 to 400+ \n",
    "  \n",
    "  - L-aa\n",
    "  \n",
    "  - PTMs\n",
    "    - Disulphide bonds\n",
    "    - Proteolytic cleavage\n",
    "    - Modification of aa\n",
    "    - Modified proteins might not refold if unfolded\n",
    "    - Phosphorylation\n",
    "      - Most abundant PTM in Eukaryotes\n",
    "    - Glycosylation\n",
    "      - folding\n",
    "      - stability\n",
    "      - signalling\n",
    "\n",
    "  - Peptide bond\n",
    "\n",
    "    - resonance\n",
    "      - rigid\n",
    "      - planar\n",
    "    - planarity restricts rotations\n",
    "\n",
    "  - Conformation preferences of aa\n",
    "\n",
    "    - Ramachandran plot\n",
    "\n",
    "      - α-helix\n",
    "        - φ ≈ -60°\n",
    "        - ψ ≈ -45°\n",
    "        - angles close together\n",
    "        - restricted rotation\n",
    "          - Cα-Cβ\n",
    "          - due to packing of R group\n",
    "      - β-sheet\n",
    "        - φ ≈ -120°\n",
    "        - ψ ≈ 120°\n",
    "        - angles farther apart\n",
    "\n",
    "\n",
    "  - without β-branching\n",
    "    - preferred in a helix\n",
    "    - e.g.\n",
    "      - Ala\n",
    "      - Glu\n",
    "      - Lys\n",
    "      - Met\n",
    "      - Leu\n",
    "\n",
    "  - β-branched residues\n",
    "    - prefer “extended” polypeptide chain\n",
    "      - β-sheets\n",
    "      - steric interactions minimised\n",
    "    - e.g.\n",
    "      - Val\n",
    "      - Ile\n",
    "      - Thr\n",
    "\n",
    "  - Conformational search\n",
    "    - influenced by intrinsic properties\n",
    "    - adopt favoured \n",
    "      - φ\n",
    "      - ψ\n",
    "\n",
    "  - Chou-Fasman Method\n",
    "\n",
    "    - empirical structure prediction method\n",
    "\n",
    "    - Pref = 1 \n",
    "      - residue found equally in α, β regions\n",
    "\n",
    "    - Pref > 1 \n",
    "      - preference for one\n",
    "\n",
    "    - Value < ~0.7\n",
    "      - breaker for region\n",
    "\n",
    "    - nucleate\n",
    "      - helix\n",
    "        - 4/6 helix-forming residues\n",
    "\n",
    "      - sheet\n",
    "        - 3/5 sheet-forming residues\n",
    "\n",
    "      - region broken\n",
    "        - average ＜ 1 for tetrapeptide \n",
    "\n",
    "      - Pro, Gly\n",
    "        - × preference for either \n",
    "        - Prefer\n",
    "          - irregular \n",
    "          - flexible loops\n",
    "        - STOP signal\n",
    "\n",
    "# 3 Secondary Structure\n",
    "\n",
    "  - β-sheets\n",
    "    - Sequence rich\n",
    "      - Val\n",
    "      - Thr\n",
    "      - Ile\n",
    "      - Tyr\n",
    "    - parallel\n",
    "      - non-contiguous\n",
    "    - anti-parallel\n",
    "      - contiguous\n",
    "    - i, i+2 periodicity\n",
    "      - stabilising side chains contacts\n",
    "    - Intra- and inter-strand interactions\n",
    "    - P & G \n",
    "      - STOP signal\n",
    "      - reverse turn facilitating\n",
    "        - anti-parallel β-hairpin\n",
    "\n",
    "  - α-helices\n",
    "    - Sequence rich\n",
    "      - Ala\n",
    "      - Leu\n",
    "      - Lys\n",
    "      - Glu\n",
    "    - periodicity of side chains\n",
    "      - i to i+3 / i+4\n",
    "      - favourable helix stabilising interactions\n",
    "    - R groups point outward\n",
    "      - influence helical propensity\n",
    "    - Complementary side chain interactions\n",
    "      - 3, 7, 11\n",
    "        - Lys, Glu, Lys\n",
    "        - Leu, Ile, Leu\n",
    "    - Amphipathic helix\n",
    "      - one polar\n",
    "        - solvent exposed\n",
    "      - one non-polar face\n",
    "        - packing against core\n",
    "    - P & G\n",
    "      - STOP signal\n",
    "\n",
    "  - Higher order assembly\n",
    "\n",
    "    - Pseudo heptad repeat\n",
    "    \n",
    "    - coiled coils via intermolecular interactions\n",
    "\n",
    "    - Parallel dimer\n",
    "      - abcdefg : abcdefg\n",
    "        - a, d\n",
    "          - Leu\n",
    "          - Val\n",
    "        - b, c, f\n",
    "          - anything\n",
    "        - g and e\n",
    "          - complementary charge\n",
    "\n",
    "    - Anti-parallel dimer\n",
    "      - abcdefg : a’b’c’d’e’f’g’\n",
    "        - a, d\n",
    "          - Leu\n",
    "          - Val\n",
    "        - b, c, f\n",
    "          - anything\n",
    "        - e = g\n",
    "        - e’ = g’\n",
    "\n",
    "  - De novo design\n",
    "\n",
    "      - Cys-His-Glu triad\n",
    "        - in natural ester hydrolases\n",
    "          - p-Nitrophenyl acetate \n",
    "          - to p-Nitrophenol\n",
    "        - mutated into centre of four-heptad repeat\n",
    "\n",
    "    - Synthetic coiled coils mimic\n",
    "      - carbonic anhydrase\n",
    "      - nitrite reductase\n",
    "\n",
    "    - Non-natural biopolymers\n",
    "\n",
    "      - Non-natural synthetic\n",
    "        - oligomers\n",
    "        - foldamers\n",
    "        - Different folding rules\n",
    "      - Peptidomimetics\n",
    "        - Resistant to proteases\n",
    "\n",
    "      - 2.5 residues per turn\n",
    "        - H-bond\n",
    "        - C=O (i)\n",
    "        - NH (i+3)\n",
    "        - N’H (i+4)\n",
    "        - more H-bond\n",
    "      \n",
    "      - H1\n",
    "        - Helical arrangement \n",
    "          - not affected by side chains\n",
    "        \n",
    "        - Self-assembly\n",
    "          - hexameric helix bundle\n",
    "          - controlled by\n",
    "            - side chains\n",
    "          - driven by\n",
    "            - hydrophobic core formation\n",
    "        \n",
    "        - Complementary surface charges\n",
    "      \n",
    "      - H2\n",
    "        - Self-assembly\n",
    "          - hexameric channel/pore\n",
    "          - controlled by\n",
    "            - side chains\n",
    "          - driven by\n",
    "            - hydrophilic pore formation\n",
    "\n",
    "        - Surface hydrophobic residues\n",
    "\n",
    "# 4 Peptide Synthesis\n",
    "\n",
    "  - Solid Synthesis\n",
    "    - Coupling\n",
    "    - Deprotection \n",
    "      - 20% piperidine\n",
    "      - DMF\n",
    "    - Coupling\n",
    "      - DCC/HOBt\n",
    "      - DCI/Oxima\n",
    "    - Capping\n",
    "      - acetic anhydride\n",
    "    - Repeat steps\n",
    "    - Cleave from resin\n",
    "      - TFA\n",
    "\n",
    "  - Protecting Group\n",
    "    \n",
    "    - Characteristics\n",
    "      - Easy, selective reaction\n",
    "      - Not affect stereochemistry of aa\n",
    "      - Stable under coupling conditions\n",
    "      - Easy to selectively remove without \n",
    "        - damaging peptide\n",
    "        - orthogonal protection\n",
    "    \n",
    "    - Amino protection\n",
    "\n",
    "      - Conversion to carbamates (urethanes)\n",
    "        - amide\n",
    "          - decreases reactivity of N\n",
    "        - ester\n",
    "          - easily hydrolysed\n",
    "          - spontaneously decarboxylate\n",
    "        \n",
    "        - Boc\n",
    "          - amine protection\n",
    "          - deprotection in H+\n",
    "            - TFA\n",
    "            - HCl\n",
    "\n",
    "        - Fmoc\n",
    "          - stable to H+\n",
    "            - orthogonal to Boc\n",
    "          - deprotection in base\n",
    "            - 20% piperidine in DMF\n",
    "\n",
    "      - Triphenylmethyl (trityl, Trt) group\n",
    "\n",
    "        - Trt\n",
    "          - steric bulk\n",
    "          - stable to base, unstable to H+\n",
    "            - orthogonal to Fmoc\n",
    "            - not orthogonal to Boc\n",
    "            - deprotection in mild H+\n",
    "              - weak acid\n",
    "              - acetic acid\n",
    "              - thanks to stability of carbocation\n",
    "      \n",
    "    - Carboxy protection\n",
    "\n",
    "      - Esterification\n",
    "\n",
    "      - methyl, ethyl\n",
    "        - occurs swiftly\n",
    "        - stable\n",
    "        - difficult to deprotect\n",
    "\n",
    "      - Benzyl\n",
    "        - resistant to TFA\n",
    "        - deprotection\n",
    "          - hydrazinolysis\n",
    "          - HBr/AcOH\n",
    "          - hydrogenolysis\n",
    "\n",
    "      - t-Butyl\n",
    "        - similar reactivity to Boc\n",
    "        - less sensitive to TFA\n",
    "\n",
    "      - Phenyl\n",
    "        - stable to H+\n",
    "        - deprotection\n",
    "          - saponification\n",
    "          - hydrogen peroxide\n",
    "        \n",
    "    - Peptide bond formation\n",
    "\n",
    "      - carboxylic acid + amine → salt\n",
    "\n",
    "      - requires carboxylic activation\n",
    "\n",
    "      - acylating reagent\n",
    "        - prepared in separate steps\n",
    "        - formed in situ\n",
    "        - e.g.\n",
    "          - Dicyclohexylcarbodiimide (DCC)\n",
    "          - derivatives (e.g. DIC)\n",
    "\n",
    "    - Side chains protection\n",
    "\n",
    "      - Most aa side chain can give unwanted reactivity\n",
    "\n",
    "        - Lys\n",
    "          - Boc\n",
    "\n",
    "        - Ser\n",
    "          - tBu\n",
    "        - Thr\n",
    "          - tBu\n",
    "        - Tyr\n",
    "          - tBu\n",
    "\n",
    "        - Asp\n",
    "          - OtBu\n",
    "        - Glu\n",
    "          - OtBu\n",
    "\n",
    "        - Cys\n",
    "          - Trt\n",
    "\n",
    "      - Other residue\n",
    "        - not necessarily\n",
    "        - yields higher\n",
    "        - purification steps easier\n",
    "\n",
    "        - Asn\n",
    "          - Trt\n",
    "\n",
    "        - Gln\n",
    "          - Trt\n",
    "\n",
    "        - Arg\n",
    "          - Pbf\n",
    "          - deprotection with TFA\n",
    "\n",
    "        - His\n",
    "          - Trt\n",
    "\n",
    "        - Trp\n",
    "          - Boc\n",
    "\n",
    "# 5 Spectroscopic techniques\n",
    "  \n",
    "  - Secondary structures\n",
    "    - CD\n",
    "      - Circular Dichroism\n",
    "      - Linearly polarised light\n",
    "        - oscillation\n",
    "        - in a single plane\n",
    "        - along the direction of propagation\n",
    "      - Circularly polarised light\n",
    "        - electric vector rotates \n",
    "        - (left or right) \n",
    "        - around the direction of propagation\n",
    "\n",
    "    - Ellipticity (θ)\n",
    "      - Chiral molecules \n",
    "        - interact differently with the left / right light\n",
    "        - producing a net absorption\n",
    "      - proteins\n",
    "        - peptide backbone\n",
    "        - “handedness” of protein secondary structure\n",
    "\n",
    "    - no net absorption\n",
    "      - Achiral\n",
    "      - racemic molecules\n",
    "\n",
    "    - Absorption bands are very \n",
    "      - sensitive to conformation\n",
    "      - arise from electronic excitation\n",
    "        - amide chromophores\n",
    "\n",
    "    - characteristic CD “finger print” \n",
    "      - secondary structures\n",
    "      - random coil\n",
    "\n",
    "    - Random coil\n",
    "      - -ve ellipticity at ~198 nm\n",
    "      - +ve above ~210 nm\n",
    "    - β-sheet\n",
    "      - -ve band at 216-218 nm\n",
    "      - +ve below ~210 nm\n",
    "    - α-helix\n",
    "      - Strong double –ve band at ~210 nm and ~222 nm\n",
    "      - Strong +ve below 200 nm\n",
    "    - Proteins\n",
    "      - linear combination\n",
    "\n",
    "    - Conformational analysis\n",
    "\n",
    "      - “in phase” \n",
    "        - on one surface of the helix\n",
    "        - His\n",
    "        - Met\n",
    "    \n",
    "      - Hypothesis\n",
    "        - Ag+\n",
    "        - bind\n",
    "        - induce helix formation\n",
    "\n",
    "    - CD is a good probe\n",
    "      - protein secondary structure and for \n",
    "      - monitoring folding transitions\n",
    "    - characteristic signatures\n",
    "      - helices\n",
    "      - sheets\n",
    "      - random coils\n",
    "\n",
    "  - Tertiary structures\n",
    "\n",
    "    - Fluorescence spectroscopy\n",
    "\n",
    "      - monitor protein denaturation\n",
    "      - intrinsic fluorescence of \n",
    "        - Phe\n",
    "        - Tyr\n",
    "        - Trp\n",
    "      \n",
    "      - Trp\n",
    "\n",
    "        - Buried in hydrophobic core\n",
    "\n",
    "          - blue shift ~20 nm \n",
    "            - from 355 nm\n",
    "            - to 309-335 nm\n",
    "\n",
    "        - Effect on quantum yield\n",
    "          - difficult to predict\n",
    "          - quencher\n",
    "            - Several aa chains\n",
    "\n",
    "    - Linear extrapolation method \n",
    "      \n",
    "      - Assumption\n",
    "        - ΔGeq[D] = ΔGeq + m[D]\n",
    "\n",
    "        - ΔGeq\n",
    "          - stability on water alone\n",
    "\n",
    "      - transition midpoint\n",
    "        - U = N\n",
    "        - Keq = 1\n",
    "        - ΔGeq[D] = 0\n",
    "        - midpoint defined\n",
    "          - [D]_50%\n",
    "          - ΔGeq = -m[D]_50%\n",
    "\n",
    "      - Extrapolated stability in water ΔG_eq depends on \n",
    "        - slope m\n",
    "        - transition midpoint [D]_50%\n",
    "\n",
    "\n",
    "      - m value\n",
    "        - correlates with the hydrophobic surface area buried on folding → \n",
    "        - larger proteins have more co-operative folding transitions →\n",
    "        - bigger m value\n",
    "\n",
    "        - similar for protein containing non-disruptive point mutations\n",
    "\n",
    "    - Two-state protein folding kinetics\n",
    "    \n",
    "      - Equilibrium\n",
    "        - differences in energy\n",
    "          - folded\n",
    "          - unfolded\n",
    "          - ΔGo_eq\n",
    "\n",
    "      - Kinetics\n",
    "        - barrier between states\n",
    "        - probe protein folds pathway\n",
    "          - two-state \n",
    "          - U and N\n",
    "            - separated by single TS\n",
    "\n",
    "      - TS\n",
    "        - highest point on energy landscape\n",
    "        - All molecules pass through TS\n",
    "        - not significantly populated\n",
    "\n",
    "        - represents balance\n",
    "          - a few stabilising interactions\n",
    "          - entropic cost of restricting the flexibility\n",
    "\n",
    "        - highly compact\n",
    "          - almost “native-like”\n",
    "          - highly unfolded\n",
    "\n",
    "\n",
    "      - Denaturants Effect\n",
    "\n",
    "        - protein structure\n",
    "          - work by stabilising the unfolded state\n",
    "          - More amide groups are exposed in unfolded state\n",
    "          - solvent exposed amide groups\n",
    "            - Gdm.Cl\n",
    "            - urea hydrogen bond\n",
    "\n",
    "          - transition state\n",
    "            - partially stabilised\n",
    "\n",
    "          - native folded state\n",
    "            - least stabilised\n",
    "\n",
    "          - transition mid-point\n",
    "            - folded = unfolded\n",
    "\n",
    "        - folding kinetics\n",
    "\n",
    "          - At some [D]\n",
    "            - free energy UD ＜ ND\n",
    "            - protein unfolds\n",
    "\n",
    "          - [D] ↑\n",
    "            - UD becomes stable as D binds\n",
    "              - UD ↓\n",
    "            - Barrier to unfolding ΔGDU ↓\n",
    "            - Barrier to folding ΔGDF ↑\n",
    "\n",
    "        - Stopped-flow rapid mixing apparatus\n",
    "\n",
    "          - collected Kinetic data\n",
    "\n",
    "          - Rapid jump in conditions\n",
    "            - triggers refolding\n",
    "\n",
    "          - Mixing must be rapid \n",
    "            - (~2-3 ms)\n",
    "            - compared with the rate of folding\n",
    "\n",
    "          - After initial mixing\n",
    "            - equilibrium is re-established\n",
    "            - molecules refold\n",
    "\n",
    "          - Detect the folding “in real time” \n",
    "            - using change in fluorescence\n",
    "            - Repeat the experiment at different final [D]\n",
    "\n",
    "\n",
    "# Q2\n",
    "\n",
    "  - Mutations\n",
    "    \n",
    "    - Type\n",
    "      - Point Mutations\n",
    "      - Silent\n",
    "      - Nonsense\n",
    "      - Missense\n",
    "        - Conservative\n",
    "        - Non-Conservative\n",
    "\n",
    "    - Sequencing\n",
    "      - Reagents\n",
    "        - Template DNA\n",
    "        - ddNTPs\n",
    "          - ddATP\n",
    "          - ddTTP\n",
    "          - ddGTP\n",
    "          - ddCTP\n",
    "        - DNA polymerase\n",
    "        - Primers\n",
    "\n",
    "      - Progress\n",
    "        - Primer Annealing \n",
    "        - Chain Extension\n",
    "        - ddNTP Binding\n",
    "        - Chain Termination\n",
    "        - Fluorescently Labeled DNA Sample\n",
    "        - Capillary Gel Electrophoresis\n",
    "        - Fluorescence Detection\n",
    "        - Sequence Analysis\n",
    "        - Reconstruction\n",
    "\n",
    "  - X-ray crystallography\n",
    "\n",
    "    - Protein size\n",
    "      - Broad range\n",
    "\n",
    "    - Protein type\n",
    "      - × membrane proteins\n",
    "\n",
    "    - Other comments\n",
    "      - √ total structure determination\n",
    "      - √ well established, productive process\n",
    "      - × “frozen”\n",
    "      - × can appear distorted\n",
    "      - × crystals from soluble proteins\n",
    "\n",
    "  - NMR spectroscopy\n",
    "\n",
    "    - Protein size\n",
    "      - < 50 kDa\n",
    "\n",
    "    - Protein type\n",
    "      - Broad range\n",
    "\n",
    "    - Other comments\n",
    "      - √ proteins studied in aqueous solution\n",
    "      - √ protein dynamics possible\n",
    "      - × often requires stable isotope labelling\n",
    "      - × membrane proteins challenging\n",
    "      - × requires very stable proteins\n",
    "\n",
    "\n",
    "  - Cryo-EM\n",
    "\n",
    "    - Express protein\n",
    "\n",
    "    - Purify protein\n",
    "\n",
    "    - Plunge-freeze\n",
    "    \n",
    "      - Dubochet's Vitrification Method\n",
    "        - sample transferred to metal mesh\n",
    "        - sample forms a thin film across the holes in the mesh\n",
    "          - shot into ethane at -190°\n",
    "        - water vitrifies around sample\n",
    "          - cooled by liquid nitrogen during the measurements in the electron microscope.\n",
    "\n",
    "    - Transfer to TEM\n",
    "\n",
    "    - Image specimen\n",
    "\n",
    "    - Analyse & reconstruct\n",
    "\n",
    "      - Frank's Image Analysis for 3D Structures\n",
    "        - Randomly oriented proteins hit by electron beam\n",
    "        - leaving a trace on image\n",
    "        - computer discriminates between traces and fuzzy background\n",
    "        - placing similar ones in same group\n",
    "        - Using thousands of similar traces\n",
    "        - computer generates a high-resolution 2D image\n",
    "        - computer calculates \n",
    "        - different 2D images relate to each other\n",
    "          - generates 3D high-resolution structure    \n",
    "\n",
    "  - Bioinformatics\n",
    "\n",
    "    - Define\n",
    "      - interdisciplinary field \n",
    "      - develops methods / software tools\n",
    "      - understanding biological data\n",
    "      - large and complex data sets\n",
    "  \n",
    "    - Compare alignment scoring matrices\n",
    "      - BLOSUM\n",
    "\n",
    "    - link between “omics”\n",
    "      - transcription\n",
    "      - translation\n",
    "      - enzymatic activity\n",
    "    \n",
    "    - Demonstrate cutting-edge applications\n",
    "      - COVID genome sequencing\n",
    "      - Predicting protein folding\n",
    "\n",
    "  - Protein design\n",
    "\n",
    "    - library scale and use of plasmids\n",
    "\n",
    "    - Rational design\n",
    "      - Small scale mutations\n",
    "        - 10-100\n",
    "      - based on extensive knowledge\n",
    "      - Rotamer Confirmations\n",
    "\n",
    "    - Directed evolution\n",
    "\n",
    "      - High throughput\n",
    "        - 10^10-10^12\n",
    "\n",
    "      - randomised mutations\n",
    "\n",
    "      - step\n",
    "        - inserted in bacteria as templates\n",
    "        - produce randomly mutated enzymes\n",
    "        - test changed enzymes\n",
    "        - selected most efficient\n",
    "        - New random mutations introduced for next\n",
    "\n",
    "      - Phage display\n",
    "        - introduced a gene in phage’s capsule \n",
    "          - phage DNA inserted into bacteria \n",
    "            - produced phage\n",
    "\n",
    "        - peptide \n",
    "          - produced from introduced gene\n",
    "          - ended up as part of capsule\n",
    "          - on phage surface\n",
    "\n",
    "        - fish out the phage \n",
    "          - antibody designed to\n",
    "            - attach to the peptide\n",
    "          - got peptide gene\n",
    "\n",
    "    - De novo design\n",
    "\n",
    "      - Computational protein design algorithms\n",
    "\n",
    "      - based on\n",
    "        - 3D structures of natural proteins\n",
    "        - folding rules\n",
    "\n",
    "      - Step\n",
    "\n",
    "        - Construct multiple-sequence alignment\n",
    "\n",
    "        - Predict local structure\n",
    "\n",
    "          - Fragments\n",
    "            - 20–200 per each segment\n",
    "        \n",
    "        - Predict residue contacts\n",
    "          - Correlated mutations\n",
    "          - Residue-residue distance constraints\n",
    "        \n",
    "        - Assemble 3D models\n",
    "          - Gradient descent\n",
    "          - Distance geometry\n",
    "          - Fragment assembly\n",
    "        \n",
    "        - Refine and rank models\n",
    "          - Landscape projection\n",
    "          - Refinement trajectory\n",
    "\n",
    "      - Protein design\n",
    "\n",
    "        - Fragment assembly\n",
    "        - Emphasize tightly packed hydrophobic core\n",
    "\n",
    "      - Protein–protein interactions\n",
    "\n",
    "        - Protein-protein docking\n",
    "        - Emphasize hydrogen bonding\n",
    "          - ensure good hydrophobic packing\n",
    "\n",
    "      - Ligand-binding proteins\n",
    "\n",
    "        - Ligand docking\n",
    "        - Emphasize hydrogen bonding\n",
    "          - ensure good hydrophobic packing\n",
    "\n",
    "      - Sensor applications\n",
    "\n",
    "        - pH switch\n",
    "\n",
    "        - lon switch\n",
    "\n",
    "        - Chemically-induced dimerization system\n",
    "\n",
    "        - Lockr\n",
    "\n",
    "# Biological Inorganic Chemistry\n",
    "\n",
    "  - Show the binding modes of the \n",
    "  side-chains of \n",
    "  amino acids \n",
    "  to d-metals.\n",
    "  - Understand the connection between the structure of a metalloprotein, \n",
    "  the electronic structure of a metal centre in biology and its function.\n",
    "  - List the structural and spectroscopic methods\n",
    "  - Explain crystal field theory and \n",
    "  apply it to derive approximate d-orbital splitting diagrams for some metal sites in biology.\n",
    "\n",
    "  - λ = 2dsinθ\n",
    "\n",
    "# X-ray\n",
    "\n",
    "  - why X-ray\n",
    "\n",
    "    - rational drug design\n",
    "      - structure based functional studies\n",
    "        - aid effective therapeutic agents and drugs development\n",
    "      - answers to structure questions\n",
    "        - global folds\n",
    "        - atomic details of bonding\n",
    "\n",
    "    - NMR \n",
    "      - no size limitation\n",
    "\n",
    "    - price\n",
    "      - a good crystal must be found\n",
    "      - only limited information \n",
    "      about the molecule's dynamic behavior is \n",
    "      available from one single diffraction experiment\n",
    "\n",
    "  - overview\n",
    "    - Single Crystal Formation\n",
    "      - single crystal exposed to X-ray source\n",
    "    - Data Collection\n",
    "      - detector\n",
    "        - scattered X-rays captured\n",
    "        - e.g., film\n",
    "      - producing diffraction pattern (precession photograph)\n",
    "    - Data Processing\n",
    "      - diffraction pattern → electron density map\n",
    "    - Model Building\n",
    "      - map + sequence = atomic structure of the protein\n",
    "\n",
    "  - features of X-ray diffractometer.\n",
    "  X-ray Source\n",
    "  rotating copper (Cu) anode \n",
    "  generates X-rays\n",
    "  50 keV electrons\n",
    "  Mirrors\n",
    "  X-ray mirrors focus and direct the X-ray beam toward the crystal.\n",
    "  Crystal Holder\n",
    "  The crystal is mounted on a rotating platform to expose it to X-rays at different angles.\n",
    "  Detector\n",
    "  Detects the diffracted X-rays\n",
    "  creating a pattern (red dots) that can be analyzed to determine the crystal structure.\n",
    "\n",
    "  - Eye and Microscope Analogy\n",
    "\n",
    "    Objects of macroscopic size are visible to us because they scatter\n",
    "    light falling on them.\n",
    "\n",
    "    The function of the eye lens is to bring together the light rays to\n",
    "    form an image on the retina.\n",
    "\n",
    "    Each scattered light ray has a particular intensity and a particular\n",
    "    phase. The relative intensities and phases determine the nature of\n",
    "    the image formed in the eye.\n",
    "\n",
    "    Information on the shape (structure) of the object is carried in the\n",
    "    intensities and phases of the light waves scattered by it.\n",
    "\n",
    "    For smaller objects the eye needs help from more powerful\n",
    "    lenses. The scattered light is collected by the microscope and\n",
    "    refracted to bring it into focus in the observer’s eye. It is not\n",
    "    necessary to collect all of the light but note that the image\n",
    "    becomes less clear as the proportion of light collected is reduced.\n",
    "\n",
    "  - Scattering\n",
    "\n",
    "    The lower limit on the size of objects which can be clearly seen with a\n",
    "    microscope is determined by the wavelength of light (400 -700 nm).\n",
    "    Objects that are much smaller than this do not give any significant\n",
    "    scattering of light.\n",
    "\n",
    "    To ‘see’ atoms, molecules and proteins (ca. 100 to 1000 times smaller\n",
    "    than the wavelength of visible light) requires the use of X-rays. In\n",
    "    principle we require an X-ray microscope in order to observe molecular\n",
    "    structure. Such an instrument does not exist. WHY?\n",
    "\n",
    "    The situation is not hopeless because the pattern of scattered X-rays\n",
    "    can be recorded and then the recombination which is impossible\n",
    "    physically can be achieved mathematically.\n",
    "\n",
    "    However, to get an interpretable scattering pattern we must use\n",
    "    crystalline material WHY?\n",
    "\n",
    "    When the method is successful it provides and image of the molecular structure. It locates the\n",
    "    components of the material which interact with the incident X-rays i.e. the electron density.\n",
    "\n",
    "    The total time taken to record the scattering pattern is usually hours or days.\n",
    "    WHAT ARE THE IMPLICATIONS OF THIS?\n",
    "\n",
    "    Some information is lost in the experiment. When the X-ray scattering is recorded the individual\n",
    "    wave amplitudes are retained as relative intensities but the relative phases are lost ÷ THE\n",
    "    PHASE PROBLEM.\n",
    "\n",
    "  - Compare and contrast magnification and diffraction as ways of viewing small objects.\n",
    "\n",
    "  - Bragg Equation\n",
    "\n",
    "    - Individual rays have\n",
    "      - path differences of a whole numberwave lengths (λ) for the waves to \n",
    "        - remain in phase\n",
    "\n",
    "      - leads to\n",
    "        - constructive interference\n",
    "        - produce diffraction spot\n",
    "    \n",
    "    - Bragg equation show\n",
    "      - lattice spacing (d) \n",
    "      - angle of incidence (θ) \n",
    "\n",
    "      - greater θ \n",
    "      - smaller d \n",
    "      - greater resolution\n",
    "\n",
    "  - Resolution\n",
    "\n",
    "    - 5-6 Å\n",
    "      - general shape of protein discerned\n",
    "      - α-helices might be identified\n",
    "\n",
    "    - 3.5-2.5 Å\n",
    "      - polypeptide back-bone traced\n",
    "\n",
    "    - 1.5 Å\n",
    "      - crystal structure approaches atomic resolution\n",
    "\n",
    "  - Intensities of Diffracted X-rays\n",
    "\n",
    "    The intensities of the spots \n",
    "    in the diffraction pattern \n",
    "    the arrangement of atoms in the crystal structure are \n",
    "    related to FOURIER TRANSFORM\n",
    "\n",
    "      electron density is the Fourier transform of the diffraction pattern.\n",
    "\n",
    "  - Process of Refining the Structure\n",
    "\n",
    "    Knowing the position of the heavy atom \n",
    "    enables us to calculate\n",
    "    phase φc\n",
    "    intensity Fc.\n",
    "\n",
    "    combine the experimental intensities Fo with φc to arrive \n",
    "    at a new calculated ρ(x,y,z)\n",
    "    identify more atoms\n",
    "    recalculate \n",
    "    new φc\n",
    "    Fc\n",
    "    \n",
    "    Combine Fo and φc\n",
    "    Have a guess\n",
    "    Re-calculate Fc and φc for all diffraction spots\n",
    "\n",
    "    reliable structure R<0.25.\n",
    "\n",
    "  - phase problem\n",
    "\n",
    "    - Some information is lost\n",
    "    - X-ray scattering is recorded\n",
    "    - wave amplitudes are retained\n",
    "    - relative phases are lost\n",
    "\n",
    "    - approaches\n",
    "\n",
    "      - Isomorphous replacement\n",
    "        - insert a heavy metal atom into the protein without destroying the crystal\n",
    "        - by soaking a protein crystal in a solution containing a heavy metal salt\n",
    "\n",
    "      - Imagine in real space\n",
    "        - creating a 'difference crystal' \n",
    "        - by subtracting the native crystal from the derivative crystal\n",
    "        - dominated by the high number of electrons associated with the heavy atom\n",
    "\n",
    "      - Patterson Synthesis\n",
    "        - leads to a map that resembles an electron density map. \n",
    "        - contains sufficient information to locate the position of the heavy atoms in the structure.\n",
    "        - calculation, all of the phases to zero\n",
    "        - use the intensities of this difference diffraction pattern\n",
    "\n",
    "  - How Reliable\n",
    "\n",
    "    - resolution\n",
    "    - R factors\n",
    "    - temperature factors, B\n",
    "\n",
    "# XAS\n",
    "\n",
    "  - X-ray Absorption Spectroscopy\n",
    "\n",
    "  - Measurement\n",
    "    - wavelengths around the absorption “edge” \n",
    "      - element of interest\n",
    "\n",
    "  - Utility\n",
    "    - electronic information\n",
    "    - geometric structure information\n",
    "    - a local region around specific elements \n",
    "    - in amorphous samples\n",
    "\n",
    "  - Phenomena Type\n",
    "    - Absorption edge\n",
    "      - energy dependent on\n",
    "        - charge density of absorbing atom \n",
    "        - oxidation state\n",
    "\n",
    "    - Absorption pre-edge\n",
    "      - energy and intensity\n",
    "      - covalency of d-orbitals\n",
    "\n",
    "    - Extended structure\n",
    "\n",
    "      - EXAFS\n",
    "\n",
    "  - Extended X-ray Absorption Fine Structure\n",
    "  \n",
    "    - Oscillations\n",
    "      - caused by interference of out-going electron wave \n",
    "        - by backscattered electrons from neighbouring atoms\n",
    "      - pattern of oscillations\n",
    "        - individual sine waves associated with scattering \n",
    "        - caused by each shell of coordinated atoms\n",
    "    \n",
    "    - interference (constructive, destructive) depends on\n",
    "      - X-rays energy (wavelength)\n",
    "      - nature of scattering atom\n",
    "      - distance from metal centre\n",
    "\n",
    "    - Observable vs. Information\n",
    "      - Frequency\n",
    "        - Distance\n",
    "      - Amplitude\n",
    "        - Coordination number\n",
    "      - Amplitude and phase\n",
    "        - Atom type\n",
    "\n",
    "    - Frequency-Distance Relationship\n",
    "      - longer metal-ligand distance\n",
    "      \n",
    "        - more periods of the photelectron wave are required to cover the distance from the metal to the ligand and back\n",
    "\n",
    "        - takes less change in photoelectron wavelength to go through one period of interference. \n",
    "        \n",
    "        - abosorption is modulated at a higher frequency\n",
    "\n",
    "    - Amplitude-Coordination Number Relationship\n",
    "\n",
    "      - coordination with two identical atoms\n",
    "        - e.g. \n",
    "          - two O donors\n",
    "          - same distance\n",
    "        - each contribute equally to amplitude of sine wave\n",
    "\n",
    "    - limitation of EXAFS\n",
    "      - only radial information\n",
    "      - not spatial information\n",
    "\n",
    "      - radial dependence\n",
    "        - shells of atoms\n",
    "      - One shell of atoms\n",
    "        - one damped sine wave\n",
    "\n",
    "    - Phase and Amplitude-Atom Relationship\n",
    "      - interaction   \n",
    "        - outgoing photoelectron wave\n",
    "        - electron density of a scattering atom, \n",
    "      - undergoes a phase shift\n",
    "\n",
    "      - Variations in electron density (atoms)\n",
    "        - different phase shifts\n",
    "      - Heavier atoms \n",
    "        - better scatterers of X-rays\n",
    "        - yeild EXAFS with larger amplitudes\n",
    "\n",
    "    - Advantages\n",
    "      - No requirement for crystalline material and \n",
    "        - different forms of a system can be investigated\n",
    "      - Element specific.\n",
    "      - Samples ~1mM in element can be probed\n",
    "      - EXAFS provides accurate ± 0.03 Å interatomic distances\n",
    "      - Studies f(time)\n",
    "\n",
    "    - Disadvantages\n",
    "      - Radiation Damage\n",
    "      - Data analysis complicated\n",
    "        - extensive fitting required\n",
    "      - Detailed angular information is not available\n",
    "      - only gives an average coordination sphere\n",
    "      - Interatomic distances restricted to the region within ca. 3.5 Å of the primary absorber\n",
    "      - difficult to distinguish\n",
    "        - N scatterers\n",
    "        - O scatterers \n",
    "\n",
    "    - Interpretation\n",
    "      - Identifies relationship between\n",
    "        - oxidation state of the metal, V\n",
    "        - metal-ligand (j) distances, r\n",
    "        - coordination number\n",
    "\n",
    "    - Synchrotron Sources\n",
    "\n",
    "      - EXAFS technique is inherently insensitive\n",
    "      - ca. 1% of intensity of edge\n",
    "      - X-ray source\n",
    "        - powerful\n",
    "          - small signal can be extracted from a large background\n",
    "\n",
    "      - Synchrotron radiation\n",
    "        - first became available in the 1970s\n",
    "        - revolutionised \n",
    "          - XAS\n",
    "          - protein crystallography\n",
    "        - bright\n",
    "          - high intensity\n",
    "          - small spot size\n",
    "          - tunable over 2-30 keV range\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "    - Bond Valence Sum\n",
    "      - analyse EXAFS data and\n",
    "      - predict EXAFS spectra for metal centres in protein\n",
    "\n",
    "\n",
    "# EPR\n",
    "\n",
    "  - Fundamentals\n",
    "\n",
    "    - Electron Paramagnetic Resonance\n",
    "      - keep\n",
    "        - electromagnetic radiation frequency constant (ν)\n",
    "      - scan\n",
    "        - magnetic field (Bo)\n",
    "        - detect dA/dBo\n",
    "      - h\n",
    "        - Planck’s constant\n",
    "      - μB\n",
    "        - Bohr magneton constant\n",
    "    - Zeeman Effect\n",
    "\n",
    "    - Hyperfine Coupling\n",
    "\n",
    "  - Compared to NMR spectroscopy\n",
    "    \n",
    "    - g\n",
    "      - chemical shift\n",
    "\n",
    "    - A\n",
    "      - coupling constant\n",
    "\n",
    "    - unique fingerprints\n",
    "      - compound identification\n",
    "\n",
    "  - EPR Spectrometer\n",
    "\n",
    "    - Microwave Bridge\n",
    "    - Waveguide\n",
    "    - Sample Space\n",
    "    - Magnet\n",
    "    - Console\n",
    "    - Microwave Frequency Bands\n",
    "\n",
    "\n",
    "  - EPR Spectra for S = ½ Transition Metal Centres\n",
    "\n",
    "    - unpaired electron\n",
    "      - associated with at least on transition metal ion\n",
    "\n",
    "    - anisotropic system\n",
    "      - g no longer equal 2.0023\n",
    "    \n",
    "      - Anisotropy in g\n",
    "\n",
    "        - gxx = gyy = gzz\n",
    "          - Isotropic \n",
    "            - Oh\n",
    "            - Td\n",
    "\n",
    "        - gxx = gyy(g∥) ≠ gzz(g⊥)\n",
    "          - Axial\n",
    "            - square pyramidal\n",
    "            - trigonal pyramidal\n",
    "            - flattened tetrahedron\n",
    "            - square plane\n",
    "\n",
    "        - gxx ≠ gyy ≠ gzz\n",
    "          - Rhombic Low symmetry\n",
    "\n",
    "  - g values and the orientation of Bo\n",
    "\n",
    "    - EPR signal\n",
    "      - orientation relative to the magnetic field Bo\n",
    "\n",
    "    - frozen solution\n",
    "      - 77 K\n",
    "    \n",
    "    - frozen solution spectrum\n",
    "      - summation all different orientations\n",
    "\n",
    "    - d transition metal compounds\n",
    "      - performed in fluid solution\n",
    "      - anisotropic molecule\n",
    "        - tumbling more rapidly\n",
    "        - than experiment timescale\n",
    "        - anisotropy is averaged\n",
    "\n",
    "    - g = 2.0023\n",
    "      - g value for a free e\n",
    "      - ‘spin only’ value\n",
    "\n",
    "  - e.g.\n",
    "\n",
    "    - Cu(II)\n",
    "\n",
    "      - Elongated octahedron\n",
    "        - g‖ > g⊥\n",
    "        - unpaired e in dx²-y² orbital\n",
    "\n",
    "      - Compressed octahedron\n",
    "        - g⊥ > g‖\n",
    "        - unpaired e in dz² orbital\n",
    "\n",
    "    - Hyperfine Couplings\n",
    "\n",
    "      - CuN2O2 coordination sphere\n",
    "\n",
    "      - I=3/2\n",
    "        - 63Cu (0.69%)\n",
    "        - 65Cu (0.31%)\n",
    "        - 2nI+1 lines\n",
    "    \n",
    "      - I=1\n",
    "        - 14N\n",
    "        - 2nI+1 lines\n",
    "      \n",
    "      - Metal hyperfine reveal\n",
    "        - %metal covalency in the orbital containing the unpaired electron.\n",
    "\n",
    "      - Cu(II) Protein\n",
    "        - Type I Blue Copper\n",
    "\n",
    "# 10 interview\n",
    "\n",
    "# 50 HCC & CRLM\n",
    "\n",
    "## 50.0 Acquisition / Processing\n",
    "\n",
    "  - TCGA database \n",
    "    - RNA expression profiles\n",
    "    - clinical data\n",
    "    - via UCSC Xena platform\n",
    "\n",
    "  - survival analysis\n",
    "    - 368 samples\n",
    "  \n",
    "  - Processing\n",
    "    - TPM format\n",
    "    - log2-transformed\n",
    "\n",
    "  - sc\n",
    "\n",
    "    - sc dataset\n",
    "      - GEO\n",
    "      - 10 HCC\n",
    "      - 9 CRLM\n",
    "\n",
    "    - Quality control criteria\n",
    "      - mitochondrial content\n",
    "        - ＜ 20%\n",
    "      - UMI counts\n",
    "        - 200-20,000\n",
    "      - gene counts\n",
    "        - 200-5,000\n",
    "\n",
    "    - Seurat functions \n",
    "      - NormalizeData\n",
    "        - Data normalization\n",
    "      - FindVariableFeatures\n",
    "        - selection of highly variable genes\n",
    "        - 2000 genes\n",
    "      - ScaleData\n",
    "        - data transformation\n",
    "        - eliminating cell cycle effects with the parameters \n",
    "        - vars.to.regress = c(\"S.Score\", \"G2M.Score\")\n",
    "\n",
    "      - Dimensionality reduction methods \n",
    "        - UMAP\n",
    "        - TSNE\n",
    "        - Louvain\n",
    "\n",
    "    - Batch effect correction\n",
    "      - Harmony\n",
    "\n",
    "    - Differential gene expression\n",
    "      - Between\n",
    "        - clusters\n",
    "        - cell types\n",
    "      - FindAllMarkers function\n",
    "        - p-value < 0.05, log2FC > 0.25\n",
    "        - expression proportion > 0.1\n",
    "\n",
    "  - ST\n",
    "\n",
    "    - HCC ST\n",
    "      - 4 HCC samples\n",
    "    - CRLM ST\n",
    "      - 4 CRLM samples\n",
    "\n",
    "    - SpaceRanger software\n",
    "      - Post-quality control\n",
    "\n",
    "    - SCTtransform algorithm\n",
    "      - data normalization\n",
    "      - selection of highly variable genes\n",
    "      - data transformation\n",
    "\n",
    "    - average number of spots\n",
    "      - 4054\n",
    "    - average UMI\n",
    "      - 15317.4\n",
    "    - gene counts\n",
    "      - 3890.1\n",
    "    - mitochondrial content\n",
    "      - 4.5%\n",
    "\n",
    "    - Seurat\n",
    "      - Analysis\n",
    "      - visualization\n",
    "\n",
    "    - CARD algorithm\n",
    "      - conditional autoregression based deconvolution\n",
    "      - Deconvolution analysis\n",
    "\n",
    "    - single-cell annotations\n",
    "      - cell type predictions for each spot\n",
    "\n",
    "    - CARD software\n",
    "      - Visualization of cell types in ST\n",
    "\n",
    "    - Signature scores\n",
    "      - Seurat\n",
    "        - GSVA package\n",
    "        - AddModelScore\n",
    "\n",
    "## 50.1 sc Expression Profile\n",
    "\n",
    "  - Markers\n",
    "\n",
    "    - epithelial cell\n",
    "      - EPCAM\n",
    "      - KRT18\n",
    "      - KRT19\n",
    "      - CDH1\n",
    "\n",
    "    - fibroblast\n",
    "      - DCN\n",
    "      - THY1\n",
    "      - COL1A1\n",
    "      - COL1A2\n",
    "\n",
    "    - endothelial cell\n",
    "      - PECAM1\n",
    "      - CLDN5\n",
    "      - FLT1\n",
    "      - RAMP2\n",
    "\n",
    "    - T cell \n",
    "      - CD3D\n",
    "      - CD3E\n",
    "      - CD3G\n",
    "      - TRAC\n",
    "\n",
    "    - NK cell \n",
    "      - NKG7\n",
    "      - GNLY\n",
    "      - NCAM1\n",
    "      - KLRD1\n",
    "\n",
    "    - B cell\n",
    "      - CD79A\n",
    "      - IGHM\n",
    "      - IGHG3\n",
    "      - IGHA2\n",
    "\n",
    "    - mast cell \n",
    "      - KIT\n",
    "      - MS4A2\n",
    "      - GATA2\n",
    "\n",
    "  - Pearson correlation matrix\n",
    "    - correlation between two continuous variables\n",
    "      - clustering results\n",
    "      - cell type classification\n",
    "    - marker gene expression\n",
    "    - heatmap\n",
    "\n",
    "  - Marker gene expression\n",
    "    - heatmaps\n",
    "    - tSNE plots\n",
    "    - confirming cell-type identification\n",
    "\n",
    "  - Comparisons\n",
    "    - cell type composition\n",
    "    - patient origin\n",
    "    - cell count\n",
    "    - transcript detection\n",
    "    - CNV status\n",
    "\n",
    "## 50.2 ST\n",
    "\n",
    "  - HCC enriched\n",
    "    - LIHCT\n",
    "    - tumor cells\n",
    "  - CRLM enriched\n",
    "    - LIHC\n",
    "    - fibroblasts\n",
    "    - endothelial cells\n",
    "    - myeloid cells\n",
    "\n",
    "  - CNV detection\n",
    "    - InferCNV software\n",
    "    - CNV score distribution\n",
    "\n",
    "  - deconvolution analysis\n",
    "    - CARD algorithm\n",
    "    - based on single-cell data\n",
    "    - cell type distribution\n",
    "\n",
    "    \n",
    "  - H&E staining images\n",
    "\n",
    "  - tumor vs. non-tumor classification\n",
    "\n",
    "    - dimensionality reduction\n",
    "    - clustering analysis\n",
    "\n",
    "\n",
    "\n",
    "    - tumor cells\n",
    "    8 subgroups\n",
    "    LIHCT1\n",
    "    LIHCT3\n",
    "    LIHCT4\n",
    "\n",
    "    tSNE plot\n",
    "    subgroup composition\n",
    "    sample origin\n",
    "    CNV status\n",
    "    patient composition\n",
    "\n",
    "    Marker genes for each subgroup were calculated\n",
    "    expression heatmaps were created\n",
    "\n",
    "\n",
    "\n",
    "    - non-tumor cells\n",
    "    5 subgroups\n",
    "    endothelial cells\n",
    "    tumor cells\n",
    "    myeloid cells\n",
    "\n",
    "    LIHCT2 enrichment\n",
    "\n",
    "    tSNE plot\n",
    "    marker gene expression heatmaps\n",
    "\n",
    "## 50.3 Subclassification Analysis of Tumor Cells\n",
    "\n",
    "Tumor cells were isolated and subjected to dimensionality reduction and clustering analysis, resulting in 16 clusters. These clusters were then merged based on their marker gene functions, yielding 7 tumor cell subgroups (Figure 3A). A bar plot displays the cell composition of each sample, showing that HCC samples have a higher content of APOA2+NEO cells, while CRLM samples have a higher content of TFF3+NEO cells. Differential gene expression analysis was performed for each subgroup, and a heatmap of differentially expressed genes was generated (Figure 3B). GO Biological Process (GOBP) enrichment analysis was conducted for the marker genes of each subgroup, revealing their corresponding functions. We compared the CNV differences among different cell types from different sample origins, finding significant differences in CNV across most cell types (Figure 3C). Transcription factor analysis was also performed for each cell type, with the results visualized in a heatmap showing different transcription factors enriched in various cell types (Figure 3D).\n",
    "\n",
    "Subsequent cell trajectory analysis revealed multiple paths leading to different endpoints, indicating the potential existence of different subtypes. The starting branches contained a large number of APOA2+NEO cells and tumor cells from HCC (Figure 3E). From the pseudotime analysis, APOA2+NEO cells were primarily at the starting points, while most other cells were at the endpoints. Genes related to pseudotime were identified, and their expression heatmap was generated (Figure 3F). Scatter plots of pseudotime for six selected tumor marker genes (APOA2, HIST1H2AG, HMGB2, IGKV1-5, LCN2, TFF3) were created (Figure 3G).\n",
    "\n",
    "## 50.3 Cell Annotation Analysis\n",
    "\n",
    "  - tumor heterogeneity clustering\n",
    "    - epithelial cells\n",
    "    - immune cells\n",
    "    - fibroblasts\n",
    "\n",
    "\n",
    "## 50.4 Subgroup Analysis of Cell Populations\n",
    "\n",
    "  - further distinction for subgroups\n",
    "    - Immune cells\n",
    "    - epithelial cells\n",
    "    - fibroblasts\n",
    "\n",
    "  - standard Seurat workflow\n",
    "  - specific markers for each subgroup\n",
    "  - UMAP plots\n",
    "\n",
    "## 50.5 CNV Analysis of Single Cells\n",
    "\n",
    "  - CNVs in tumor cell subgroups\n",
    "    - InferCNV\n",
    "    - references\n",
    "      - fibroblasts\n",
    "      - endothelial cells\n",
    "    - identify malignant cells within subgroups\n",
    "\n",
    "## 50.6 Pseudotime Analysis of Single Cells\n",
    "\n",
    "  - Monocle2 software\n",
    "    - pseudotime analysis\n",
    "      - epithelial\n",
    "      - macrophage\n",
    "      - CAF cell\n",
    "\n",
    "  - DDRTree\n",
    "    - dimensionality reduction algorithm\n",
    "    - parameters set to default\n",
    "    - cell differentiation\n",
    "\n",
    "## 50.7 Transcription Factor Analysis\n",
    "\n",
    "  - SCENIC software\n",
    "    - transcription factor analysis across cell subgroups\n",
    "    - default parameters\n",
    "      - RcisTarget\n",
    "      - GRNBoost motif databases\n",
    "\n",
    "  - RcisTarget package\n",
    "    - overexpressed transcription factor binding motifs\n",
    "\n",
    "  - AUCell package\n",
    "    - scored the activity of each regulatory module for each cell type\n",
    "\n",
    "## 50.8 Cell-Cell Communication Analysis\n",
    "\n",
    "  - cell-cell communication\n",
    "    - CellChat package\n",
    "    - normalized gene expression matrix was imported\n",
    "\n",
    "  - Data preprocessing\n",
    "    - identifyOverExpressedGenes\n",
    "    - identifyOverExpressedInteraction\n",
    "    - ProjectData\n",
    "    - default parameters\n",
    "\n",
    "  - ligand-receptor interactions\n",
    "    - computeCommunProb\n",
    "    - filterCommunication\n",
    "    - computeCommunProbPathway\n",
    "\n",
    "  - cell communication network\n",
    "    - aggregateNet function\n",
    "\n",
    "## 50.9 Statistical Analysis\n",
    "\n",
    "  - R 4.1.3\n",
    "    - data processing\n",
    "    - statistical analysis\n",
    "    - plotting\n",
    "\n",
    "  - Pearson correlation coefficient\n",
    "    - correlation between two continuous variables\n",
    "\n",
    "  - chi-square test\n",
    "    - categorical variables\n",
    "\n",
    "  - continuous variables\n",
    "    - Wilcoxon rank-sum test\n",
    "    - T-test\n",
    "\n",
    "  - survminer package\n",
    "    - determine optimal cutoff values\n",
    "\n",
    "  - survival package\n",
    "    - Cox regression\n",
    "    - Kaplan-Meier analyses"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- “萧何”用户使用手册\n",
    "\n",
    "    - “分形”知识结构\n",
    "\n",
    "        - 模块化\n",
    "\n",
    "        - 自相似\n",
    "\n",
    "        - 分布与集成的辩证统一\n",
    "        \n",
    "        - “郡县”与“分封”制度的优劣\n",
    "\n",
    "    - 认知负荷理论\n",
    "\n",
    "        - 最小记忆原理\n",
    "\n",
    "    - “刻意”"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
