{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# All in One\n",
    "\n",
    "  - 1 Quantification\n",
    "\n",
    "      - General UV Absorbance\n",
    "\n",
    "        - Major\n",
    "          - Try (Trp)\n",
    "          - Tyr\n",
    "        - Minor \n",
    "          - Cys (–S-S-)\n",
    "\n",
    "\n",
    "        - Easy\n",
    "        - Sample Recoverable\n",
    "        - Interference\n",
    "          - Chromophores\n",
    "          - Nucleic acids \n",
    "\n",
    "      - Far UV Absorbance\n",
    "\n",
    "        - 190 nm\n",
    "\n",
    "        - Advantages\n",
    "          - Detect from peptide bonds\n",
    "\n",
    "        - Disadvantages\n",
    "          - Interference at 190 nm\n",
    "            - O2\n",
    "          - Interference at 205 nm\n",
    "            - buffer\n",
    "            - heme\n",
    "            - pyridoxal group\n",
    "          - Low sensitivity (205 nm → 190 nm)\n",
    "          - Calibration Requirement\n",
    "\n",
    "      - Near UV Absorbance\n",
    "\n",
    "        - 280 nm\n",
    "\n",
    "        - Major\n",
    "          - Try (Trp)\n",
    "            - 5690\n",
    "          - Tyr\n",
    "            - 1280\n",
    "        - Minor\n",
    "          - Cys\n",
    "            - 120\n",
    "\n",
    "        - A₂₈₀ (1mg/ml) = (5690 Trp + 1280 Tyr + 120 Cys) / M\n",
    "\n",
    "        - Reliable and Easy\n",
    "        - No Buffer Absorption\n",
    "        - Correction for Nucleic Acids\n",
    "          - Protein (mg/ml) =\n",
    "          - 1.55\n",
    "          - × A₂₈₀\n",
    "          - (- 0.76)\n",
    "          - × A₂₆₀\n",
    "\n",
    "      - Lowry Method\n",
    "\n",
    "        - Based on 2 reactions\n",
    "\n",
    "          - Biuret Reaction\n",
    "            - Cu²⁺ + \n",
    "            - peptide bonds + \n",
    "            - alkaline conditions → \n",
    "            - Biuret chromophore\n",
    "\n",
    "          - Folin-Ciocalteu Reaction\n",
    "            - Tyr, Try (Trp) +\n",
    "            - phosphomolybdate (H₃PMo₁₂O₄₀) +\n",
    "            - phosphotungstate (H₃PW₁₂O₄₀) → \n",
    "            - Heteropolymolybdenum blue\n",
    "\n",
    "        - Strong Blue color\n",
    "          - Detectable within 500-750 nm\n",
    "          - Commonly chosen for measurement\n",
    "            - 660 nm\n",
    "            - 750 nm\n",
    "\n",
    "        - Advantages\n",
    "          - High Sensitivity\n",
    "            - 100-Fold than Biuret\n",
    "            - 0.01 mg/ml\n",
    "          - Strong Color Development\n",
    "          - Wider Wavelength Range\n",
    "            - 500-750 nm\n",
    "\n",
    "        - Disadvantages\n",
    "          - Bind preference\n",
    "            - Tyr\n",
    "            - Try (Trp)\n",
    "          - Interference\n",
    "            - Red\n",
    "          - Complexity\n",
    "\n",
    "      - BCA assay\n",
    "        - Bicinchoninic Acid\n",
    "        - Similar to Lowry method\n",
    "        - Cu(II→I) under HO-\n",
    "        - Cu(I) + BCA = purple-colored complex\n",
    "        - 562 nm\n",
    "\n",
    "        - Advantages\n",
    "          - One-Step Process (Lowry two-step)\n",
    "          - Sensitivity ＞ Lowry method\n",
    "          - Suitable for low protein concentrations\n",
    "\n",
    "        - Disadvantages\n",
    "          - Cu(I) amount affect by\n",
    "            - protein concentration\n",
    "            - incubation time\n",
    "          - Requiring precise control\n",
    "\n",
    "      - Bradford method\n",
    "        - Coomassie Blue G250 dye\n",
    "        - Free dye in 3 ionic forms\n",
    "          - Cationic\n",
    "            - orange-red\n",
    "            - 470 nm\n",
    "          - Anionic\n",
    "            - blue\n",
    "            - bind to proteins\n",
    "            - 590 nm\n",
    "\n",
    "        - 2 absorption spectra Overlap\n",
    "          - Non-linear standard curve\n",
    "\n",
    "        - Simple and quick assay\n",
    "\n",
    "        - Bind preference\n",
    "          - Arg\n",
    "          - Lys\n",
    "\n",
    "  - 2 Electrophoresis\n",
    "\n",
    "    - Polyacrylamide Gels\n",
    "\n",
    "      - Polymerisation\n",
    "\n",
    "        - Acrylamide\n",
    "          - CH2=CHCONH2\n",
    "\n",
    "        - N,N’-methylene-bisacrylamide\n",
    "          - CH2(NHCOCH=CH2)2\n",
    "\n",
    "      - Radical generation catalysis\n",
    "\n",
    "        - APS\n",
    "          - ammonium persulfate\n",
    "        - TEMED\n",
    "          - N,N,N’,N’-tetramethylenediamine\n",
    "\n",
    "    - SDS-PAGE\n",
    "\n",
    "      - Stacking gel\n",
    "\n",
    "        - Short gel on top\n",
    "        - Sample Stacking\n",
    "        - Low conc. acrylamide\n",
    "        - Low ionic strength\n",
    "        - Large pore\n",
    "        - pH\n",
    "          - 6.9\n",
    "        - Charge carrier\n",
    "          - Cl-\n",
    "        - glycine form\n",
    "          - Zwitterion\n",
    "\n",
    "      - Resolving gel\n",
    "        - pH\n",
    "          - 8-9\n",
    "        - glycine form\n",
    "          - Anion\n",
    "        - Charge carrier\n",
    "        - glycinate ions\n",
    "\n",
    "        - Tris glycine pH\n",
    "          - 8-9\n",
    "\n",
    "      - Denaturing\n",
    "        - SDS\n",
    "          - Anionic detergent\n",
    "        - β-mercaptoethanol\n",
    "          - reduces disulfide bridges\n",
    "\n",
    "      - Migration\n",
    "        - size\n",
    "\n",
    "    - Native PAGE\n",
    "\n",
    "      - Preserving \n",
    "        - tertiary structures\n",
    "        - quaternary structures\n",
    "      - Enables study\n",
    "        - subunits interactions\n",
    "        - different modifications\n",
    "      - Migration dependent on\n",
    "        - size\n",
    "        - shape\n",
    "        - charge\n",
    "      - Applications\n",
    "        - Apoferritin\n",
    "          - 24 subunits\n",
    "          - Self-assemble → protein nanocage\n",
    "            - different ratios\n",
    "              - modified apoferritin\n",
    "                - Z_HER2-Apoferritin\n",
    "                - 27 kDa\n",
    "              - unmodified apoferritin\n",
    "                - Apoferritin\n",
    "                - 22 kDa\n",
    "\n",
    "    - Gradient SDS-PAGE\n",
    "\n",
    "      - Acrylamide conc. ↑\n",
    "      - Pore size ↓\n",
    "      - Wide protein MW\n",
    "      - Higher resolution\n",
    "      - Complicated to Set Up\n",
    "\n",
    "    - IEF\n",
    "      - Isoelectric Focusing\n",
    "      - Low% polyacrylamide\n",
    "      - Ampholytes\n",
    "      - Electric field induces →\n",
    "        - pH gradient\n",
    "      - Proteins stop at pI\n",
    "\n",
    "    - 2D Electrophoresis\n",
    "      - 1st Dimension\n",
    "        - pI\n",
    "      - 2nd Dimension\n",
    "        - size\n",
    "\n",
    "      - Proteomics\n",
    "\n",
    "        - Define Biological Question\n",
    "        - Conduct Sample Preparation\n",
    "        - Perform 2-D Gel Electrophoresis\n",
    "        - Capture results through Imaging\n",
    "        - Extract proteins from gel\n",
    "        - Protein Digestion for analysis\n",
    "        - MS: Determine protein identity\n",
    "        - Global Bioinformatics Analysis\n",
    "\n",
    "  - 6 Yeast two/three-Hybrid\n",
    "\n",
    "    - Native yeast GAL4 protein\n",
    "\n",
    "      - transcriptional activator\n",
    "\n",
    "      - DNA-Binding domain (DB)\n",
    "        - aa 1-147\n",
    "        - bind UAS\n",
    "\n",
    "      - Activation Domain (AD)\n",
    "        - aa 768-881\n",
    "        - directs RNA pol. II\n",
    "        - transcribe gene downstream UAS\n",
    "\n",
    "    - Selection Reporter\n",
    "      - lacZ\n",
    "        - β-galactosidase\n",
    "      - Nutritional marker \n",
    "        - adenine\n",
    "        - histidine\n",
    "        - in media lacking adenine / histidine\n",
    "      - other reporter systems\n",
    "\n",
    "    - Advantages\n",
    "      - In vivo interaction\n",
    "      - Detect transient interaction\n",
    "      - identify single aa in interaction\n",
    "      - identify new protein interactant\n",
    "\n",
    "    - Disadvantages\n",
    "      - Difficult and time consuming\n",
    "      - Many false +ve\n",
    "      - Yeast grow slowly\n",
    "\n",
    "    - three-Hybrid\n",
    "\n",
    "  - 14 PTM\n",
    "\n",
    "    - Phosphorylation\n",
    "      - HO-\n",
    "        - by kinase\n",
    "        - covalently addition\n",
    "        - reversible by phosphatases\n",
    "      - change protein interactions\n",
    "      - control protein across membranes\n",
    "      - Abnormal phosphorylation\n",
    "        - cancer\n",
    "        - degenerative disorders\n",
    "\n",
    "    - Glycosylation\n",
    "      - + Carbohydrates → glycoprotein\n",
    "      - N-Glycans linked\n",
    "        - Asn\n",
    "      - O-Glycans linked \n",
    "        - Ser\n",
    "        - Thr\n",
    "      - Essential in protein folding\n",
    "      - Major role\n",
    "        - cell–cell recognition\n",
    "        - signalling\n",
    "\n",
    "  - 15 Role Protein\n",
    "\n",
    "    - Actin\n",
    "      - Structural proteins\n",
    "      - Most abundant protein\n",
    "      - Main component of cytoskeleton\n",
    "      - 42 kDa \n",
    "      - Globular protein\n",
    "      - Forms fibrils throughout cell\n",
    "\n",
    "    - Ferritin\n",
    "      - Mineral storage\n",
    "      - Function for Fe\n",
    "        - oxidise\n",
    "        - incorporate\n",
    "        - keeps in non-toxic form\n",
    "      - 24-mer structure\n",
    "        - large cavity\n",
    "          - iron storage\n",
    "        - separated from solution\n",
    "          - by 2 nm protein shell\n",
    "\n",
    "    - Ion channels\n",
    "      - Membrane transport\n",
    "      - Integral membrane protein\n",
    "      - Circular arrangement\n",
    "        - packed around waterfilled pore\n",
    "        - location\n",
    "          - membrane\n",
    "          - lipid bilayer\n",
    "\n",
    "\n",
    "\n",
    "# 100 E. coli Strains\n",
    "\n",
    "  - BL21\n",
    "    - deficient in lon and ompT proteases\n",
    "      - avoid excessive protein degradation\n",
    "\n",
    "  - BLR\n",
    "    - recA- 重组酶A缺失\n",
    "      - ↑ plasmid monomer yields\n",
    "      - Stabilize repetitive sequence plasmids\n",
    "\n",
    "  - B834\n",
    "    - Methionine auxotroph 甲硫氨酸缺陷型菌株\n",
    "      - labels S35/Se for crystallography\n",
    "\n",
    "  - AD494\n",
    "    - Thioredoxin reductase deficient (trxB-) 硫氧还蛋白还原酶缺失型\n",
    "      - Aid disulfidebond formation\n",
    "\n",
    "  - Rosetta strains\n",
    "    - express rare codons proteins\n",
    "\n",
    "  - MCS\n",
    "    - Multiple Cloning Site\n",
    "    - insert gene through multiple restriction sites\n",
    "\n",
    "# 200 Thermostable DNA polymerase\n",
    "\n",
    "  - Automation\n",
    "  - Accuracy\n",
    "  - Processivity\n",
    "  - Stability\n",
    "\n",
    "  - e.g.\n",
    "\n",
    "    - Taq Polymerase\n",
    "      - Thermus aquaticus\n",
    "      - High processivity\n",
    "      - Lower fidelity (no 3’-5’ exonuclease)\n",
    "      - Activity lost\n",
    "        - 50%\n",
    "        - 30 min\n",
    "        - 95°C\n",
    "      - Add an extra A base at 3’-end\n",
    "\n",
    "    - Vent Polymerase\n",
    "      - Thermococcus litoralis\n",
    "      - Lower processivity than Taq\n",
    "      - Higher fidelity (3’-5’ exonuclease)\n",
    "      - Activity lost\n",
    "        - 10%\n",
    "        - 60 min\n",
    "        - 98°C\n",
    "      - Forms blunt-ended\n",
    "\n",
    "    - Deep Vent Polymerase\n",
    "      - Pyrococcus\n",
    "      - Lower processivity than Taq\n",
    "      - Higher fidelity (3’-5’ exonuclease)\n",
    "      - Extremely stable at high temperatures\n",
    "      - Grows at\n",
    "        - 2000 metres\n",
    "        - 104°C\n",
    "\n",
    "    - Therminator DNA Polymerase\n",
    "      - Thermococcus\n",
    "      - Grows at\n",
    "        - 2500 metres\n",
    "      - Activity lost\n",
    "        - 50% \n",
    "        - 6.7 h\n",
    "        - 95°C\n",
    "\n",
    "    - KOD DNA Polymerase\n",
    "      - Thermococcus kodakaraensis\n",
    "      - Higher fidelity (3’-5’ exonuclease)\n",
    "      - High processivity\n",
    "        - Twice of Taq\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# 3 Molecular Biology basics\n",
    "\n",
    "  - Cloning techniques\n",
    "\n",
    "    - Plasmids\n",
    "      - Circular\n",
    "      - extra-chromosomal\n",
    "      - Replicate independently of chromosomal DNA\n",
    "      - Plasmids Occurrence\n",
    "        - bacteria\n",
    "        - archaea\n",
    "        - some eukaryotes\n",
    "      - Carry antibiotic resistance\n",
    "      - Use in Cloning\n",
    "        - \"vector\"\n",
    "        - Antibiotic Resistance Gene → selection\n",
    "        - MCS\n",
    "\n",
    "    - Restriction Enzymes\n",
    "      - active site\n",
    "        - Mg2+\n",
    "      - Cut DNA specifically\n",
    "        - Bgl11\n",
    "          - A↓GATCT\n",
    "        - AluI\n",
    "          - AG↓CT\n",
    "        - BamHI\n",
    "          - G↓GATCC\n",
    "        - BfaI\n",
    "          - C↓TAG\n",
    "        - DpnII\n",
    "          - ↓GATC\n",
    "        - EcoRI\n",
    "          - G↓AATTC\n",
    "        - HaeIII\n",
    "          - GG↓CC\n",
    "        - HindIII\n",
    "          - A↓AGCTT\n",
    "        - NotI\n",
    "          - GC↓GGCCGC\n",
    "        - NruI\n",
    "          - TCG↓CGA\n",
    "        - RsaI\n",
    "          - GT↓AC\n",
    "        - SacI\n",
    "          - GAGCT↓C\n",
    "        - SmaI\n",
    "          - CCC↓GGG\n",
    "        - Tru9I\n",
    "          - T↓TAA\n",
    "        - PstI\n",
    "          - CTGCA↓G\n",
    "      - sticky ends\n",
    "\n",
    "    - DNA Ligases\n",
    "      - active site\n",
    "        - Lys\n",
    "      - make new phosphate bond\n",
    "      - 2 ATP phosphate bonds hydrolysis\n",
    "\n",
    "\n",
    "  - PCR\n",
    "\n",
    "    - Copying Interest Gene\n",
    "\n",
    "      - Start/Stop define target DNA\n",
    "      - Gene copied through repeated cycles\n",
    "        - DNA Melting\n",
    "        - Primer Annealing\n",
    "        - Extension\n",
    "\n",
    "    - PCR Cloning\n",
    "      - Primers add restriction sites\n",
    "      - Sequencing ensures accuracy\n",
    "\n",
    "  - pET-based expression in E. coli\n",
    "\n",
    "    - pET System\n",
    "      - Prokaryotic Expression\n",
    "      - Control Mechanism\n",
    "        - Genes under T7 promoter\n",
    "        - T7 RNA polymerase initiates expression\n",
    "\n",
    "      - T7 RNA Polymerase\n",
    "        - provided by prophage\n",
    "          - λDE3 lysogen\n",
    "        - LacI-lacO blockade\n",
    "        - IPTG binds lacI, removes inhibition\n",
    "        - IPTG\n",
    "\n",
    "  - Tagging\n",
    "\n",
    "    - Tags\n",
    "      - Peptides added (fused) \n",
    "        - N-terminal\n",
    "        - C-terminal\n",
    "      - aid, at genetic level \n",
    "        - isolation\n",
    "        - purification\n",
    "        - solubility\n",
    "\n",
    "    - Tags can impact\n",
    "      - protein folding\n",
    "      - biological activity\n",
    "      - crystallization\n",
    "      - use with care\n",
    "\n",
    "    - His\n",
    "\n",
    "      - IMAC\n",
    "        - Immobilised Metal Affinity Chromatography\n",
    "        - Metals used in IMAC\n",
    "          - Ni2+\n",
    "          - Co2+\n",
    "          - Cu2+\n",
    "          - Zn2+\n",
    "\n",
    "      - Hexahistidine\n",
    "        - His₆\n",
    "\n",
    "    - MBP\n",
    "\n",
    "      - pMAL fusion system\n",
    "        - Protein Maltose-Binding Protein Fusion\n",
    "      - Gene inserted downstrem of malE\n",
    "      - Fusion with MBP\n",
    "      - MBP, protease cleavage \n",
    "        - site between\n",
    "          - malE and \n",
    "          - inserted gene\n",
    "        - e.g. protease\n",
    "          - factor Xa\n",
    "          - enerokinase\n",
    "          - genenase I\n",
    "      - solubility ↑\n",
    "        - insoluble proteins\n",
    "\n",
    "      - Paramyosin 肌肉蛋白\n",
    "\n",
    "# 4 Protein Purification\n",
    "\n",
    "  - Protein chromatography\n",
    "\n",
    "    - Different partitioning of proteins between \n",
    "      - stationary phase \n",
    "        - adsorbent\n",
    "      - mobile phase\n",
    "        - Buffer solution\n",
    "        - Composition change\n",
    "          - control protein binding/elution\n",
    "\n",
    "    - Specificity\n",
    "      - altered by adsorbent\n",
    "      - efficiency\n",
    "      - selectivity\n",
    "\n",
    "    - Adsorbent\n",
    "      - 90% solvent\n",
    "      - 10% porous matrix\n",
    "    \n",
    "    - Matrix\n",
    "      - hydrophilic\n",
    "      - chemically & physically stable\n",
    "      - no functional groups → indiscriminate protein binding\n",
    "      - controllable porosity, rigid to allow good flow rates\n",
    "\n",
    "    - Protein separation\n",
    "      - size\n",
    "      - shape\n",
    "      - net charge\n",
    "      - hydrophobicity\n",
    "        - HIC\n",
    "      - thiol function\n",
    "        - Covalent Chromatography\n",
    "      - substrate binding\n",
    "        - Biospecific Affinity Chromatography\n",
    "\n",
    "  - Size Exclusion Chromatography\n",
    "\n",
    "    - Gel filtration\n",
    "\n",
    "    - Protein MW determine by\n",
    "      - shape\n",
    "        - globular (round) ＞ rod-like\n",
    "\n",
    "      - size\n",
    "        - Small ＞ Large\n",
    "\n",
    "  - Ion Exchange Chromatography\n",
    "\n",
    "    - Most widely used\n",
    "\n",
    "    - Ion Exchange principle\n",
    "\n",
    "      - competition\n",
    "        - charged protein\n",
    "        - competing ion\n",
    "      - oppositely charged column\n",
    "\n",
    "      - bind to column\n",
    "        - concentration of competing ions ＜ protein\n",
    "\n",
    "      - elute from column\n",
    "        - ionic strength increased\n",
    "        - competing ions remove the protein\n",
    "\n",
    "    - Q Sepharose\n",
    "      - Quaternary Ammonium\n",
    "      - Anion exchanger\n",
    "      - binds –vely charged protein\n",
    "\n",
    "    - S Sepharose\n",
    "      - Sulfonic ions\n",
    "      - Cation exchanger\n",
    "      - binds +vely charged protein\n",
    "\n",
    "    - protein at pI\n",
    "      - fully ionised\n",
    "      - no net charge\n",
    "\n",
    "    - buffer pH\n",
    "      - affect protein charge\n",
    "      - make protein bind to Q/S column\n",
    "\n",
    "  - HIC\n",
    "\n",
    "    - Hydrophobic Interaction Chromatography\n",
    "\n",
    "    - Hydrophobic molecules\n",
    "      - self-associate in aq.\n",
    "\n",
    "    - Hydrophobic residues\n",
    "      - hidden inside protein\n",
    "      - some exposed on surface\n",
    "        - bind to adsorbent hydrophobic ligand\n",
    "\n",
    "    - Adsorption to Hydrophobic ligand\n",
    "      - favoured at high ionic strength\n",
    "        - Anti-chaotropic salts\n",
    "          - SO42-\n",
    "          - Cl-\n",
    "          - enhance hydrophobic interaction\n",
    "\n",
    "    - Elution achieved by\n",
    "      - ↓ salt conc.\n",
    "      - ↓ ionic strength\n",
    "\n",
    "    - e.g.\n",
    "      - Butyl Sepharose 4 Fast Flow\n",
    "      - Octyl Sepharose CL-4B\n",
    "        - Cross-Linked\n",
    "      - Phenyl Superose/Sepharose\n",
    "        - Phenyl Sepharose High Performance\n",
    "        - Phenyl Sepharose CL-4B\n",
    "        - Phenyl Sepharose 6 Fast Flow (low/high sub)\n",
    "      - Alkyl Superose\n",
    "\n",
    "  - IMAC\n",
    "\n",
    "    - Immobilised Metal Ion Affinity Chromatography\n",
    "    - proteins-metal complexes\n",
    "\n",
    "    - Adsorbent \n",
    "      - metal chelator\n",
    "        - Cu2+\n",
    "      - linkage group\n",
    "        - iminodiacetic acid\n",
    "      - suitable matrix\n",
    "        - agarose matrix\n",
    "\n",
    "    - Binding protein by\n",
    "      - His\n",
    "      - Cys\n",
    "      - Try\n",
    "\n",
    "    - Elution achieved by\n",
    "      - histidine/imidazole\n",
    "      - descending pH gradient of histidine/imidazole\n",
    "      - compete with protein\n",
    "\n",
    "  - Biospecific Affinity Chromatography\n",
    "\n",
    "    - Protein-protein interactions\n",
    "      - immobilised transferrin receptor\n",
    "      - Specific antibodies\n",
    "\n",
    "    - Protein-DNA interactions\n",
    "      - Heparin columns\n",
    "        - Polysaccharide \n",
    "        - Containing sulfate groups\n",
    "        - mimicking DNA phosphate groups\n",
    "\n",
    "    - Biomimetic dyes \n",
    "\n",
    "      - Cibacron Blue F3G-A\n",
    "        - Blue Sepharose\n",
    "        - analogue of adenylyl-containing cofactors\n",
    "        - purification\n",
    "          - NAD+ binding enzymes\n",
    "          - ATP binding enzymes\n",
    "\n",
    "      - Procion Red HE-3B\n",
    "        - NAD+ binding enzymes\n",
    "        - NADP+ binding enzymes\n",
    "\n",
    "    - Trypsin-like Ser proteases purification\n",
    "      - Benzamidine\n",
    "      - Arginine\n",
    "      - Lysine\n",
    "\n",
    "    - Purification tau\n",
    "      - Over-expression\n",
    "      - Blue Sepharose\n",
    "      - Mono Q\n",
    "      - Gel Filtration\n",
    "\n",
    "  - Inclusion Bodies\n",
    "\n",
    "    - Insoluble proteins\n",
    "      - produce in E. coli\n",
    "      - form inclusion bodies\n",
    "      \n",
    "    - Inclusion bodies\n",
    "      - membrane bound vacuoles inside bacteria\n",
    "      - isolated once bacteria are lysed\n",
    "      - difficult to disrupt\n",
    "      - harsh conditions\n",
    "        - e.g.\n",
    "          - urea\n",
    "          - detergents\n",
    "          - guanidinium hydrochloride\n",
    "        - cause protein denature\n",
    "        - need refold\n",
    "\n",
    "    - Refolding process\n",
    "      - gradual remove denaturants\n",
    "      - allowing protein fold gradually\n",
    "      - via several intermediate states\n",
    "\n",
    "# 5 Protein Mutagenesis\n",
    "\n",
    "  - Chemical Mutagenesis\n",
    "\n",
    "    - GC → AT\n",
    "\n",
    "    - Hydroxylamine\n",
    "      - Converts C \n",
    "      - To N4-hydroxycytosine\n",
    "\n",
    "    - N-methyl-N' nitrosoguanidine\n",
    "      - Converts G \n",
    "      - To O6-methyl guanine\n",
    "\n",
    "  - Mutator Bacterial Strains\n",
    "\n",
    "    - mutD loss\n",
    "      - DNA pol. III\n",
    "    - mutT loss\n",
    "      - prevents A:8-oxidoG\n",
    "      - by hydrolysing 8-oxidoG\n",
    "    - mutY loss\n",
    "      - glycosylase\n",
    "      - correct G:A\n",
    "      - correct A:8-oxidoG\n",
    "\n",
    "  - PCR-Mutagenesis\n",
    "    - Taq polymerase\n",
    "    - Lacks 3’-5’ proofreading activity\n",
    "    - Using Mn2+ ↑ error to 30%\n",
    "    - Problems\n",
    "      - Multiple mutations\n",
    "      - Silent mutations\n",
    "      - Non-sense mutations 提前终止\n",
    "      - No control nature of mutation\n",
    "      - Mutations outside cloned gene\n",
    "\n",
    "  - Site-directed mutagenesis\n",
    "\n",
    "    - M13\n",
    "\n",
    "      - Bacteriophage M13\n",
    "\n",
    "        - ssDNA virus\n",
    "        - Replication\n",
    "          - ssDNA → ds RF\n",
    "          - in E. coli\n",
    "        - Recovery\n",
    "          - RF M13 DNA\n",
    "            - recovered from E. coli\n",
    "          - ssM13\n",
    "            - recovered from virus\n",
    "        - RF DNA → MCS\n",
    "\n",
    "      - Phagemids 噬菌粒\n",
    "\n",
    "        - maintain as plasmids in E. coli\n",
    "        \n",
    "        - Plasmids\n",
    "          - replication origin\n",
    "          - gene expression systems\n",
    "\n",
    "        - Phagemids Composition\n",
    "          - replication origin\n",
    "          - packaging signal\n",
    "        \n",
    "        - Infection by helper phage\n",
    "          - Phage replication Activate\n",
    "          - Phagemid ssDNA → phage particles\n",
    "          - Recognizes RF plasmid → ssDNA\n",
    "\n",
    "    - Overlapping PCR\n",
    "\n",
    "    - DpnI digestion\n",
    "\n",
    "# 7 Phage\n",
    "\n",
    "  - Ff Structure\n",
    "\n",
    "    - ss circular DNA\n",
    "\n",
    "    - 11 genes grouped\n",
    "      - DNA replication\n",
    "        - II\n",
    "        - V\n",
    "        - X\n",
    "      - capsid\n",
    "        - VII\n",
    "        - IX\n",
    "        - VIII\n",
    "        - III\n",
    "        - VI\n",
    "      - membrane associated assembly\n",
    "        - I\n",
    "        - XI\n",
    "        - IV\n",
    "\n",
    "    - Intergenic (IG) region \n",
    "      - encode no proteins\n",
    "      - origin sites for +/- strands synthesis\n",
    "      - packaging signal\n",
    "        - phage particles assembly initiation\n",
    "        - hairpin region\n",
    "\n",
    "  - Ff Life cycle\n",
    "\n",
    "    - Infection\n",
    "    - Conversion to Replicative Form\n",
    "    - Phage Genome Replication\n",
    "    - Phage Protein Expression\n",
    "    - Assembly\n",
    "    - Release\n",
    "\n",
    "  - principle of phage display\n",
    "\n",
    "    - Fuse protein/peptide to capsid protein\n",
    "    - Display capsid protein on surface\n",
    "    - Build libraries\n",
    "      - per phage:\n",
    "      - unique random peptide/protein\n",
    "    - Screen using appropriate ligands\n",
    "\n",
    "  - Natural peptides\n",
    "    - Mapping epitopes\n",
    "      - monoclonal antibodies\n",
    "      - polyclonal antibodies\n",
    "    - Generating immunogens\n",
    "\n",
    "  - Random peptides\n",
    "    - Mapping epitopes\n",
    "      - monoclonal antibodies\n",
    "      - polyclonal antibodies\n",
    "    - Identifying peptide ligands\n",
    "    - Mapping substrate sites\n",
    "      - proteases \n",
    "      - kinases\n",
    "\n",
    "  - Protein and protein domains\n",
    "    - Directed proteins evolution\n",
    "    - High affinity antibodies isolation\n",
    "    - cDNA expression screening\n",
    "\n",
    "  - Display site\n",
    "    \n",
    "    - site gene\n",
    "      - III\n",
    "      - VIII\n",
    "    - fusions with capsid proteins \n",
    "      - pIII\n",
    "      - pVIII\n",
    "    - Not all sequences displayed well\n",
    "\n",
    "  - Phage Common problems\n",
    "\n",
    "    - particle assembly\n",
    "    - stability\n",
    "    - infectivity\n",
    "\n",
    "  - Different vector systems\n",
    "\n",
    "    - Type 3 or 8\n",
    "      - either gIII or gVIII\n",
    "      - both gene\n",
    "\n",
    "    - Type 3+3 or 8+8\n",
    "      - either gIII or gVIII\n",
    "      - only one gene\n",
    "\n",
    "    - Type 33 or 88\n",
    "      - either gIII or gVIII\n",
    "      - extra copy: wt gIII or gVIII\n",
    "\n",
    "    - Valency\n",
    "      - pVIII-fusion ＞ pIII-fusion\n",
    "\n",
    "      - Valency modifications advantages\n",
    "        - low affinity interactions\n",
    "        - display ‘difficult’ proteins\n",
    "\n",
    "  - Cysteine problem\n",
    "\n",
    "  - pIII \n",
    "    - 8 cysteines paired\n",
    "      - C7-C36\n",
    "      - C46-C53\n",
    "      - C188-C201\n",
    "      - C354-C371\n",
    "\n",
    "    - Unpaired cysteines problems\n",
    "      - odd-numbered Cys affect pIII systems\n",
    "      - even-numbered Cys perform better\n",
    "      - favors short inter-C sequences peptides\n",
    "\n",
    "  - pVIII has no cysteines\n",
    "\n",
    "# 8 MS\n",
    "\n",
    "  - MALDI\n",
    "\n",
    "    - Matrix Assisted Laser Desorption Ionisation\n",
    "\n",
    "    - Role of matrix\n",
    "      - ionise sample embedded in crystalline matrix\n",
    "      - absorb energy from laser UV\n",
    "      - causing sample desorption in gas phase\n",
    "      - lead to H+ exchange / ionisation\n",
    "\n",
    "    - MALDI ionisation\n",
    "      - proteins\n",
    "      - peptides\n",
    "      - sugars\n",
    "      - oligonucleotides\n",
    "      - synthetic polymers\n",
    "\n",
    "    - Singly charged ions\n",
    "      - large proteins e.g.\n",
    "        - bovine serum albumin (BSA)\n",
    "        - MW 66.5 kDa\n",
    "\n",
    "  - ESI\n",
    "\n",
    "    - Electrospray Ionisation\n",
    "    - Proteins with multiple charge\n",
    "\n",
    "\n",
    "  - Protein Digestion\n",
    "    - Trypsin\n",
    "      - Commonly used\n",
    "      - C-terminal side\n",
    "      - Lys\n",
    "      - Arg\n",
    "\n",
    "    - e.g. LDFPQRILLWKAGGT\n",
    "      - LDFPQR\n",
    "      - ILLWK\n",
    "      - AGGT\n",
    "\n",
    "  - LC-MS/MS\n",
    "    - Sequencing\n",
    "      - MS (Q-TOF)\n",
    "        - Quadrupole-Time Of Flight\n",
    "        - Q1\n",
    "          - select desired m/z peptide ion\n",
    "        - collision cell q2\n",
    "          - Collision induced dissociation (CID)\n",
    "            - peptide ions\n",
    "              - 2+\n",
    "              - 3+\n",
    "            \n",
    "            - N-terminal\n",
    "              - a\n",
    "              - b\n",
    "              - c\n",
    "\n",
    "            - C-terminal\n",
    "              - x\n",
    "              - y\n",
    "              - z\n",
    "\n",
    "        - TOF mass analyser\n",
    "\n",
    "    - modification identification\n",
    "      - Phosphorylation\n",
    "      - Glycosylation\n",
    "      - Oxidation\n",
    "\n",
    "  - Fragments\n",
    "    - b, y ions dominate\n",
    "    - Given a reasonable series of b / y ions\n",
    "      - Peptide sequenced partially\n",
    "    - Impossible to distinguish\n",
    "      - Leu\n",
    "      - Ile\n",
    "    - Require high resolution\n",
    "      - Lys\n",
    "      - Gln\n",
    "\n",
    "  - Identify protein\n",
    "    - 6 or 7 contiguous aa\n",
    "    - 18-21 DNA bases\n",
    "\n",
    "# 9 AUC\n",
    "\n",
    "  - Analytical Ultracentrifugation\n",
    "\n",
    "  - high-speed centrifuge → biomolecules sedimentation\n",
    "\n",
    "  - Optical detection systems\n",
    "    - monitored sedimentation\n",
    "      - protein solute absorbs UV\n",
    "      - solvent doesn’t\n",
    "    - specific, protein absorbs at particular wavelengths e.g. \n",
    "      - aromatic aa\n",
    "      - bound metal\n",
    "      - cofactor e.g.\n",
    "        - NAD+/H\n",
    "        - FAD/H2\n",
    "\n",
    "    - Interference\n",
    "      - Relative refractive index between\n",
    "        - sample\n",
    "        - reference solution\n",
    "      - Universal detection\n",
    "        - × need absorbing protein\n",
    "      - not specific\n",
    "\n",
    "    - Fluorescence\n",
    "      - sensitive\n",
    "      - specific\n",
    "        - defined excitation\n",
    "        - emission wavelengths\n",
    "\n",
    "  - Svedberg equation\n",
    "    - s =\n",
    "      - DM\n",
    "      - (1-vρ)\n",
    "      - 1/RT\n",
    "\n",
    "  - Non-equilibrium Sedimentation\n",
    "    - very fast\n",
    "      - 60,000 rpm\n",
    "    - no back-diffusion\n",
    "    - 3 forces\n",
    "      - centrifugal force\n",
    "        - spin of centrifuge\n",
    "      - buoyancy force\n",
    "        - solute displacing solvent molecules\n",
    "      - frictional force\n",
    "        - restriction of solute passing through solvent \n",
    "    - s determination\n",
    "      - diffusion coefficient D\n",
    "        - shape\n",
    "      - molecular weight M\n",
    "        - size\n",
    "    - e.g.\n",
    "      - two non-interacting proteins\n",
    "        - 2 distinct sedimentation events\n",
    "        - 2 s determined\n",
    "      - proteins interact\n",
    "        - boundary due to complex\n",
    "\n",
    "\n",
    "  - equilibrium Sedimentation\n",
    "    - slower\n",
    "      - 10,000 rpm\n",
    "    - balanced by diffusion back up the cell\n",
    "    - take many hours\n",
    "    - molecular weight M\n",
    "    - replaced by MS\n",
    "    - show protein self-association\n",
    "    - e.g. protein in monomer-dimer equ.\n",
    "      - low conc. (absorbance)\n",
    "        - monomeric\n",
    "      - high conc. (absorbance)\n",
    "        - dimeric\n",
    "\n",
    "\n",
    "# 10 SPR\n",
    "\n",
    "  - Physical Principle\n",
    "\n",
    "    - total internal reflection\n",
    "      - Light passing through a prism\n",
    "\n",
    "    - Surface Plasmon Resonance\n",
    "\n",
    "      - Surface Plasmon\n",
    "        - e- oscillations\n",
    "        - prism upper surface\n",
    "          - coated with thin modified gold\n",
    "            - Lys\n",
    "            - Cys\n",
    "\n",
    "      - Resonance\n",
    "        - incident light\n",
    "          - between light and surface\n",
    "        - plasmons achieve\n",
    "          - reflected light intensity → minimum\n",
    "\n",
    "  - Application\n",
    "\n",
    "    - immobilising protein +\n",
    "      - e.g. antibodies\n",
    "    - gold\n",
    "    - passing interacting molecules\n",
    "    - θ_SPR changes\n",
    "      - sensitive to refractive index\n",
    "      - binding can be detected and studied\n",
    "    - done in flow cell\n",
    "    - Pure analyte samples are required\n",
    "\n",
    "  - Sensorgram\n",
    "\n",
    "    - analyte injection\n",
    "      - association: analyte and immobilised protein\n",
    "\n",
    "    - pure buffer injection\n",
    "      - induces dissociation of complex\n",
    "\n",
    "    - regeneration buffer injection\n",
    "      - strip the complex\n",
    "      - residual bound analyte removal\n",
    "\n",
    "    - normal buffer\n",
    "      - resets system for another cycle\n",
    "\n",
    "  - Data\n",
    "    - Kinetics\n",
    "      - association rate\n",
    "          - P = k1[A][P]\n",
    "      - dissociation rate\n",
    "          - P = k-1[AP]\n",
    "    - Thermodynamics\n",
    "      - dissociation constant\n",
    "        - [A][P]/[AP] = k-1 / k1\n",
    "    - Accurate range\n",
    "      - nM\n",
    "      - μM\n",
    "\n",
    "# 11 Fluorescence, FRET\n",
    "\n",
    "  - Fluorescence\n",
    "    - Stokes shift\n",
    "      - Change in wavelength\n",
    "    - Quantum yield\n",
    "      - Fraction\n",
    "      - emit fluorescence\n",
    "      - excited molecules\n",
    "    - Photobleaching\n",
    "      - photochemical destruction of fluorophore\n",
    "      - cause by Extended exposure to intense excitation light\n",
    "\n",
    "    - Fluorophore\n",
    "      - Polarised chromophore\n",
    "      - Intrinsic fluorophores \n",
    "        - aromatic aa\n",
    "          - tryptophan (indole ring)\n",
    "        - flavin\n",
    "        - NADH cofactors\n",
    "      - Extrinsic fluorophores\n",
    "        - attached or bound to protein\n",
    "\n",
    "      - efficiency\n",
    "        - dependent on environment\n",
    "          - λ (emission)\n",
    "          - quantum yield (signal)\n",
    "\n",
    "\n",
    "    - Monitor protein folding\n",
    "      - different fluorescence\n",
    "        - surface tryptophan residue exposed to water\n",
    "        - buried inside the protein core\n",
    "      - no tryptophan\n",
    "        - extrinsic fluorophore introduced chemically\n",
    "\n",
    "fluorophore + protein non-covalently\n",
    "λ(emission), quantum yield (signal) change\n",
    "\n",
    "bovine serum albumen (BSA)\n",
    "\n",
    "BSA binds lipids in transport through blood\n",
    "\n",
    "extrinsic fluorescent compound ANS + BSA\n",
    "fluorescence change\n",
    "moves from the \n",
    "polar environment\n",
    "water to the \n",
    "non-polar environment\n",
    "protein \n",
    "\n",
    "  - Anisotropy \n",
    "    - fluorescence difference in directions\n",
    "    - detect biomolecular interactions\n",
    "    - uses plane polarised light\n",
    "\n",
    "    - large molecules\n",
    "      - tumble slowly\n",
    "      - emit highly polarised light\n",
    "    \n",
    "    - small molecules \n",
    "      - tumble rapidly\n",
    "      - emit depolarised light\n",
    "      - reduced anisotropy\n",
    "\n",
    "    - e.g.\n",
    "      - [DNA_fluor+CAP+RNA pol] complex\n",
    "        - fluorescent-labelled DNA\n",
    "        - RNA pol enzyme\n",
    "        - small protein ‘CAP’\n",
    "      - RNA pol conc. ↑\n",
    "      - slower tumbling\n",
    "      - anisotropy ↑\n",
    "\n",
    "  - FRET\n",
    "\n",
    "    - Förster Resonance Energy Transfer\n",
    "      - Two different fluorophores\n",
    "      - Donor selectively irradiated\n",
    "      - Acceptor quenched fluorescence\n",
    "        - overlaps\n",
    "          - donor emission spectrum\n",
    "          - acceptor excitation spectrum\n",
    "        - donor / acceptor close in space\n",
    "\n",
    "    - Förster distance\n",
    "      - distance at 50% FRET efficiency\n",
    "\n",
    "    - Measuring spatial proximity\n",
    "      - relative motion, conformational flexibility of a protein\n",
    "      - dissociation of 2 linked fluorescent protein\n",
    "\n",
    "# 12 NMR\n",
    "\n",
    "  - 2D\n",
    "    - signal correlated with each other\n",
    "    - coupled nuclei\n",
    "      - homonuclear\n",
    "      - heteronuclear\n",
    "    - coupling\n",
    "      - bonds (scalar, J)\n",
    "      - space (dipolar, d)\n",
    "\n",
    "  - COSY\n",
    "    - Correlated Spectroscopy\n",
    "    - 1H-1H\n",
    "    - homonuclear\n",
    "    - through bond\n",
    "\n",
    "  - NOESY\n",
    "    - Nuclear Overhauser Effect Spectroscopy\n",
    "    - 1H-1H\n",
    "    - homonuclear\n",
    "    - through space\n",
    "    - nuclear Overhauser effect (nOe)\n",
    "      - between protons held rigidly in close proximity due to secondary structure\n",
    "    - Patterns in 1H1H NOESY between contiguous AAs can reveal the presence of an alpha helix or beta sheet \n",
    "\n",
    "    - Heteronuclear\n",
    "      - important for larger proteins (>100 aas)\n",
    "      - line broadening in 1H-1H experiment\n",
    "      - Coupling between 13C, 15N, 1H\n",
    "      - require 13C-, 15N-labelled protein\n",
    "\n",
    "  - 3D\n",
    "    - The process of assigning signals in\n",
    "    2D NMR to specific AAs, then\n",
    "    determining their spatial relationships\n",
    "    is a complex process (even with\n",
    "    knowledge of the sequence)\n",
    "    - It requires considerable skill and time,\n",
    "    but can result in a 3D structure\n",
    "    - The process is performed in\n",
    "    combination with molecular modelling\n",
    "\n",
    "  - slow tumbling\n",
    "\n",
    "    - small fluorescent ligand + \n",
    "    - large protein → \n",
    "    - tumbling rate ↓\n",
    "\n",
    "    - Tumbling increase in horizontally polarised signal\n",
    "\n",
    "  - CSP\n",
    "    - chemical shift perturbation\n",
    "    - By titrating in a ‘binder’\n",
    "      - small molecule ligand\n",
    "      - another protein\n",
    "\n",
    "    - map binding sites\n",
    "      - aa CSP peak\n",
    "        - where the binder interacts\n",
    "\n",
    "  - native ESI-MS\n",
    "\n",
    "    - study protein complexes\n",
    "    - volatile buffer\n",
    "      - ammonium acetate\n",
    "    - give lower charge states\n",
    "      - protein is still folded\n",
    "      - surface area to accept protons\n",
    "\n",
    "# 13 Evolution\n",
    "\n",
    "  - Divergent evolution\n",
    "    - 趋异进化\n",
    "    - Darwin’s finches\n",
    "    - ABO blood groups\n",
    "      - A and B alleles\n",
    "        - differ at 4 aa\n",
    "      - O gene \n",
    "        - premature stop\n",
    "        - too short to work\n",
    "\n",
    "  - Convergent evolution\n",
    "    - 趋同进化\n",
    "    - Top marine predators\n",
    "      - Similar body plan\n",
    "      - But came from Different land creatures\n",
    "      - Functionally convergent\n",
    "      - Doesn’t look similar\n",
    "    - Enzymes work same reaction\n",
    "      - e.g. from fungus\n",
    "        - Cellobiohydrolase\n",
    "        - Exoglucanase 1\n",
    "      - differ in fold / mechanism\n",
    "      - same result\n",
    "\n",
    "# 16 Protein structures\n",
    "\n",
    "  - Peptide\n",
    "    - ribosome\n",
    "      - 4.5 MDa\n",
    "      - 4 different RNA molecules\n",
    "      - 82 proteins\n",
    "\n",
    "    - tRNAs\n",
    "      - 25 kDa\n",
    "      - 20 aminoacyl-tRNA synthetases\n",
    "      - All to add ~110 Da aa\n",
    "\n",
    "    - Ramachandran plot\n",
    "      - Real data from protein structures\n",
    "      - Only some conformations occur\n",
    "        - α-helix\n",
    "        - β-sheet\n",
    "      - More flexible\n",
    "        - Glycine\n",
    "      - Less flexible\n",
    "        - Proline\n",
    "\n",
    "  - Secondary structure\n",
    "\n",
    "    - α-helix\n",
    "      - stabilized by H bond\n",
    "      - 3.6 aa / turn\n",
    "      - 2 side different\n",
    "        - Non-polar\n",
    "        - Polar\n",
    "\n",
    "    - β-sheet\n",
    "      - parallel\n",
    "      - anti-parallel\n",
    "      - Linker\n",
    "        - loops\n",
    "        - turns (random coils)\n",
    "\n",
    "  - Tertiary structure\n",
    "\n",
    "    - salt bridges\n",
    "      - charged side chains\n",
    "      - Ionic interactions\n",
    "\n",
    "    - disulfide bridges\n",
    "      - Cys\n",
    "\n",
    "    - secondary structure → Motif e.g.\n",
    "      - β-α-β loop\n",
    "\n",
    "    - Motif → domain e.g.\n",
    "      - α-β barrel\n",
    "\n",
    "    - domain → Protein\n",
    "      - domain have specific function\n",
    "        - binding to substrate\n",
    "        - receptor\n",
    "\n",
    "  - Quaternary structure\n",
    "\n",
    "    - single peptide chain\n",
    "      - monomeric\n",
    "\n",
    "    - multiple polypeptides\n",
    "      - identical polypeptides\n",
    "        - from same gene\n",
    "        - homo- prefix\n",
    "      - different polypeptides\n",
    "        - from different genes\n",
    "        - hetero- prefix\n",
    "      - Dimers\n",
    "      - Tetramers\n",
    "      - Octamers\n",
    "\n",
    "    - bonding interactions \n",
    "      - H-bonding\n",
    "      - salt bridges\n",
    "      - disulfide bridges\n",
    "\n",
    "    - Subunit\n",
    "      - Each polypeptide unit\n",
    "\n",
    "  - Proteins in motion\n",
    "    - change shape \n",
    "      - enzyme binding substrate\n",
    "    - more distant \n",
    "      - inhibitor binding\n",
    "    - couple to other reactions \n",
    "      - ATP hydrolysis providing energy\n",
    "    - Allosteric\n",
    "      - functional units assemblies\n",
    "      - quaternary structural transitions\n",
    "        - response to conc. ligand changes\n",
    "\n",
    "Hemoglobin (Hb)\n",
    "response to allosteric effectors\n",
    "cooperative binding\n",
    "one subunit binding affects other\n",
    "hemoglobin sigmoidal curve\n",
    "\n",
    "Myoglobin\n",
    "\n",
    "# 17 Cofactors\n",
    "\n",
    "  - Prosthetic groups\n",
    "    - Aid structural stability\n",
    "    - Porphyrin\n",
    "\n",
    "  - Apoenzyme\n",
    "    - enzyme without cofactor\n",
    "\n",
    "  - Coenzymes\n",
    "\n",
    "    - Functionally complement with aa\n",
    "\n",
    "    - NAD\n",
    "      - Nicotinamide adenine dinucleotide\n",
    "      - Aid redox reactions\n",
    "        - NAD+ accepts\n",
    "        - H- from -OH\n",
    "      - Bind transiently\n",
    "\n",
    "    - TPP\n",
    "      - Thiamine pyrophosphate\n",
    "      - Vitamin B1 derivative\n",
    "      - Involved in forming / breaking C-C bonds\n",
    "      - e.g.\n",
    "        - Pyruvate decarboxylase\n",
    "\n",
    "    - Cobalamin\n",
    "      - Vitamin B12\n",
    "      - Co cycles (I, II, III)\n",
    "      - R group\n",
    "        - CN\n",
    "        - HO\n",
    "        - Me\n",
    "      - e.g.\n",
    "        - Methionine synthase\n",
    "\n",
    "    - Biotin\n",
    "      - Vitamin B7\n",
    "      - Carboxylate Transfer\n",
    "      - e.g.\n",
    "        - Pyruvate carboxylase\n",
    "\n",
    "    - PLP\n",
    "      - Pyridoxal phosphate\n",
    "      - Vitamin B6\n",
    "      - Forms Schiff base\n",
    "      - Function\n",
    "        - amino transferases\n",
    "        - eliminations\n",
    "        - isomerisations\n",
    "\n",
    "# 18 Kinetics\n",
    "\n",
    "  - Assumption\n",
    "    - Substrate excess\n",
    "    - Rapid equilibrium\n",
    "    - Enzyme unchanged\n",
    "      - Free E\n",
    "      - ES\n",
    "\n",
    "  - Assume [ES] steady state\n",
    "    - Formation\n",
    "      - ES =k1[E][S]\n",
    "    - Breakdown\n",
    "      - ES = k-1[ES]+k2[ES]\n",
    "    - Formation = Breakdown\n",
    "\n",
    "  - v = \n",
    "    - (Vmax * [S])\n",
    "    - / (Km + [S])\n",
    "\n",
    "  - Vmax\n",
    "    - all enzyme involved\n",
    "    - limited by k2\n",
    "\n",
    "  - Km, Michaelis constant\n",
    "\n",
    "# 25 Enzymology\n",
    "\n",
    "  - Biotechnology\n",
    "\n",
    "  - Role\n",
    "\n",
    "    - Metals\n",
    "\n",
    "      - Metals in biological systems\n",
    "        - Metalloproteins\n",
    "          - Electron transport\n",
    "          - Oxygen carriers\n",
    "          - Structural role\n",
    "          - Metalloenzymes\n",
    "            - Hydrolytic enzymes\n",
    "            - Two electrons redox enzymes\n",
    "            - Multi electron pair redox enzymes\n",
    "            - Rearrangements\n",
    "        - Structural or communication role\n",
    "\n",
    "      - Group 1/2 metals\n",
    "        - Group 1/2 metals: structural role in\n",
    "          - body (e.g. bones)\n",
    "          - protein folding\n",
    "        - Group 1/2 metals: Aid in\n",
    "          - charge maintenance\n",
    "          - osmotic balance\n",
    "          - nerve impulses\n",
    "      - Transition metal ions\n",
    "        - structural role\n",
    "        - electron carries\n",
    "        - oxygen transporters\n",
    "        - catalysts in redox/non-redox\n",
    "\n",
    "    - aa\n",
    "\n",
    "      - Proteins bind to metal\n",
    "      - Prosthetic groups bind to metals\n",
    "      - According to the HSAB\n",
    "\n",
    "  - Serine proteases\n",
    "\n",
    "    - Triad\n",
    "      - Asp\n",
    "        - align His\n",
    "        - act as Acid\n",
    "      - His\n",
    "        - activate Ser\n",
    "        - act as Base\n",
    "      - Ser\n",
    "        - act as Nucleophile\n",
    "\n",
    "    - Two steps\n",
    "      - Formation of acyl-enzyme + Ser\n",
    "      - Hydrolysis of acyl-enzyme\n",
    "\n",
    "    - No substrate specificity\n",
    "      - main chain binds enzyme loop\n",
    "      - Backbone H-bonds form\n",
    "      - Interaction resembles\n",
    "        - short anti-parallel β-sheet\n",
    "\n",
    "    - selectivity controlled by pocket\n",
    "\n",
    "      - Trypsin\n",
    "        - Aspartate\n",
    "        - Negative charge\n",
    "      - Chemotrypsin\n",
    "        - Serine\n",
    "        - Small\n",
    "      - Elastase\n",
    "        - Valine\n",
    "        - Threonine\n",
    "        - Bulky\n",
    "\n",
    "\n",
    "  - Cysteine proteases (Thiol proteases)\n",
    "\n",
    "    - e.g.\n",
    "      - papain 木瓜蛋白酶\n",
    "      - ficin 无花果蛋白酶\n",
    "      - bromelain 菠萝蛋白酶\n",
    "\n",
    "    - no Asp need\n",
    "      - pKa: Cys(1.92)＜Ser(13)\n",
    "\n",
    "    - Mechanism for papain\n",
    "      - Catalytic dyad\n",
    "        - Cys\n",
    "          - 25\n",
    "          - nucleophilic attack\n",
    "        - His\n",
    "          - 159\n",
    "          - general acid/base\n",
    "      - Oxyanion hole stabilize by\n",
    "        - Cys25\n",
    "        - Gln19 \n",
    "        - H-bonding interactions\n",
    "      - Substrate binding\n",
    "        - 7 aa\n",
    "\n",
    "  - Acid proteases (Aspartic proteases)\n",
    "\n",
    "    - Asp in active site\n",
    "    - e.g.\n",
    "      - pepsin 胃蛋白酶\n",
    "      - chymosin 凝乳蛋白酶\n",
    "      - HIV-protease\n",
    "    - acid/base cat. mechanism\n",
    "\n",
    "  - Metalloprotease\n",
    "\n",
    "    - cleave peptide bond\n",
    "    - Require Zn & Co\n",
    "    - Exopeptidases\n",
    "      - Process from 2 end\n",
    "      - cleave terminal / penultimate peptide bond\n",
    "      - Releasing aa / dipeptides\n",
    "    - Endopeptidases\n",
    "      - Break from non-terminal aa\n",
    "      - Specific for certain aa\n",
    "\n",
    "  - Zn\n",
    "\n",
    "    - e.g.\n",
    "      - Carbonic Anhydrase\n",
    "        - reversible CO2 to -COOH\n",
    "      - Carboxypeptidase A\n",
    "        - Exopeptidases\n",
    "      - SuperOxide Dismutase\n",
    "        - maintain protein structure and stability\n",
    "      - β-lactamase\n",
    "        - hydrolytic enzymes\n",
    "      - alcohol dehydrogenase\n",
    "      \n",
    "    - × catalyse redox reaction\n",
    "\n",
    "    - Carboxypeptidase A\n",
    "      - Exopeptidases\n",
    "\n",
    "      - Coordination, Carboxypeptidase A\n",
    "        - 2 His\n",
    "        - a Glu\n",
    "        - a water/hydroxide\n",
    "        - Distorted Td\n",
    "      - Remove Zn\n",
    "        - low pH, chelating agent\n",
    "          - loss activity\n",
    "          - no effect on structure\n",
    "        - Re-established: Zn + apo enzyme\n",
    "      - Cleaving peptide from C-term, favoured:\n",
    "        - Phe\n",
    "        - Tyr\n",
    "        - Trp\n",
    "        - Leu\n",
    "      - Zn2+ polarises C=O\n",
    "      - Residues act as acid/base\n",
    "      - concerted mechanism\n",
    "\n",
    "    - Carbonic Anhydrase\n",
    "      \n",
    "      - Catalyse \n",
    "        - reversible CO2 to -COOH\n",
    "      - Regulate\n",
    "        - pH\n",
    "        - haemoglobin function\n",
    "      - Coordination, Carbonic Anhydrase\n",
    "        - 3 His\n",
    "        - 1 HO-\n",
    "        - Distorted Td\n",
    "        - 1 nearby Thr/His aided formation of hydroxide\n",
    "      - One of fastest enzymes\n",
    "        - diffusion limited\n",
    "      - Zn2+ + H2O → HO- attack CO2\n",
    "\n",
    "  - Fe\n",
    "\n",
    "    - Most abundant transition metal in human body \n",
    "      - (~4-6 g)\n",
    "\n",
    "    - Fe associated-diseases\n",
    "      - anaemia\n",
    "      - haemochromatosis\n",
    "      - siderosis\n",
    "\n",
    "    - Transport\n",
    "      - Transferrins\n",
    "        - Bind Fe(III)\n",
    "        - In pH dependent manner\n",
    "      - 2 Fe and 2 Carbonate in Distorted Oh\n",
    "\n",
    "    - Storage\n",
    "      - Ferritins\n",
    "        - Phase transition: metal ions —— solid\n",
    "      - In anaerobic\n",
    "        - Fe(II) (soluble)\n",
    "      - In aerobic\n",
    "        - Fe(III) (solubility of ~10^-18 M)\n",
    "      - In protein centre\n",
    "        - Fe(III) stored Fe₂O₃·xH₂O\n",
    "    \n",
    "    - Structural motifs\n",
    "      - haem\n",
    "        - Porphyrin\n",
    "      - non-haem\n",
    "      - Fe-S clusters\n",
    "\n",
    "    - Roles\n",
    "\n",
    "      - Oxygen transport and storage\n",
    "        - Haemoglobin 血红蛋白\n",
    "          - oxygen transport\n",
    "        - Myoglobin 肌红蛋白\n",
    "          - oxygen storage\n",
    "        - Both contain haem centre, coordinating\n",
    "          - Fe(II)\n",
    "            - deoxy\n",
    "          - Fe(III)\n",
    "            - oxy\n",
    "\n",
    "      - Electron transfer\n",
    "        - Cytochrome c\n",
    "          - Mitochondrial Cyt c transfers e- in respiratory chain\n",
    "          - Haem-Fe centre\n",
    "          - Coordination, Cyt c\n",
    "            - His\n",
    "            - Met \n",
    "            - in axial positions\n",
    "\n",
    "        - Fe-S proteins\n",
    "          - Fe in Td\n",
    "          - S2- / —S- from Cys\n",
    "            - 4Fe4S\n",
    "            - 3Fe4S\n",
    "            - 2Fe2S\n",
    "          - Fast e- transfer tuned by H-bonds around\n",
    "\n",
    "      - Catalysis\n",
    "        - Some Fe-S clusters have hydrolytic activity\n",
    "        - Haem / non-haem centres found in Fe enzymes\n",
    "\n",
    "      - Oxygen activation\n",
    "        - SuperOxide Dismutase\n",
    "        - Catalase 过氧化氢酶\n",
    "        - Lower activity → O2-derived radicals accumulation → oxidative stress\n",
    "        - Oxygen Activation Type\n",
    "          - Dioxygenases\n",
    "          - Monooxygenases\n",
    "          - Oxidases\n",
    "\n",
    "    - P450\n",
    "\n",
    "      - Two main roles\n",
    "        - Endogenous Biosynthesis/degradation\n",
    "          - specific for substrate\n",
    "            - steroids\n",
    "            - fatty acids\n",
    "          - High regio- and stereoselectivity\n",
    "        - Exogenous Oxidative Metabolism\n",
    "          - e.g. drugs\n",
    "          - Low selectivity\n",
    "\n",
    "      - Haem-based\n",
    "      - Monooxygenases\n",
    "      - 2e-/2H+ hydroxylation\n",
    "      - Breaking strong C-H bonds\n",
    "      - Triggered by substrate\n",
    "      - Fe cycles (II, III, IV)\n",
    "      - aided by non-innocent ligand porphyrin ring\n",
    "\n",
    "  - Cu\n",
    "\n",
    "    - Hemocyanin 血蓝蛋白\n",
    "      - Cu-based\n",
    "      - oxygen transport protein\n",
    "      - In invertebrates\n",
    "\n",
    "    - Type I\n",
    "      - mononuclear Cu sites\n",
    "      - intense blue colour\n",
    "      - high reduction potential\n",
    "      - Type I Cu coordinated by\n",
    "        - 2 His\n",
    "        - 1 Cys\n",
    "        - 4th ligand (e.g. Met)\n",
    "\n",
    "    - Type II\n",
    "      - mononuclear Cu sites\n",
    "      - N or N/O coordination \n",
    "      - square planar / JT distorted Oh\n",
    "      - Colour not intense\n",
    "\n",
    "    - Type III\n",
    "      - 2 Cu ← His\n",
    "      - EPR silent \n",
    "        - due to antiferromagnetic coupling\n",
    "\n",
    "    - LPMOs\n",
    "      - Lytic Polysaccharide Monooxygenases 裂解多糖单加氧酶\n",
    "      - Type II\n",
    "      - Coordination, LPMOs\n",
    "        - 3 H2O\n",
    "        - a His\n",
    "        - a Tyr\n",
    "        - “His brace”\n",
    "      - 2e-/2H+ hydroxylation\n",
    "      - Selective oxidation\n",
    "        - C1\n",
    "        - C4\n",
    "      - breaking strong C-H bonds\n",
    "      - triggered by substrate\n",
    "      - Cu cycles (I, II(III)) producing O2 intermediates similar P450\n",
    "\n",
    "    - GalOx\n",
    "      - Function\n",
    "        - -OH + O2 → -CHO\n",
    "        - Anti-bacterial H2O2 production\n",
    "        - 2e-/2H+ removed from -OH, reduce O2 to H2O2\n",
    "      - Coordination, GalOx\n",
    "        - 2 His\n",
    "        - 2 Tyr\n",
    "        - 1 H2O\n",
    "        - modified Tyr-Cys\n",
    "        - distorted square pyramidal\n",
    "      - Many substrates\n",
    "        - not just galactose\n",
    "        - aliphatic\n",
    "        - aryl alcohols\n",
    "        - sometimes with high substrate selectivity\n",
    "      - Active site\n",
    "        - modified Tyr-Cys\n",
    "        - Aid in e- transfer\n",
    "      - Cu cycles (I, II) aided by active site residues\n",
    "\n",
    "  - SODs (Superoxide Dismutase)\n",
    "\n",
    "    - ROS\n",
    "      - Produced by O2 activation\n",
    "      - 3 Anti-oxidant enzymes\n",
    "        - Superoxide Dismutase 歧化酶\n",
    "          - HO2 → H2O2 + O2\n",
    "        - Catalase 过氧化氢酶\n",
    "          - Fe-haem enzyme\n",
    "          - H2O2 → O2 + water\n",
    "        - Glutathione Peroxidase\n",
    "          - Selenoprotein\n",
    "            - Seleno-cysteine, Sec\n",
    "          - Reduce H202 → water\n",
    "          - Oxidising glutathione 2GSH → GSSG\n",
    "            - Glu\n",
    "            - Cys\n",
    "            - Gly\n",
    "\n",
    "    - Fe/MnSOD\n",
    "\n",
    "      - eukaryotes/prokaryotes\n",
    "      - homodimer/homotetramer\n",
    "      - Active site is coordinated by\n",
    "        - 3 His\n",
    "        - 1 Asp\n",
    "      - MnSOD in\n",
    "        - all mitochondria \n",
    "        - some bacteria\n",
    "\n",
    "    - NiSOD\n",
    "\n",
    "      - last class to be discovered\n",
    "      - prokaryotes\n",
    "      - homohexamer\n",
    "      - 1 Ni active site/subunit\n",
    "      - Hydrophobic core → assembly of quaternary structure\n",
    "      - Ni binding hook\n",
    "        - Binding site disordered without metal\n",
    "        - Pro causes turn\n",
    "\n",
    "    - Cu&ZnSOD\n",
    "\n",
    "      - In all eukaryotes\n",
    "      - Both Cu / Zn in active site\n",
    "      - Zn role in SOD\n",
    "        - structural role\n",
    "      - Cu role in SOD\n",
    "        - catalysis\n",
    "      - Bridged by 1 imidazolate\n",
    "\n",
    "# 26 Q1\n",
    "\n",
    "  - Protein folding\n",
    "    - Diseases\n",
    "      - Partially unstructured proteins\n",
    "        - aggregation \n",
    "      - Neurodegenerative disease\n",
    "    - Function\n",
    "    - Design\n",
    "\n",
    "    - Golf course landscape\n",
    "      - Single low energy conformation\n",
    "      - Random search of conformational space\n",
    "      - Very slow folding\n",
    "\n",
    "    - Rugged funnel landscape\n",
    "      - Reduces conformational search\n",
    "      - Hierarchical folding – structure complexity increases gradually\n",
    "      - Allows fast folding (ms to s)\n",
    "\n",
    "    - Proteins stabilized by non-bonding interactions\n",
    "      - H-bond\n",
    "      - Hydrophobic\n",
    "      - Electrostatic\n",
    "      - Van der Walls\n",
    "\n",
    "    - Factors drive protein denaturation\n",
    "      - Temperature\n",
    "      - pH\n",
    "      - Detergents\n",
    "      - High concentration of water soluble compounds (e.g. alcohols)\n",
    "      - Salts\n",
    "\n",
    "  - Protein composition\n",
    "    - 50 to 400+ L-aa\n",
    "    - PTMs\n",
    "      - Disulphide bonds\n",
    "      - Proteolytic cleavage\n",
    "      - Modification of aa\n",
    "    - Phosphorylation\n",
    "      - Most abundant PTM in Eukaryotes\n",
    "    - Glycosylation\n",
    "      - folding\n",
    "      - stability\n",
    "      - signalling\n",
    "    - Modified proteins might not refold if unfolded\n",
    "\n",
    "  - Ramachandran plot\n",
    "\n",
    "    - α-helix\n",
    "      - φ ≈ -60°\n",
    "      - ψ ≈ -45°\n",
    "      - angles close together\n",
    "    - β-sheet\n",
    "      - φ ≈ -120°\n",
    "      - ψ ≈ 120°\n",
    "      - angles farther apart\n",
    "\n",
    "  - Thermofluor assay\n",
    "    - Protein Stability\n",
    "      - Tm\n",
    "        - melting temperatur\n",
    "        - midpoint in unfolding process\n",
    "      - Cofactors, ligands can stabilise\n",
    "        - Cu on stability of LPMOs\n",
    "    - Thermofluor analysis\n",
    "      - differential scanning fluorimetry\n",
    "\n",
    "    - Protein + fluorescent dye + cuvette\n",
    "      - fluorescence quenched by water\n",
    "      - T ↑\n",
    "      - protein unfold\n",
    "      - dye bind hydrophobic residues\n",
    "      - expelling water\n",
    "      - fluorescence ↑\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "  - spectroscopic techniques anlysis\n",
    "    - secondary protein structures\n",
    "    - tertiary protein structures\n",
    "\n",
    "  - effect of denaturants\n",
    "    - stability\n",
    "    - energy barriers in two state system\n",
    "\n",
    "  - Secondary Structure Prediction\n",
    "\n",
    "    - β-sheets\n",
    "      - Sequence rich\n",
    "        - Val\n",
    "        - Thr\n",
    "        - Ile\n",
    "        - Tyr\n",
    "      - P & G \n",
    "        - STOP signal\n",
    "        - reverse turn facilitating\n",
    "          - anti-parallel β-hairpin\n",
    "\n",
    "    - α-helices\n",
    "      - Sequence rich\n",
    "        - Ala\n",
    "        - Leu\n",
    "        - Lys\n",
    "        - Glu\n",
    "      - periodicity of side chains\n",
    "        - i to i+3 / i+4\n",
    "        - favourable helix stabilising interactions\n",
    "      - Amphipathic helix\n",
    "        - one polar\n",
    "          - solvent exposed\n",
    "        - one non-polar face\n",
    "          - packing against core\n",
    "      - P & G\n",
    "        - STOP signal\n",
    "\n",
    "\n",
    "  - peptide bond characteristics\n",
    "    - resonance\n",
    "      - rigid\n",
    "      - planar\n",
    "    - planarity restricts rotations\n",
    "\n",
    "  - principle of higher order assembly of coiled coils structures\n",
    "\n",
    "    - importance of the heptad repeat\n",
    "    - Pseudo heptad repeat\n",
    "    - abcdefg:a’b’c’d’f’g’\n",
    "\n",
    "  - Parallel dimer\n",
    "    - a b c d e f g : a b c d e f g\n",
    "      - a, d\n",
    "        - Leu\n",
    "        - Val\n",
    "      - b, c, f\n",
    "        - anything\n",
    "      - g and e \n",
    "        - complementary charges\n",
    "\n",
    "  - Anti-parallel dimer\n",
    "    - a b c d e f g : a’ b’ c’ d’ e’ f’ g’\n",
    "      - a, d\n",
    "        - Leu\n",
    "        - Val\n",
    "      - b, c, f\n",
    "        - anything\n",
    "      - e = g\n",
    "      - e’ = g’\n",
    "\n",
    "  - De novo design\n",
    "\n",
    "  Cys-His-Glu triad (found in\n",
    "  natural ester hydrolases)\n",
    "  mutated into the centre of the\n",
    "  four-heptad repeat\n",
    "\n",
    "  Hydrolases p-Nitrophenyl\n",
    "  acetate to p-Nitrophenol\n",
    "\n",
    "  Synthetic coiled coils designed\n",
    "  to mimic carbonic\n",
    "  anhydrase/nitrite reductase\n",
    "\n",
    "  - non-natural aa in foldamer structures\n",
    "\n",
    "    Non-natural synthetic oligomers (foldamers) follow \n",
    "    different folding rules\n",
    "\n",
    "    Peptidomimetics\n",
    "    resistant to proteases\n",
    "\n",
    "    2.5 residues per turn\n",
    "    H-bond: C=O (i), NH (i+3) and N’H (i+4)\n",
    "    more H-bond\n",
    "\n",
    "    Self-assembly of a hexameric helix bundle\n",
    "    Helical arrangement not affected by side chains\n",
    "    Self-assembly controlled by side chains, driven by hydrophobic core formation\n",
    "    Complementary surface charges\n",
    "\n",
    "    Self-assembly of a hexameric channel/pore\n",
    "    Self-assembly controlled by side chains, driven by hydrophilic pore formation\n",
    "    Surface hydrophobic residues\n",
    "\n",
    "  - Peptide Synthesis\n",
    "\n",
    "    - Solid Synthesis\n",
    "      - Coupling\n",
    "      - Deprotection \n",
    "        - 20% piperidine/DMF\n",
    "      - Coupling\n",
    "        - DCC/HOBt\n",
    "        - DCI/Oxima\n",
    "      - Capping\n",
    "        - acetic anhydride\n",
    "      - Repeat steps\n",
    "      - Cleave from resin\n",
    "        - TFA\n",
    "\n",
    "\n",
    "    - Protecting Group\n",
    "      - Boc\n",
    "        - Lys\n",
    "\n",
    "      - tBu\n",
    "        - Ser\n",
    "        - Thr\n",
    "        - Tyr\n",
    "\n",
    "      - OtBu\n",
    "        - Asp\n",
    "        - Glu\n",
    "\n",
    "      - Trt\n",
    "        - Cys\n",
    "\n",
    "# 27 Q2\n",
    "\n",
    "- Translate between genetic code and primary sequence\n",
    "- Design DNA/RNA sequences to introduce mutations\n",
    "- Compare alignment scoring matrices to biochemistry of proteins\n",
    "\n",
    "\n",
    "- Understand how protein structure is coupled to biological function\n",
    "\n",
    "- Know the key techniques in determining the structure of proteins\n",
    "- Appreciate the advantages and limitations of different analytical techniques\n",
    "- Identify what analytical technique may be most useful in set situations\n",
    "\n",
    "- NMR spectroscopy\n",
    "\n",
    "- Cryo-EM\n",
    "  - Purify protein\n",
    "  - Plunge-freeze\n",
    "  - Transfer to TEM\n",
    "  - Image specimen\n",
    "  - Analyse & reconstruct\n",
    "\n",
    "\n",
    "- Define bioinformatics\n",
    "  - Bioinformatics is an interdisciplinary field that \n",
    "  - develops methods and software tools for understanding biological data in particular \n",
    "  - when the data sets are large and complex\n",
    "- Understand link between the key “omics”\n",
    "  - transcription\n",
    "  - translation\n",
    "  - enzymatic activity\n",
    "- Demonstrate cutting-edge applications of bioinformatics\n",
    "  - COVID genome sequencing\n",
    "  - Predicting protein folding\n",
    "\n",
    "\n",
    "- Understand library scale and use of plasmids\n",
    "- Describe the key methods of protein design\n",
    "- Compare protein design methods and identify key uses\n",
    "- Appreciate protein design in sensor applications\n",
    "\n",
    "# 28 Q3\n",
    "\n",
    "- Show the binding modes of the side-chains of amino acids to d-metals.\n",
    "- Understand the connection between the structure of a metalloprotein, the electronic structure of a metal centre in biology and its function.\n",
    "- List the structural and spectroscopic methods that will be described in this course.\n",
    "- Explain crystal field theory and apply it to derive approximate d-orbital splitting diagrams for some metal sites in biology.\n",
    "\n",
    "- nλ = 2dsinθ\n",
    "\n",
    "\n",
    "## 28.1 X-ray\n",
    "- Describe why you would want to undertake a protein X-ray crystallographic study.\n",
    "\n",
    "- overview of X-ray crystallography.\n",
    "  - Single Crystal Formation\n",
    "    - single crystal is exposed to X-rays from an X-ray source\n",
    "  - Data Collection\n",
    "    - scattered X-rays are captured on a detector (e.g., film)\n",
    "    - producing diffraction pattern (precession photograph)\n",
    "  - Data Processing\n",
    "    - diffraction pattern → electron density map\n",
    "  - Model Building\n",
    "    - map + sequence = atomic structure of the protein\n",
    "\n",
    "- features of X-ray diffractometer.\n",
    "X-ray Source\n",
    "rotating copper (Cu) anode \n",
    "generates X-rays\n",
    "50 keV electrons\n",
    "Mirrors\n",
    "X-ray mirrors focus and direct the X-ray beam toward the crystal.\n",
    "Crystal Holder\n",
    "The crystal is mounted on a rotating platform to expose it to X-rays at different angles.\n",
    "Detector\n",
    "Detects the diffracted X-rays\n",
    "creating a pattern (red dots) that can be analyzed to determine the crystal structure.\n",
    "\n",
    "- Compare and contrast magnification and diffraction as ways of viewing small objects.\n",
    "\n",
    "- Understand the implications of the phase problem in crystallographic determinations \n",
    "\n",
    "Some information is lost\n",
    "X-ray scattering is recorded\n",
    "wave amplitudes are retained\n",
    "relative phases are lost\n",
    "\n",
    "- outline approaches to solve it.\n",
    "\n",
    "Isomorphous replacement\n",
    "the insertion of a heavy metal atom into the protein without destroying the crystal. \n",
    "This can be achieved by soaking a protein crystal in a solution containing a heavy metal salt.\n",
    "\n",
    "Imagine in real space \n",
    "creating a 'difference crystal' by subtracting the native crystal from the derivative crystal. \n",
    "dominated by the high number of electrons associated with the heavy atom.\n",
    "\n",
    "PATTERSON SYNTHESIS\n",
    "leads to a map that resembles an electron density map. \n",
    "contains sufficient information to locate the position of the heavy atoms in the structure.\n",
    "\n",
    "calculation in which we set all of the phases to zero and just \n",
    "use the intensities of this difference diffraction pattern. \n",
    "\n",
    "\n",
    "- Appreciate that regular arrays of matter give rise to diffraction patterns that are related to the structure of the material by the Fourier Transform.\n",
    "\n",
    "- Derive the Bragg Equation and \n",
    "  - use it to determine the resolution of a crystallographic study.\n",
    "- process of refining a structure.\n",
    "\n",
    "- Assess the quality of a protein X-ray crystallographic study.\n",
    "\n",
    "## 28.2 XAS\n",
    "\n",
    "- origin of features\n",
    "- Appreciate the information content in each feature\n",
    "- advantages and disadvantages of EXAFS as a structural tool.\n",
    "- Use the Bond Valence Sum to analyse EXAFS data and predict EXAFS spectra for metal centres in proteins.\n",
    "- Understand the need for synchrotron sources for XAS (and protein crystallographic) studies.\n",
    "\n",
    "## 28.3 EPR\n",
    "\n",
    "- Describe the EPR experiment and how EPR spectra are presented.\n",
    "- EPR g- and A-values.\n",
    "- Classify EPR spectra as \n",
    "isotropic, \n",
    "axial or \n",
    "rhombic and \n",
    "\n",
    "- effect of anisotropy on the form of frozen solution EPR spectra\n",
    "\n",
    "- Provide an explanation for the variation of the g-value and how this relates to electronic structure\n",
    "\n",
    "- Provide interpretations for the EPR spectra of Cu(II) containing compounds and proteins\n",
    "\n",
    "# 50 HCC & CRLM\n",
    "\n",
    "## 50.0 Acquisition / Processing\n",
    "\n",
    "  - TCGA database \n",
    "    - RNA expression profiles\n",
    "    - clinical data\n",
    "    - via UCSC Xena platform\n",
    "\n",
    "  - survival analysis\n",
    "    - 368 samples\n",
    "  \n",
    "  - Processing\n",
    "    - TPM format\n",
    "    - log2-transformed\n",
    "\n",
    "  - sc\n",
    "\n",
    "    - sc dataset\n",
    "      - GEO\n",
    "      - 10 HCC\n",
    "      - 9 CRLM\n",
    "\n",
    "    - Quality control criteria\n",
    "      - mitochondrial content\n",
    "        - ＜ 20%\n",
    "      - UMI counts\n",
    "        - 200-20,000\n",
    "      - gene counts\n",
    "        - 200-5,000\n",
    "\n",
    "    - Seurat functions \n",
    "      - NormalizeData\n",
    "        - Data normalization\n",
    "      - FindVariableFeatures\n",
    "        - selection of highly variable genes\n",
    "        - 2000 genes\n",
    "      - ScaleData\n",
    "        - data transformation\n",
    "        - eliminating cell cycle effects with the parameters \n",
    "        - vars.to.regress = c(\"S.Score\", \"G2M.Score\")\n",
    "\n",
    "      - Dimensionality reduction methods \n",
    "        - UMAP\n",
    "        - TSNE\n",
    "        - Louvain\n",
    "\n",
    "    - Batch effect correction\n",
    "      - Harmony\n",
    "\n",
    "    - Differential gene expression\n",
    "      - Between\n",
    "        - clusters\n",
    "        - cell types\n",
    "      - FindAllMarkers function\n",
    "        - p-value < 0.05, log2FC > 0.25\n",
    "        - expression proportion > 0.1\n",
    "\n",
    "  - ST\n",
    "\n",
    "    - HCC ST\n",
    "      - 4 HCC samples\n",
    "    - CRLM ST\n",
    "      - 4 CRLM samples\n",
    "\n",
    "    - SpaceRanger software\n",
    "      - Post-quality control\n",
    "\n",
    "    - SCTtransform algorithm\n",
    "      - data normalization\n",
    "      - selection of highly variable genes\n",
    "      - data transformation\n",
    "\n",
    "    - average number of spots\n",
    "      - 4054\n",
    "    - average UMI\n",
    "      - 15317.4\n",
    "    - gene counts\n",
    "      - 3890.1\n",
    "    - mitochondrial content\n",
    "      - 4.5%\n",
    "\n",
    "    - Seurat\n",
    "      - Analysis\n",
    "      - visualization\n",
    "\n",
    "    - CARD algorithm\n",
    "      - conditional autoregression based deconvolution\n",
    "      - Deconvolution analysis\n",
    "\n",
    "    - single-cell annotations\n",
    "      - cell type predictions for each spot\n",
    "\n",
    "    - CARD software\n",
    "      - Visualization of cell types in ST\n",
    "\n",
    "    - Signature scores\n",
    "      - Seurat\n",
    "        - GSVA package\n",
    "        - AddModelScore\n",
    "\n",
    "## 50.1 sc Expression Profile\n",
    "\n",
    "  - Markers\n",
    "\n",
    "    - epithelial cell\n",
    "      - EPCAM\n",
    "      - KRT18\n",
    "      - KRT19\n",
    "      - CDH1\n",
    "\n",
    "    - fibroblast\n",
    "      - DCN\n",
    "      - THY1\n",
    "      - COL1A1\n",
    "      - COL1A2\n",
    "\n",
    "    - endothelial cell\n",
    "      - PECAM1\n",
    "      - CLDN5\n",
    "      - FLT1\n",
    "      - RAMP2\n",
    "\n",
    "    - T cell \n",
    "      - CD3D\n",
    "      - CD3E\n",
    "      - CD3G\n",
    "      - TRAC\n",
    "\n",
    "    - NK cell \n",
    "      - NKG7\n",
    "      - GNLY\n",
    "      - NCAM1\n",
    "      - KLRD1\n",
    "\n",
    "    - B cell\n",
    "      - CD79A\n",
    "      - IGHM\n",
    "      - IGHG3\n",
    "      - IGHA2\n",
    "\n",
    "    - mast cell \n",
    "      - KIT\n",
    "      - MS4A2\n",
    "      - GATA2\n",
    "\n",
    "  - Pearson correlation matrix\n",
    "    - correlation between two continuous variables\n",
    "      - clustering results\n",
    "      - cell type classification\n",
    "    - marker gene expression\n",
    "    - heatmap\n",
    "\n",
    "  - Marker gene expression\n",
    "    - heatmaps\n",
    "    - tSNE plots\n",
    "    - confirming cell-type identification\n",
    "\n",
    "  - Comparisons\n",
    "    - cell type composition\n",
    "    - patient origin\n",
    "    - cell count\n",
    "    - transcript detection\n",
    "    - CNV status\n",
    "\n",
    "## 50.2 ST\n",
    "\n",
    "  - HCC enriched\n",
    "    - LIHCT\n",
    "    - tumor cells\n",
    "  - CRLM enriched\n",
    "    - LIHC\n",
    "    - fibroblasts\n",
    "    - endothelial cells\n",
    "    - myeloid cells\n",
    "\n",
    "  - CNV detection\n",
    "    - InferCNV software\n",
    "    - CNV score distribution\n",
    "\n",
    "  - deconvolution analysis\n",
    "    - CARD algorithm\n",
    "    - based on single-cell data\n",
    "    - cell type distribution\n",
    "\n",
    "    \n",
    "  - H&E staining images\n",
    "\n",
    "  - tumor vs. non-tumor classification\n",
    "\n",
    "    - dimensionality reduction\n",
    "    - clustering analysis\n",
    "\n",
    "\n",
    "\n",
    "    - tumor cells\n",
    "    8 subgroups\n",
    "    LIHCT1\n",
    "    LIHCT3\n",
    "    LIHCT4\n",
    "\n",
    "    tSNE plot\n",
    "    subgroup composition\n",
    "    sample origin\n",
    "    CNV status\n",
    "    patient composition\n",
    "\n",
    "    Marker genes for each subgroup were calculated\n",
    "    expression heatmaps were created\n",
    "\n",
    "\n",
    "\n",
    "    - non-tumor cells\n",
    "    5 subgroups\n",
    "    endothelial cells\n",
    "    tumor cells\n",
    "    myeloid cells\n",
    "\n",
    "    LIHCT2 enrichment\n",
    "\n",
    "    tSNE plot\n",
    "    marker gene expression heatmaps\n",
    "\n",
    "## 50.3 Subclassification Analysis of Tumor Cells\n",
    "\n",
    "Tumor cells were isolated and subjected to dimensionality reduction and clustering analysis, resulting in 16 clusters. These clusters were then merged based on their marker gene functions, yielding 7 tumor cell subgroups (Figure 3A). A bar plot displays the cell composition of each sample, showing that HCC samples have a higher content of APOA2+NEO cells, while CRLM samples have a higher content of TFF3+NEO cells. Differential gene expression analysis was performed for each subgroup, and a heatmap of differentially expressed genes was generated (Figure 3B). GO Biological Process (GOBP) enrichment analysis was conducted for the marker genes of each subgroup, revealing their corresponding functions. We compared the CNV differences among different cell types from different sample origins, finding significant differences in CNV across most cell types (Figure 3C). Transcription factor analysis was also performed for each cell type, with the results visualized in a heatmap showing different transcription factors enriched in various cell types (Figure 3D).\n",
    "\n",
    "Subsequent cell trajectory analysis revealed multiple paths leading to different endpoints, indicating the potential existence of different subtypes. The starting branches contained a large number of APOA2+NEO cells and tumor cells from HCC (Figure 3E). From the pseudotime analysis, APOA2+NEO cells were primarily at the starting points, while most other cells were at the endpoints. Genes related to pseudotime were identified, and their expression heatmap was generated (Figure 3F). Scatter plots of pseudotime for six selected tumor marker genes (APOA2, HIST1H2AG, HMGB2, IGKV1-5, LCN2, TFF3) were created (Figure 3G).\n",
    "\n",
    "## 50.3 Cell Annotation Analysis\n",
    "\n",
    "  - tumor heterogeneity clustering\n",
    "    - epithelial cells\n",
    "    - immune cells\n",
    "    - fibroblasts\n",
    "\n",
    "\n",
    "## 50.4 Subgroup Analysis of Cell Populations\n",
    "\n",
    "  - further distinction for subgroups\n",
    "    - Immune cells\n",
    "    - epithelial cells\n",
    "    - fibroblasts\n",
    "\n",
    "  - standard Seurat workflow\n",
    "  - specific markers for each subgroup\n",
    "  - UMAP plots\n",
    "\n",
    "## 50.5 CNV Analysis of Single Cells\n",
    "\n",
    "  - CNVs in tumor cell subgroups\n",
    "    - InferCNV\n",
    "    - references\n",
    "      - fibroblasts\n",
    "      - endothelial cells\n",
    "    - identify malignant cells within subgroups\n",
    "\n",
    "## 50.6 Pseudotime Analysis of Single Cells\n",
    "\n",
    "  - Monocle2 software\n",
    "    - pseudotime analysis\n",
    "      - epithelial\n",
    "      - macrophage\n",
    "      - CAF cell\n",
    "\n",
    "  - DDRTree\n",
    "    - dimensionality reduction algorithm\n",
    "    - parameters set to default\n",
    "    - cell differentiation\n",
    "\n",
    "## 50.7 Transcription Factor Analysis\n",
    "\n",
    "  - SCENIC software\n",
    "    - transcription factor analysis across cell subgroups\n",
    "    - default parameters\n",
    "      - RcisTarget\n",
    "      - GRNBoost motif databases\n",
    "\n",
    "  - RcisTarget package\n",
    "    - overexpressed transcription factor binding motifs\n",
    "\n",
    "  - AUCell package\n",
    "    - scored the activity of each regulatory module for each cell type\n",
    "\n",
    "## 50.8 Cell-Cell Communication Analysis\n",
    "\n",
    "  - cell-cell communication\n",
    "    - CellChat package\n",
    "    - normalized gene expression matrix was imported\n",
    "\n",
    "  - Data preprocessing\n",
    "    - identifyOverExpressedGenes\n",
    "    - identifyOverExpressedInteraction\n",
    "    - ProjectData\n",
    "    - default parameters\n",
    "\n",
    "  - ligand-receptor interactions\n",
    "    - computeCommunProb\n",
    "    - filterCommunication\n",
    "    - computeCommunProbPathway\n",
    "\n",
    "  - cell communication network\n",
    "    - aggregateNet function\n",
    "\n",
    "## 50.9 Statistical Analysis\n",
    "\n",
    "  - R 4.1.3\n",
    "    - data processing\n",
    "    - statistical analysis\n",
    "    - plotting\n",
    "\n",
    "  - Pearson correlation coefficient\n",
    "    - correlation between two continuous variables\n",
    "\n",
    "  - chi-square test\n",
    "    - categorical variables\n",
    "\n",
    "  - continuous variables\n",
    "    - Wilcoxon rank-sum test\n",
    "    - T-test\n",
    "\n",
    "  - survminer package\n",
    "    - determine optimal cutoff values\n",
    "\n",
    "  - survival package\n",
    "    - Cox regression\n",
    "    - Kaplan-Meier analyses"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- “萧何”用户使用手册\n",
    "\n",
    "    - “分形”知识结构\n",
    "\n",
    "        - 模块化\n",
    "\n",
    "        - 自相似\n",
    "\n",
    "        - 分布与集成的辩证统一\n",
    "        \n",
    "        - “郡县”与“分封”制度的优劣\n",
    "\n",
    "    - 认知负荷理论\n",
    "\n",
    "        - 最小记忆原理\n",
    "\n",
    "    - “刻意”"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
